<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html401/loose.dtd">
<html>
<!-- Created on October 27, 2013 by texi2html 1.82
texi2html was written by: 
            Lionel Cons <Lionel.Cons@cern.ch> (original author)
            Karl Berry  <karl@freefriends.org>
            Olaf Bachmann <obachman@mathematik.uni-kl.de>
            and many others.
Maintained by: Many creative people.
Send bugs and suggestions to <texi2html-bug@nongnu.org>
-->
<head>
<title>Hex 7.0.9 User Manual</title>

<meta name="description" content="Hex 7.0.9 User Manual">
<meta name="keywords" content="Hex 7.0.9 User Manual">
<meta name="resource-type" content="document">
<meta name="distribution" content="global">
<meta name="Generator" content="texi2html 1.82">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
<!--
a.summary-letter {text-decoration: none}
blockquote.smallquotation {font-size: smaller}
pre.display {font-family: serif}
pre.format {font-family: serif}
pre.menu-comment {font-family: serif}
pre.menu-preformatted {font-family: serif}
pre.smalldisplay {font-family: serif; font-size: smaller}
pre.smallexample {font-size: smaller}
pre.smallformat {font-family: serif; font-size: smaller}
pre.smalllisp {font-size: smaller}
span.roman {font-family:serif; font-weight:normal;}
span.sansserif {font-family:sans-serif; font-weight:normal;}
ul.toc {list-style: none}
-->
</style>


</head>

<body lang="en" bgcolor="#FFFFFF" text="#000000" link="#0000FF" vlink="#800080" alink="#FF0000">

<a name="Hex-8_002e0_002e0-User-Manual"></a>
<h1 class="majorheading"><em>Hex</em> 8.0.0 User Manual</h1>
<a name="Protein-Docking-Using-Spherical-Polar-Fourier-Correlations"></a>
<h2 class="heading">Protein Docking Using Spherical Polar Fourier Correlations</h2>
<a name="Copyright-_00a9-1996_002d2013-David-W_002e-Ritchie"></a>
<h3 class="subheading">Copyright &copy; 1996-2013 David W. Ritchie</h3>
<img src="hex_manual_logo.jpg" alt="hex_manual_logo">
<a name="Dave-Ritchie"></a>
<h1 class="majorheading">Dave Ritchie</h1>
<a name="Team-Orpailleur_002c-INRIA-Nancy-Grand-Est_002c-LORIA_002c-54506-Vavdoeuvre_002dles_002dNancy_002c-France"></a>
<h3 class="subheading">Team Orpailleur, INRIA Nancy Grand Est, LORIA, 54506 Vavdoeuvre-les-Nancy, France</h3>




<a name="Introduction"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[ &lt; ]</td>
<td valign="middle" align="left">[<a href="#What-the-Heck-is-Hex_003f" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[ &lt;&lt; ]</td>
<td valign="middle" align="left">[ Up ]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h1 class="chapter">1. Introduction</h1>

<a name="index-Hex_002c-Dictionary-Definition"></a>
<p>&ldquo;hex /heks/ <em>v.</em> &amp; 
 <em>n.  US &ndash; v.</em> <strong>1</strong>. practise witchcraft.
                      <strong>2</strong>. bewitch. 
   &ndash; <em>n.</em> <strong>1</strong>. a magic spell. 
               <strong>2</strong>. a witch.
[GK <em>hex</em> six].&rdquo; 
(The Concise Oxford Dictionary).
</p>
<br>

<a name="index-Impossible-Things"></a>
<p>&ldquo;Why, sometimes I&rsquo;ve believed as many as six impossible things
      before breakfast.&rdquo; 
(Lewis Carroll).
</p>

<hr size="6">
<a name="What-the-Heck-is-Hex_003f"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Introduction" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-8_002e0_002e0" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">1.1 What the Heck is <em>Hex</em>?</h2>

<p><em>Hex</em> is an interactive molecular graphics program for calculating
and displaying feasible docking modes of pairs of protein and
DNA molecules. <em>Hex</em> can also calculate protein-ligand docking,
assuming the ligand is rigid, and it can superpose pairs of molecules using
only knowledge of their 3D shapes.
<em>Hex</em> has been available for about 12 years now, but as far as I
know, it is still the only docking and superpostion program to use
<em>spherical polar Fourier</em> (SPF) correlations to accelerate the 
calculations,
and its still one of the few docking programs which has built-in graphics
to view the results. Also, as far as I know, it is <b>the first protein
docking program to be able to use modern graphics processor units (GPUs)
to accelerate the calculations</b>.
</p>
<p>The graphical nature of <em>Hex</em> came about
largely because I wanted to visualise the results of such docking
calculations in a natural and seamless way, without having to export
unmanageably many (and usually quite big) coordinate files to one of the
many existing molecular graphics programs.
For this reason, the graphical capabilities in <em>Hex</em> are generally
relatively primitive compared to professional molecular graphics packages,
but the main aim here is to do docking, not to make publication-quality images.
</p>

<a name="index-Parametric-Functions"></a>
<a name="index-Surface-Skins"></a>
<a name="index-Electrostatics"></a>
<a name="index-Correlation"></a>
<a name="index-FFT"></a>
<a name="index-Fast-Fourier-Transform"></a>

<p>In <em>Hex</em>&rsquo;s docking calculations, each molecule is modelled using
3D expansions of real orthogonal spherical polar basis functions
to encode both surface shape
and electrostatic charge and potential distributions.
Essentially, this allows
each property to be represented by a vector of coefficients
(which are the components of the basis functions).
<em>Hex</em> represents the surface shapes of proteins using a two-term
surface skin plus van der Waals steric density model,
whereas the electrostatic model is derived from classical electrostatic theory.
By writing expressions for the <em>overlap</em> of pairs of
parametric functions, one can obtain an overall docking score
as a function of the six degrees of freedom in a rigid body docking search. 
With suitable scaling factors, this docking score can be
interpreted as an interaction energy, which we seek to minimise.
</p>
<p>Due to the special orthogonality property of the basis functions, the
<em>correlation</em> 
(or <em>overlap</em> as a function of translation/rotation operations)
between a pair of 3D functions can be calculated
using expressions which involve only the original expansion coefficients.
In many respects, this approach is similar to conventional
fast Fourier transform (FFT) docking methods which use Cartesian grid 
representations of protein shape and other properties, and which then 
use translational FFTs to perform the docking correlations.
However, the Cartesian grid approach only accelerates a
docking search in three translational degrees of freedom whereas the
SPF approach allows the effect of rotations and translations to 
be calculated directly from the original expansion coefficients.
</p>
<p>Even though the FFT part of a docking search may be fast, 
the overall speed of calculation still depends very much on the initial
&quot;set-up&quot; costs and the final &quot;post-processing costs&quot; of filtering
and perhaps clustering the results. <em>Hex</em> is fast because it
uses FFT correlations as much as possible, and because the
&quot;set-up&quot; costs are much lower in the SPF approach than in Cartesian
grid-based approaches. It also turns out that the FFT part of the
calculation maps very well to the GPU hardware. Thus, further speed-ups
can be expected if you have a suitable graphics card.
</p>
<p>Although it is not always easy to compare the performance of different docking 
algorithms because a lot depends on the size of the translational or 
rotation steps used, for example, I would still claim that <em>Hex</em> is at 
least 10 times faster than conventional FFT docking algorithms. 
<em>Hex</em> is also very easy to use. 
However, <b>to use <em>Hex</em> most effectively, it can sometimes require 
some thought 
when setting up the calculation, especially when setting up the starting 
orientations of the proteins to be docked.</b>
</p>
<p>In the spherical polar approach, it is natural to assign the six rigid
body degrees of freedom as five Euler rotation angles and an
intermolecular separation. 
Thus, in complete contrast to Cartesian based FFT approaches,
the rotational part of a docking search is 
the &ldquo;easy bit&rdquo; and modelling translations becomes the &ldquo;hard part.&rdquo;
Fortunately, however, only a few translations (typically about 40
steps of 0.75 &Aring;ngstrom)
are required to complete a six dimensional docking search.
One advantage of the spherical polar approach is 
that it is easy to constrain the docking search to one or both binding sites, 
when this knowledge is available, simply by constraining one or two of the 
angular degrees of freedom. This can reduce docking times to a matter of 
minutes on a modern workstation. 
</p>
<p>As of <em>Hex</em> version 5.0, the rotational part of the
search may be accelerated by one-dimensional (1D), three-dimensional (3D),
or five-dimensional (5D) rotational correlations. Earlier versions 
just used multiple 1D correlations. The 1D correlations are fast, 
but the 3D correlations in <em>Hex</em> 5.0 are about twice as fast. However,
for basic shape-based docking, 3D is also (counter-intuitively) faster
than 5D, largely due to the relatively high &quot;set-up&quot; cost of the 5D
correlation expression. 
</p>
<p>As of <em>Hex</em> version 6.0, the 1D and 3D calculations may optionally
be performed on one or more GPUs, which can give a considerable speed-up 
compared to using conventional CPUs. In this case (also counter-intuitively), 
1D correlations are much faster than 3D correlations on the GPU.
Additionally, the CPU-based calculations have also been re-written to
use multi-threading in order to support parallelisation on both Windows and
and Linux-based multi-core systems. Thus, significant performance improvements
can be expected from version 6.0 if you have suitable hardware.
</p>


<a name="index-Molecular-Similarity"></a>
<p>Closely related to the <em>protein docking problem</em>
is the <em>molecular similarity problem</em>
- i.e. how to find the relative orientation of a pair of similar molecules
such that some measure of the similarity (difference) between the molecules
is maximised (minimised). Both problems involve translating and rotating
one or both molecules into the desired orientation. However, to a first
approximation, the similarity problem can be reduced to a three
dimensional rotational search by initially placing both molecules in
a common coordinate system. Although <em>Hex</em> will remain primarily
a docking program, the 3D superposition calculations implemented in
<em>Hex</em> demonstrate the potential for performing fast 3D superpositions
using the SPF correlation approach. Work is in progress to develop this
approach further as a separate program for high throughput ligand screening.
</p>

<hr size="6">
<a name="New-Features-in-Hex-Version-8_002e0_002e0"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#What-the-Heck-is-Hex_003f" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#How-to-Read-this-Manual" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">1.2 New Features in <em>Hex</em> Version 8.0.0</h2>

<p>This version includes several bug fixes since 6.3 and 6.12. Behind the scenes,
the code has been re-structured quite considerably. 
But I have been too busy with other things to document all the changes. 
A lot of these changes were numbered &quot;Hex 7.X&quot;, but that was only released 
as a beta version to people who needed it.
</p>
<ul>
<li> The CUDA version now is built using CUDA 5.0.
</li><li> Fixed a nasty bug in CUDA for non-standard SPF expansion orders.
</li><li> Now uses FTGL for better fonts in the graphics window.
</li><li> The maximum correlation order has been increased to N=36.
</li><li> More of the translation matrix code is multi-threaded.
</li><li> A non-graphical version is available for headless workstations or clusters.
</li></ul>


<hr size="6">
<a name="How-to-Read-this-Manual"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-8_002e0_002e0" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">1.3 How to Read this Manual</h2>

<p>This Manual attempts to describe the main features of <em>Hex</em> by
(a) mentioning each feature at least once, and (b) by giving some examples
of how to use the program most effectively.
In the following sections, italic text is used to refer to <em>Menu Item</em>
and <em>Button Names</em>, or other important <em>concepts</em> within the
program. Typewriter text is used to
indicate a sequence of <code>menu selections</code> or <code>button actions</code> 
that are required to
perform a particular function. This type of text is also used when listing
the contents of some of the example files provided with the program.
Bold face text is used to highlight
<strong>file</strong> or <strong>directory</strong> names that refer to the installation and
use of <em>Hex</em>. 
If you really want to read the whole of the Manual, print out the
<a href="hex_manual.pdf">PDF version</a>.
Otherwise, browse it on-line as HTML using <em>Hex</em>&rsquo;s <code>Help</code> button.
</p>

<a name="index-GUI"></a>
<p>The graphical user interface (GUI) in <em>Hex</em> is intended to be
easy to use.
Generally, most actions cause an immediate effect on the display so that
once you&rsquo;ve loaded a protein or DNA molecule (preferably two such molecules),
most of the program&rsquo;s features can be understood by experiment. If you&rsquo;re
comfortable with this approach, go right ahead. Much of this Manual can
probably be skipped. Just keep an eye on the <em>Messages Window</em> 
for any information messages.
The parts that everyone should read are the sections on
Docking (see section <a href="#Docking-Molecules">Docking Molecules</a>), and
Superposition (see section <a href="#Molecular-Matching">Molecular Matching</a>).
If you want to try more than just one or two docking
calculations, you should also look at the section on using macros
(see section <a href="#Macros">Macros</a>)
and the contents of the <strong>examples</strong> directory 
(see section <a href="#Examples-Directory">Examples Directory</a>).
It should be noted that algorithmic details of the docking and superposition
calculations are not given here. Its assumed that you have copies of the
relevant publications (see section <a href="#References">References</a>).
</p>

<hr size="6">
<a name="Getting-Started"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#How-to-Read-this-Manual" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Short_002dCut-Buttons" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Getting-Started"></a>
<a name="Getting-Started-1"></a>
<h1 class="chapter">2. Getting Started</h1>

<p>If you haven&rsquo;t already done so, please download and install <em>Hex</em> from 
<a href="http://www.loria.fr/~ritchied/hex/"><em>Hex&rsquo;s Home Page</em></a>. 
See section <a href="#Installation-Guide">Installation Guide</a> for details. It should be easier to follow
this Manual if you have <em>Hex</em> up and running in front of you.
Been there, done it? Great! Lets get started...
<a name="index-PDB"></a>
<a name="index-Receptor"></a>
<a name="index-Ligand"></a>
<a name="index-Complex"></a>
<em>Hex</em> reads protein and DNA molecular structures from PDB-format files.
As of version 5.0, 
<em>Hex</em> can also read SDF-format small-molecule structure files. 
PDB files can be downloaded from the main Protein Data Bank repository
at <a href="http://www.rcsb.org/pdb/">Rutgers University</a>.
Up to three input files can be loaded into <em>Hex</em> at any one time.
These are treated
as a <em>receptor</em>, a <em>ligand</em> and a <em>reference complex</em>.
We&rsquo;ll ignore these distinctions for now and just load a single protein.
Go to the <em>File</em> menu and select:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">File ... Open ... Receptor
</pre></td></tr></table>

<a name="index-File-Selection-Panel"></a>
<p>When you release the
mouse button, a new <em>File Selection</em> menu panel should appear. Edit the
<em>Filter</em> text area to specify the directory containing your PDB file(s)
and press the <em>Filter</em> button. 
Alternatively, you could navigate to the 
<strong>hex/examples</strong> directory and load one of the provided example
PDB files.
You should now see one or more PDB files
listed in the <em>Files</em> box (if not, use the <em>File Selection</em>
controls to navigate to the <strong>hex/examples</strong> directory).
Pick a PDB file by double clicking on it (or by highlighting it and
picking <em>OK</em>).  You should now see a skeletal display of the
molecule(s) from your chosen PDB file. Figure 1 below shows the
scene obtained after loading the example file
<strong>3hfl_fv.pdb</strong>, which is the Fv fragment of the HyHel-5 antibody.
</p>
<div class="float">
 <img src="3hfl_fv_loaded.jpg" alt="3hfl_fv_loaded">

<br>
</div><p><strong><strong>Figure 1.</strong> The initial <em>Hex</em> scene showing the 
HyHel-5 antibody Fv domain as a molecular skeleton.
</strong>
</p>
<p>In Figure 1, you will also see a small
graphic in the top left of the scene that represents the
<em>(x,y,z)</em> coordinate axes.
Note that the <em>z</em>-axis
points towards the right and the <em>x</em>-axis points away from you, into the
scene.
<em>Hex</em> uses this convention because most displays have a greater width
than height and when two molecules are loaded it is convenient to assign
the <em>z</em>-direction to the intermolecular axis. Also, looking at a pair of
molecules &ldquo;side-on&rdquo; somehow seems more natural.
<a name="index-Mouse-Buttons"></a>
<a name="index-Zoom"></a>
<a name="index-Slider"></a>
Anyway, you can translate and rotate the scene with the mouse buttons.
Assuming you have a three-button mouse, dragging with button
1 (left button) translates the scene, and dragging with <code>Button 2</code> (middle
button) rotates it. This rotation is always about an axis perpendicular
to the direction of motion of <code>Button 2</code>. The right button also rotates
the scene, but about different axes. A right-left motion of this button
gives an anticlockwise rotation. 
You will probably find that <code>Button 2</code> feels
more natural for most movements but that occasionally <code>Button 3</code> is needed
to complete a manoeuvre. If you have a one-button mouse (e.g. on a Mac),
you can hold down the <em>Ctrl</em> or <em>Meta</em> (Command/Special) keys with 
<code>Button 1</code> (which might be the only button) to simulate buttons 2 and 3, 
respectively. 
On Linux, the <em>Ctrl</em> or <em>Alt</em> keys do the same thing.
Some window managers (e.g. Gnome) may interpret <em>Alt</em>-drag as a
window movement shortcut. There should be a window manager option
that allows this key binding to be changed to make the <em>Alt</em> available
to <em>Hex</em>.
The <em>Slider</em> on the lower left
border may be used to zoom the scene in and out. The other borders around
the graphics window contain various pull-down menus and control buttons.
These are described in more detail below.
</p>


<hr size="6">
<a name="Short_002dCut-Buttons"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Getting-Started" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Function-Buttons" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Short_002dCut-Buttons"></a>
<a name="index-Menu-Buttons"></a>
<a name="index-Buttons"></a>
<a name="Short_002dCut-Buttons-1"></a>
<h2 class="section">2.1 Short-Cut Buttons</h2>

<a name="index-Home-Position"></a>
<p>The column of buttons
on the right-hand border of the main window implement handy or frequently
used operations.
For example, <em>Hex</em> stores
a <em>Home Position</em> for the scene. Pressing the <em>House</em> button at
the top right border resets the scene to the current home position.
If you have oriented the molecule into a view that you like, you can make
this orientation the <em>Home Position</em> position by pressing the
<em>Lock</em> button, below the <em>Home</em> button.
The <em>Unlock</em> button resets the <em>Home Position</em> to its
the original setting (<em>z</em>-axis to the right, etc.).
<a name="index-Axes_002c-Axis"></a>
<a name="index-Intermolecular-Axis"></a>
The next button (the &ldquo;6.6&rdquo; icon) is a <em>text</em> toggle, 
to control the display of summary text in the graphics window. 
The <em>Axes</em> button (two arrows at right angles)
toggles the display of the coordinate axes
(useful for screen shots).
Below this is the <em>Intermolecular Axis</em> button (a double-headed arrow).
This draws a white line between the centroids of the receptor and ligand
molecules.
If only one molecule has been loaded, a short white line is still drawn
along the <em>z</em>-axis.
</p>
<a name="index-Solid-Model-Button"></a>
<a name="index-Solid-Surface-Button"></a>
<a name="index-Harmonics-Button"></a>
<a name="index-Spherical-Harmonic-Surfaces"></a>
<a name="index-Cartoon-Button"></a>
<p>The <em>Solid Models</em> button toggles the display of solid models, the
default being van Der Waals spheres. The type of solid model to display
may be selected from the <code>Solid Models</code> control panel
(see section <a href="#Solid-Models">Solid Models</a>).
The <em>Solid Surface</em> button toggles the calculation and display
of solid surfaces (see section <a href="#Solid-Surfaces">Solid Surfaces</a>).
Similarly, the <em>Harmonics</em> button (which looks like a balloon),
toggles the calculation and display of spherical harmonic
molecular surfaces (see section <a href="#Spherical-Harmonic-Surfaces">Spherical Harmonic Surfaces</a>),
and the <em>Cartoon</em> button (which is supposed to look like part
of a protein ribbon) toggles the cartoon displays.
</p>
<a name="index-Sidechains-Button"></a>
<a name="index-Solid-Motion-Button"></a>
<p>The <em>Sidechain</em> button
toggles the display of protein sidechains and DNA bases.
With large molecules, rotating and translating the scene can be much faster
if only the backbone trace is drawn. This is especially true if
solid models or surface meshes are being displayed.
The final <em>Solid Motion</em> button (the black square with an arrow)
can be used to toggle whether solid shapes or just bond skeletons are 
drawn when moving molecules within the scene.
</p>
<p>As you might expect, <em>Hex</em> is normally used to display two molecules 
at once.
Go to the <em>File</em> menu and click on:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Open ... Ligand
</pre></td></tr></table>

<a name="index-R-Slider"></a>
<a name="index-Intermolecular-Distance"></a>
<a name="index-Home-View"></a>
<p>to open another molecule.
You could open the same molecule for the <em>ligand</em> as you have opened for
the <em>receptor</em>, in which case the two molecules will be superposed. 
Figure 2, below, shows the result of loading the example lysozyme structure
<strong>3hfl_ly.pdb</strong>, which is the natural antigen (ligand) of the
HyHel-5 antobody.
In this example, 
the <em>R slider</em> bar on the bottom border was used to separate 
the molecules, and the van der Waals display mode and intermolecular
axis were enabled using the right-hand border short-cut buttons.
Try experimenting with the other short-cut buttons. 
You should now find that the mouse buttons translate and
rotate both molecules in the scene.
</p>
<div class="float">
 <img src="3hfl_fv_ly_vdw.jpg" alt="3hfl_fv_ly_vdw">

<br>
</div><p><strong><strong>Figure 2.</strong> A <em>Hex</em> scene showing the 
HyHel-5 antibody Fv domain and lysozyme in van der Waals mode, and with
the intermolecular axis drawn in white.
</strong>
</p>

<hr size="6">
<a name="Function-Buttons"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Short_002dCut-Buttons" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Keyboard-Keys" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Menu-Buttons-1"></a>
<a name="index-Buttons-1"></a>
<a name="Function-Buttons-1"></a>
<h2 class="section">2.2 Function Buttons</h2>

<a name="index-Picking"></a>
<a name="index-Editing"></a>
<a name="index-Pointer-Mode-Button"></a>
<p>The column of buttons
on the left hand border of the main window implement various
<em>picking</em> and <em>editing</em> functions.
These operations differ from the short-cut buttons in that
they generally involve changing the operating mode of the mouse. The inset
icon in the top left border displays the current <em>mouse mode</em>.
The default, <em>Pointer Mode</em>
(a left arrow), represents the basic scene manipulation
mode, described above.
You can revert to this mode by pressing the <em>Left Arrow</em>
button, which is the first button in the left-hand border.
</p>
<a name="index-ID-Button"></a>
<a name="index-Fonts"></a>
<a name="index-Hammer-Button"></a>
<p>The next <em>ID</em> button selects
atom picking mode. In this mode, picking an atom (i.e. pressing and releasing
<code>Button 1</code> over an atom) will identify the atom by displaying its chain,
residue and atom numbers. The atom coordinates and charge (Q) will also
be displayed in the <em>Messages</em> window.
The <code>Graphics ... Fonts</code> menu can be used to select the font
in which atom labels are displayed.
In stereo displays, picking uses the right-eye
object coordinates, since most people are right-eye-dominant.
Picking the same atom again will remove the atom ID label.
The <em>Hammer</em> button (the last button in this border) will clear all 
picked ID&rsquo;s. This button generally acts to <em>undo</em>, or <em>kill</em>, the 
previous operation(s), depending on the current <em>mouse mode</em>.
<a name="index-Origin-Editing-Button"></a>
<a name="index-Clipping-Button"></a>
<a name="index-Clipping-Clipping-the-Scene"></a>
Below the <em>ID</em> button is the <em>Manoeuvring Molecules</em> button
(which looks like a lightning bolt),
and below this is the <em>Origin Editing</em> button (a line with two crosses).
This is followed by the <em>Clipping Mode</em> button (three faces of a cube)
which controls scene clipping (see section <a href="#Clipping-the-Scene">Clipping the Scene</a>).
</p>
<a name="index-Origin-Editing"></a>
<a name="index-Orientation-Editing"></a>
<p>Manoeuvring molecules (also called <em>editing</em>) is something
you may need to do before running a docking calculation. 
For example, before docking, you should always ensure that the
molecules are located reasonably close to each other and that
their centroids are a reasonable distance (e.g. 30 &Aring;ngstroms) apart.
The white intermolecular axis connects the default
centroids of each molecule. 
If you have a good idea about the binding epitopes of one or
both proteins, then it would be a good idea to rotate one or both
proteins such that the intermolecular axis passes through the centre
of the epitope. 
<b>Before you start a docking calculation, you should always 
consider carefully the locations of the molecular centroids and the
relative orientations of your proteins with respect to the intermolecular 
axis.</b>
The <em>Origin Editing</em> button allows you to
move these centroids and the <em>Orientation Editing</em> button
allows you to rotate each molecule
about its centroid and to translate each molecule individually
relative to the scene.
These operations are described in more detail in
See section <a href="#Manoeuvring-Molecules">Manoeuvring Molecules</a>.
Note that the <em>R slider</em>
bar can be used to change the intermolecular separation permanently, 
as described above,
but it is recommended that you use this control only for temporary changes
to the scene (e.g. when viewing the result of docking or superposition
calculations) and that you make permanent changes only in <em>editing</em> mode.
</p>

<hr size="6">
<a name="Keyboard-Keys"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Function-Buttons" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Keyboard-Keys"></a>
<a name="Keyboard-Keys-1"></a>
<h2 class="section">2.3 Keyboard Keys</h2>

<a name="index-Full-Screen-Mode"></a>
<a name="index-Stereo-Displays"></a>
<a name="index-Orthographic-Projection"></a>
<a name="index-Perspective-Projection"></a>
<a name="index-Checkerboard"></a>
<a name="index-Escape"></a>
<a name="index-Control_002fC"></a>
<a name="index-Control_002fQ"></a>
<a name="index-Killing-Jobs"></a>
<a name="index-B-Key"></a>
<a name="index-P-Key"></a>
<a name="index-S-Key"></a>
<a name="index-Page_002fUp"></a>
<a name="index-Page_002fDown"></a>
<a name="index-Arrow-Keys"></a>
<p>All of the features in <em>Hex</em> are controlled using the GUI components.
However, when you place the mouse in the main graphics window,
several keyboard keys can be used as additional short-cut ways of
controlling the program. 
For example, <code>F</code> toggles between full-screen 
and windowed mode, 
<code>G</code> toggles the GUI border buttons and sliders in full-screen mode,
<code>S</code> toggles stereographics on or off (if stereo is available), 
<code>P</code> toggles between perspective
and orthographic projections, and <code>B</code> toggles a checker-board background
(which can help enhance perspective projections).
Hitting the escape key (<code>Esc</code>)
or typing <code>Ctrl/Q</code> will exit the program. 
Typing <code>Ctrl/C</code> will interrupt and exit the program even if it is 
in the middle of a calculation.
</p>
<p>When viewing the calculated orientations from a docking run, the
keypad <code>Page/Down</code> and <code>Page/Up</code> 
keys may be used to scroll through the predicted orientations, 
and the <code>Home</code> and <code>End</code> keys may be used to jump to the initial
(before docking) and last docked orientations, respectively.
The keyboard <em>arrow keys</em> 
are also mapped to these actions
(<code>Down</code>, <code>Up</code>, <code>Left</code>, <code>Right</code>, respectively).
</p>

<hr size="6">
<a name="Molecular-Graphics"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Keyboard-Keys" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Basic-Display-Styles" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Getting-Started" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Molecular-Graphics"></a>
<a name="Molecular-Graphics-1"></a>
<h1 class="chapter">3. Molecular Graphics</h1>


<a name="index-Molecule-Control-Panel"></a>
<a name="index-Solid-Models"></a>
<a name="index-Solid-Surfaces"></a>
<a name="index-Spherical-Harmonic-Surfaces-1"></a>
<a name="index-Dot-Surfaces"></a>
<a name="index-Graphics-Menu"></a>
<a name="index-Fog"></a>
<p><em>Hex</em> can display protein and DNA molecules in several ways.
The simplest display styles are
controlled by the settings in the <em>Molecule Control</em> panel and some
of the short-cut buttons on the right-hand border.
Molecules can also be displayed as
<em>solid models</em> (see section <a href="#Solid-Models">Solid Models</a>),
<em>solid surfaces</em> (see section <a href="#Solid-Surfaces">Solid Surfaces</a>),
<em>spherical harmonic surfaces</em> (see section <a href="#Spherical-Harmonic-Surfaces">Spherical Harmonic Surfaces</a>), and
<em>dot surfaces</em> (see section <a href="#Dot-Surfaces">Dot Surfaces</a> panel).
These display types are controlled by additional control panels under
the main <em>Graphics</em> menu.
This menu also contains options to
allow the foreground and background colours to be selected,
the position and colour of a directional light to be modified,
and a simple <em>Fog</em> effect to be applied.
</p>


<hr size="6">
<a name="Basic-Display-Styles"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Solid-Models" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Basic-Display-Styles"></a>
<a name="Basic-Display-Styles-1"></a>
<h2 class="section">3.1 Basic Display Styles</h2>

<a name="index-Display-Styles"></a>
<a name="index-Molecule-Control-Panel-1"></a>
<a name="index-Colour-Schemes"></a>
<a name="index-Bond-Widths"></a>
<p>The basic appearance of protein and DNA molecules is controlled by
the <em>Molecule Control</em> panel.
The default style is to draw all covalent bonds
as a skeleton, with each half-bond colour-coded by atom type.
Probably the two most useful controls in this panel are the
<em>Show Sidechains</em> toggle (which displays a protein in backbone-only or
backbone+sidechain mode) and the <em>Colour Scheme</em> selector,
which allows a limited degree of control over the colours
used to draw each molecule.
Line widths may be changed using the <em>Bond Widths</em> control.
</p>
<a name="index-Polar-Hydrogens"></a>
<a name="index-Hydrogen-Atoms"></a>
<a name="index-Sidechains-Button-1"></a>
<p>When a protein or DNA molecule is loaded, <em>Hex</em> adds any missing polar
hydrogen atoms using a set of templates from the
<strong>atom_templates.dat</strong> file in the <strong>data</strong> directory.
Polar hydrogens can be displayed using the <em>Show Hydrogens</em> toggle.
The <em>Show Sidechains</em> toggle has an equivalent short-cut button on the
right-hand border of the main window (which is supposed to look like a
phenylalanine residue).
Setting <em>Show Sidechains</em> to <em>off</em>
(which also turns off hydrogen atom display) makes
manipulating complex scenes easier.
</p>
<a name="index-Radial-Cutoff"></a>
<a name="index-Viewing-Large-Molecules"></a>
<p><a name="Radial-Cutoff-Distance"></a>
When viewing large molecules, its sometimes useful to draw only those atoms
within a given distance of the molecular origin.
The <em>Apply Radial Cutoff</em> button may be used to enable this behaviour,
and the <em>Radial Cutoff</em> slider may be used to adjust the radial distance
threshold. Distances are relative to the current scene rotation centre,
which may be changed using <em>Origin Editing</em> button in the left-hand border
(the crossed circle icon).
</p>
<a name="index-Colour-Schemes-1"></a>
<a name="index-Colour-Files"></a>
<p>The <em>Colour Scheme</em> selector allows the molecular skeleton to be
coloured from a fixed palette of colours or by using very simple commands
in a text file to specify particular
colours for specific residues. Some examples of colour files can be found
in the <strong>examples</strong> directory. The general format for a colour file is:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">chain  first_residue  last_residue  colour_name
</pre></td></tr></table>

<p>where <code>colour_name</code>
is any one of the X11 colour names.
The colour names may be found by listing the
<strong>rgb_colours.dat</strong> file in the <strong>data</strong> directory.
Alternatively, each time a colour is picked from the <em>Colour Chart</em>
its name is printed to the terminal window.
Shown below is an example colour file <strong>3hfl.col</strong> 
from the <strong>examples</strong>
directory, which specifies the colours
to use when drawing an antibody/antigen complex. Subsets of the residues in
each chain are assigned different colours to highlight the positions of the
antibody hypervariable loops:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">L 1  214 grey
L 25  33 red
L 90  97 blue

H 1  225 RoyalBlue1
H 25  33 red
H 52  56 yellow
H 95 102 green

Y 1  999 OrangeRed
Y 41  53 yellow
Y 67  70 red
Y 84  84 blue
</pre></td></tr></table>

<p>Colour files may also be used to colour atoms explicitly,
using the syntax:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">atom_name   first_atom   last_atom    colour_name
</pre></td></tr></table>


<a name="index-Molecule-Control-Panel-2"></a>
<a name="index-Symmetry-Types"></a>
<a name="index-Crystal-Symmetry"></a>
<a name="index-Biological-Symmetry"></a>
<p>The <em>Molecule Control</em> panel also contains toggles to display 
<em>molecular centroids</em> and <em>average molecular ellipsoids</em>. 
The ellipsoids are
calculated from low order (L=2) spherical harmonic surfaces.
This control panel also allows the display of any
crystallographically related molecules whose coordinates are not given 
explicitly in the PDB file. Use the <em>Symmetry Type</em>
selector to select these additional symmetry elements. 
Selecting <em>Crystal</em> causes the scene to be generated using the 
SMTRY transformations from the PDB file,
<em>Biological</em> selects the transformations from the <em>BIOMT</em> records,
and <em>None</em> (the default) just uses the explicit atoms coordinates as 
usual. 
Note that atom coordinates are only replicated at display time using OpenGL 
display lists, so that even quite large molecules can be displayed 
reasonably quickly. 
A fun one to try is the tobacco mosaic virus (PDB code 1RMV), especially when 
low resolution harmonic surfaces are enabled. Another nice structure is the 
viral coat protein of the phi-X174 bacteriophage (PDB code 1CD3), which just 
happens to be the 10,000th entry in the PDB.
By selecting the <code>File ... Save ... Symmetry ... Biological</code> option, 
<em>Hex</em> will write out all the symmetry-related atom
coordinates when saving a structure to disc. This is useful
in cases where the biologically active molecule is a dimer (or has some
sort of symmetry) but only the minimal coordinate set appears in the PDB
data file. Some further editing of the output file (e.g. to rename 
duplicated chains) may be required if the chains are to be individually
coloured, for example.
</p>

<hr size="6">
<a name="Solid-Models"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Basic-Display-Styles" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Cartoons" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Solid-Models-1"></a>
<a name="Solid-Models-1"></a>
<h2 class="section">3.2 Solid Models</h2>


<a name="index-Solid-Model-Control-Panel"></a>
<a name="index-van-der-Waals-Spheres"></a>
<p>The <code>Graphics ... Solid Models</code> control panel contains several options
for drawing lit solid models of molecular scenes, including van der Waals
spheres, licorice, and tube representations. Aromatic rings may be
drawn as hexagonal or pentagonal &ldquo;blocks&rdquo;, which can produce some
interesting displays, especially  when viewing DNA molecules. 
The position and colour of the light may be controlled using the controls 
in the <code>Graphics ... Lighting</code> control panel. 
</p>

<hr size="6">
<a name="Cartoons"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Solid-Models" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Solid-Surfaces" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Cartoons"></a>
<a name="Cartoons-1"></a>
<h2 class="section">3.3 Cartoons</h2>


<a name="index-Cartoon-Control-Panel"></a>
<a name="index-Ribbon-Cartoons"></a>
<a name="index-Stride"></a>

<p>The <code>Graphics ... Cartoons</code> control panel contains several options
for drawing proteins as ribbon cartoons. The default behaviour is to draw
a cartoon over the molecular skeleton. To view only the cartoon, you 
can disable the molecular skeleton using the <em>Display Molecule</em> button in 
the <code>Controls ... Molecule</code> panel. There are several options on the
<em>Cartoon Control</em> panel which allow the colour, width, and thickness
of ribbon cartoons to be controlled. Different parts of the ribbon can
also be coloured according to secondary structure type (i.e. helix, sheet, 
turn). The type of secondary structure is currently calculated using 
Frishman and Argos&rsquo; 
<a href="ftp://ftp.ebi.ac.uk/pub/software/unix/stride"><strong>Stride program</strong></a>. 
</p>

<hr size="6">
<a name="Solid-Surfaces"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Cartoons" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Spherical-Harmonic-Surfaces" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Solid-Surfaces-1"></a>
<a name="Solid-Surfaces-1"></a>
<h2 class="section">3.4 Solid Surfaces</h2>


<a name="index-Gaussian-Densities"></a>
<a name="index-Gaussian-Surfaces"></a>
<a name="index-Marching-Tetragons"></a>
<p>Solid molecular surfaces may be drawn using the
<code>Graphics ... Solid Surfaces</code> control panel. These surfaces are 
calculated using a novel <em>marching tetragons</em> algorithm to contour
a Gaussian density representation of the atoms of each molecule. 
The surface skins used in the docking correlations 
are calculated using this Gaussian density approach. 
</p>


<a name="index-Colour-Ramp"></a>
<a name="index-Electrostatic-Potential"></a>
<a name="index-Charge-Density"></a>
<p>The <em>Colour Mode</em> selector allows surfaces to be 
coloured by atom colour (the default), electrostatic
potential or charge density, or by using the &ldquo;classic&rdquo; blue-red colour
scheme used in earlier versions of <em>Hex</em>.
For electrostatic surfaces, a <em>colour ramp</em> is calculated
which initialy shows positive potentials/charges in blue, negative 
values in red, and intermediate values in white. The 
<code>Surfaces ... Colour Ramp</code> button may be used to activate a 
simple graphical controller for the colour ramp.
The electrostatic potential and charge density displays are calculated
from the <em>in vacuo</em> global charge density expansion using the
current speherical polar docking expansion order, N. 
</p>


<a name="index-Relative-Permittivity"></a>
<p>The colours in a charge density 
display may appear somewhat &ldquo;washed-out&rdquo; compared to (perhaps more
familiar) potential displays. This is because the potential
is calculated directly from the charge density using Poisson&rsquo;s equation,
and the del-squared form of this equation strongly emphasises any local 
variation in the charge density.
So if the potentials &ldquo;look right&rdquo; then the charge density is also correct.
NB. <em>Hex</em> uses a relative permittivity value of 8, instead of the
more usual value of around 80. So, in addition to <em>Hex</em>&rsquo;s <em>in vacuo</em> 
assumption, the numerical values for the potential are likely to differ 
from other software.
</p>


<a name="index-Cavities"></a>
<p>Some molecules may have internal cavities, and these can produce 
one or more contour surfaces in additional to the primary external surface.
Other molecular structures (e.g. structures with many waters, or 
well-separated domains) may also give multiple surfaces when contoured.
As the contouring algorithm implicitly produces positively oriented
surfaces (outward normals, positive total volumes), it is convenient to
assign each calculated surface to one of two possible classes:
<em>Primary</em> (positive volumes, outward normals), or 
<em>Secondary</em> (negative volumes, inward normals).
By default, only <em>Primary</em> surfaces are displayed, 
but this may be changed using the <em>Draw Surface</em> control.
</p>


<a name="index-Gaussian-Surfaces-1"></a>
<a name="index-Sigma-Surface"></a>
<a name="index-Tau-Surface"></a>
<a name="index-Docking-Control-Panel"></a>
<p>In addition to the default Gaussian surface, the <em>Surface Type</em>
selector provides the option to draw contoured density functions
of the <em>Sigma</em> and <em>Tau</em> surface functions which are used by
the docking algorithm. These two density functions allow the shape 
functions used in docking calculations to be visualised.
<em>Sigma</em> is the external skin density function, typically calculated with
a probe radius of 1.4&Aring;, and <em>Tau</em> is the interior density function
(effectively, the van der Waals volume).
For relatively small molecules, and when using high expansion orders (N),
the <em>Tau</em> density can be seen to give a remarkably good representation 
of the initial (atomic) Gaussian density.
The order of the 3D expansion (default N=25) is taken from the 
<code>Docking ... Final Search</code> parameter in the <em>Docking Control</em> panel. 
Both the <em>Sigma</em> and <em>Tau</em> surfaces are contoured using
a hard-wired density value of 0.25.
However, reconstructing each 3D density from the shape expansion coefficient 
vectors is a relatively expensive calculation.
This display mode is mainly intended to illustrate the internal representations
used in docking. Figure 3 shows some example shape density isosurfaces.
In this figure, the highest polynomial power in the SPF expansion, L,
is related to the order N in the Docking Control panel by L=N-1.
</p>

<div class="float">
 <img src="all_hfl_6_L_small.jpg" alt="all_hfl_6_L_small">

<br>
</div><p><strong><strong>Figure 3.</strong> 
Illustration of the SPF steric density representations at various 3D
expansion orders for the complex between the
HyHel-5 antibody Fv domain (left) and quail lysozyme (right).
From top left to bottom right:
steric density isosurfaces
shown at expansion orders L=12, 16, 20, 24, and 26,
with the subunits separated by 15 &Aring; to give a full view of each domain.
The bottom right pair shows the van der Waals surfaces
from which the SPF expansions are derived.
</strong>
</p>

<a name="index-Translation-Matrices"></a>
<p>The <em>Sigma/Tau Shift</em> slider controls how the shape (and electrostatic)
expansion coefficients are computationally translated using the T(R)
translation matrices prior to reconstructing the surface from translated
expansion coefficients. Thus it is possible to view graphically
how the representation used in the docking correlation degrades with 
increasing distance from the origin.  
In this display mode, the receptor surface is translated in the negative 
z direction, and the ligand surface is translated in positive z for
ease of viewing. You should see a blurring of those portions of each
molecule furthest from the origin, although the shapes of the surface 
regions near the origin are very well preserved after translation.
NB. In <em>Hex</em>&rsquo;s docking correlations, all computational translations 
are applied entirely to the receptor.
</p>
<p>Please be aware that drawing surfaces can use a lot of memory. Drawing the
surface of a large molecule using a fine (0.25-0.5 &Aring;ngstrom) grid can use 
<em>hundreds of megabytes of memory</em>. Attempting to draw a complex surface 
on a machine with insufficient memory could cause the machine to hang.
</p>


<hr size="6">
<a name="Spherical-Harmonic-Surfaces"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Solid-Surfaces" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Dot-Surfaces" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Spherical-Harmonic-Surfaces-2"></a>
<a name="Spherical-Harmonic-Surfaces-1"></a>
<h2 class="section">3.5 Spherical Harmonic Surfaces</h2>

<p>In <em>Hex</em> version 5.0, 
spherical harmonic (SH) molecular surfaces are generated from the marching
tetragons surface described above. Essentially, all of the vertices
from the trangulated surface are first projected onto an icosahedral 
tesselation of the sphere. If more than one surface point maps
to the same tesselation vertex, then the point with the largest 
radial distance from the origin is selected. Hence, a bounding envelope
of sample points is constructed. Any gaps in the tesselation which might
remain are &ldquo;filled in&rdquo; by interpolating from values of nearby 
non-null tesselation vertices. 
<em>Hex</em> doesn&rsquo;t use spherical harmonic surfaces in any of its
calculations. It only displays them. However, the
commercial ParaFit program (also written by Dave Ritchie) can rapidly 
superpose and spherical harmonic surfaces and surface properties 
calculated by the ParaSurf program. Both ParaSurf and ParaFit are
available from 
<a href="http://www.ceposinsilico.com/"><strong>Cepos Insilico Ltd.</strong></a>.
<em>Hex</em> can read ParaSurf SDF files, and it can display
the key ParaSurf spherical harmonic surface properties.
</p>

<a name="index-Harmonic-Surface-Control-Panel"></a>
<a name="index-Spherical-Harmonic-Surfaces-3"></a>
<p>The <em>Harmonic Surface</em> control panel controls the way in which 
spherical harmonic molecular surfaces are calculated and displayed.
You can view the molecular envelope by selecting:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">Graphics ... Harmonic Surface ... Enable Surface (toggle on)
</pre></td></tr></table>


<a name="index-Tessellation-Mesh-Order"></a>
<p>The icosahedral sampling resolution is controlled by 
the <em>Mesh Order</em> slider.
The <em>order</em> of the spherical harmonic expansion can be selected using the
<em>Order (L)</em> slider in the <em>Harmonics Control</em> panel.
The SH envelope may be displayed in different styles and colours using the 
<em>Display Mode</em> and <em>Line Colours</em> selectors.
Figure 4 shows the SH surfaces for the antibody/lysozyme example, obtained
by pressing the <em>Harmonics</em> short-cut button on the right hand
border.
</p>
<div class="float">
 <img src="3hfl_fv_ly_harmonics.jpg" alt="3hfl_fv_ly_harmonics">

<br>
</div><p><strong><strong>Figure 4.</strong> Illustration of spherical harmonic surfaces to order
L=12 for the HyHel-5 antibody Fv domain (left) and lysozyme (right).
</strong>
</p>
<hr size="6">
<a name="Dot-Surfaces"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Spherical-Harmonic-Surfaces" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Changing-the-Scene-Origin" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Dot-Surfaces-1"></a>
<a name="Dot-Surfaces-1"></a>
<h2 class="section">3.6 Dot Surfaces</h2>

<p>Dot surfaces are somewhat of a historical remnant in <em>Hex</em>. 
They used to be used as an approximate but fast way
to calculate the <em>molecular surface</em>, the 
<em>solvent-accessible surface</em> and the <em>van der Waals surface</em>
for docking.  
However, docking and superposition
calculations now use a much more accurate method based on
contouring Gaussian density functions to calculate these two surfaces.
Nonetheless, it is still sometimes useful to be able to draw a dot surface
because <em>Hex</em> uses all the atoms from an input PDB or SDF file to draw
the surfaces. Hence dot surfaces may be used to verify the presence of atoms
that might not be recognised or drawn properly in other drawing modes.
Like the SH surfaces, dot surfaces are now calculated directly from the 
marching tetragons surface, as described above.
</p>

<hr size="6">
<a name="Changing-the-Scene-Origin"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Dot-Surfaces" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Clipping-the-Scene" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Changing-the-Scene-Origin"></a>
<a name="Changing-the-Scene-Origin-1"></a>
<h2 class="section">3.7 Changing the Scene Origin</h2>

<a name="index-Scene-Centre"></a>
<a name="index-Molecular-Centroids"></a>
<a name="index-Orientation-Control-Panel"></a>
<p>Normally, rotation and scale (zoom) operations are applied relative to the
scene origin. When just one molecule is loaded, the scene origin coincides
with the centroid of the molecule. However, when two molecules are loaded,
the scene origin is taken as the midpoint between the two molecular centroids.
This behaviour can be changed by using the first selector
in the <em>Orientation Control</em> panel to select a given molecular
centroid or a specific atom to act as the origin.
The <em>Select Origin</em> button (a circle with a diagonal cross through it)
on the left hand border may also be used to select the zoom/rotation centre.
</p>
<a name="index-Docking-Motion"></a>
<p>Although docking calculations normally move both the ligand and the receptor,
the default display behaviour is to keep the receptor fixed in space and
to assign all the motion to the ligand as successive docking solutions are
viewed. This behaviour can be changed using the <em>Docking Motion</em>
selector in the <em>Orientation Control</em> panel.
</p>

<hr size="6">
<a name="Clipping-the-Scene"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Changing-the-Scene-Origin" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Animation" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Clipping"></a>
<a name="Clipping-the-Scene-1"></a>
<h2 class="section">3.8 Clipping the Scene</h2>


<p>The <em>Clipping</em> button (three faces of a cube) on the left hand border 
selects <em>scene clipping</em> mode.
You can define up to six clipping planes in the scene, several of which may be
moved together using a single slider.
However, controlling clipping is probably
the most difficult part of the program to master, since this
involves the use of all three mouse buttons and several keyboard keys,
in addition to the basic graphical widgets.
</p>
<p>When scene clipping mode is selected, you should see a wire frame drawing of 
six skeletal clipping planes centred in the scene. 
A new slider control is also drawn in the bottom right border. 
This new slider may be used to move selected clipping planes through the 
scene, in plane-perpendicular directions.
</p>
<p>In clipping mode, 
the mouse buttons may be used to translate and rotate the clipping
skeleton in the usual way. Note however, that the rest of the scene remains
fixed. You can use the keyboard <code>Space Bar</code> to toggle between moving the
skeleton about the scene and moving the scene about the skeleton.
</p>
<p>Picking one of the skeletal planes 
(using <code>Button 1</code> for picking) <em>selects</em> that plane and 
<em>attaches</em> it to the scene. 
The plane is now drawn in white, 
either as a filled shape or as an outline, 
depending on which face you happen to be viewing.
The mouse buttons now move only this plane relative to the scene (or, if
you have hit the <code>Space Bar</code>, 
the mouse moves the plane and the scene together).
The filled face will become the clipping face of that plane.
</p>
<p><code>Button 2</code> may be used to <em>activate</em>
the selected plane (just press <code>Button 2</code> anywhere in the scene). That is,
every part of the scene above the filled face should become clipped. Pressing
<code>Button 2</code> again toggles clipping. Try moving the clipping plane though
the scene using the right slider, 
or by using the mouse and the <code>Space Bar</code> to control the movement. 
Try using the <code>Delete Key</code> (keyboard backspace)
to toggle the sense of the clipping plane. Once you are happy with the
position of the clipping plane, you can revert to <em>Pointer Mode</em> 
(select the <em>Left Arrow</em> button) or you can proceed to pick and 
activate further clipping planes. 
In pointer mode, the right hand slider continues to operate the
most recently activated clipping plane.
</p>
<a name="index-Clipping-Slider"></a>
<p>It is also possible to make the slider move two planes simultaneously. 
Having activated a clipping
plane, now try picking it with <code>Button 3</code>. This draws the plane in yellow
and ties its motion to the slider. Now pick and activate the clipping plane
(using Buttons 1 and 2) on the opposite face of the clipping skeleton. 
The right
slider should now move both planes together through the scene. You can tie
multiple planes to the slider, although selecting two perpendicular planes
is probably the most sensible option. As you might expect, picking
a plane with <code>Button 3</code> for a second time unties it from the slider.
</p>

<p>At any time while in clipping mode, you can undo the most recent operation 
using the <em>Hammer</em> button. 
Three successive clicks on the  <em>Hammer</em> will clear all clipping 
settings and will return the display to an unclipped scene.
</p>

<a name="index-Keyboard-Keys-1"></a>
<p>If necessary, you can use the keyboard <code>Plus</code> and <code>Minus</code> keys to 
zoom the clipping skeleton independently of the scene.
If the scene contains molecular surfaces (dots, lines or polygons), 
you can use the keyboard <code>Equals</code> key to toggle between clipping both 
the molecule and its surface (the default) and clipping just the surface,
leaving the molecular skeleton fully visible.
</p>

<hr size="6">
<a name="Animation"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Clipping-the-Scene" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Full-Screen-Mode" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Animation"></a>
<a name="Animation-1"></a>
<h2 class="section">3.9 Animation</h2>

<a name="index-Animation-Control-Panel"></a>
<p>The <em>Animation Control</em> panel allows
you to run a &ldquo;movie&rdquo; of either the results of a docking calculation, or
a sequence of models from an NMR structure, for example, and it
controls how the scene &ldquo;spins&rdquo; if you perform a mouse drag-release action. 
</p>
<a name="index-Frame-Rate"></a>
<a name="index-Movie-Type"></a>
<a name="index-Receptor-Models"></a>
<a name="index-Ligand-Models"></a>
<p>After a
docking run, pressing <em>Start</em> in the <code>Graphics ... Animation</code> 
control panel will cause <em>Hex</em> to draw each docking solution in turn. 
The rate at which orientations are drawn is controlled by the 
<em>Frame Rate</em> slider. Similarly, if you have loaded a PDB file
that contains several NMR model structures, setting the
<em>Movie Type</em> to <em>Receptor Models</em> or 
<em>Ligand Models</em>, as appropriate,
and pressing <em>Start</em> will show a movie of the sequence of models.
The orientations, or frames, of both types of movie may be shown just
once, or cycled forever. The movie can be stopped at the current
frame by pressing the <em>Stop</em> button. The speed of the movie is
controlled by the  <em>Frame Rate</em> slider.
</p>
<p>The maximum frame rate achievable will depend on the speed of the CPU
and on whether your machine has hardware-accelerated graphics. The
<em>Frame Rate</em> thus defines a requested rate. Actual performance
will vary. But please note, <em>Hex</em> never hogs the CPU in a tight 
loop on input events, even during an animation, as do many other programs 
(which shall remain nameless). Thus it should still be possible to perform
other activities while a movie is running without things becoming sluggish.
</p>
<a name="index-Spin-Angle"></a>
<p>You can <em>spin</em> the scene by
dragging with <code>Button 1</code> or <code>Button 2</code>
to start a rotation and by releasing
the button while still moving the mouse to initiate a continuous rotation 
(spinning). 
When spinning, the rotatation angle is incremented by the 
current value in the <em>Spin Angle</em> slider, and new rotational increments
are drawn at the current <em>Frame Rate</em>. If desired, the mouse button
action that initiates spinning can be enabled/disabled with the 
<em>Enable Spinning</em> toggle. Spinning is enabled by default, and it is
possible to spin a movie (if thats what you really want to see!).
</p>
<p>Any of the usual user interface controls may be used while an animation
is running. If using these controls changes the graphical complexity of
the scene, <em>Hex</em> will take a second or two to adjust the drawing 
speed back to the requested frame rate.
</p>


<a name="index-GUI-Control-Toggle"></a>
<a name="index-F-Key"></a>
<a name="index-G-Key"></a>


<hr size="6">
<a name="Full-Screen-Mode"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Animation" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Stereo-Displays" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="Full-Screen-Mode-1"></a>
<h2 class="section">3.10 Full Screen Mode</h2>

<p>Personally, I think full-screen mode is the best way to view molecular
graphics scenes. You can press <code>F</code> to switch back and forth 
between full-screen and window-manager modes. By default, <em>Hex</em> starts
in window-manager mode.
You can also press <code>G</code> 
at any time to toggle the display of the GUI controls when in full-screen mode.
<code>G</code> only toggles the setting, not the display. So you won&rsquo;t see the
effect until you enter full-screen mode.
</p>
<p>In both the Linux and Windows versions, you should always use <code>F</code> to
get full-screen mode because this forces the window border off.
Conversely, don&rsquo;t use the window manager&rsquo;s &ldquo;maximise&rdquo; control because 
this forces the window to keep a border, and <code>F</code> won&rsquo;t then work.
</p>
<p>In Windows XP, you will probably still see the Windows Taskbar in 
full-screen mode. To get rid of this, open the
<code>Taskbar and Start Menu</code> in the <em>Windows XP Control Panel</em> and 
uncheck the <em>Keep the taskbar on top of other windows</em> item. I don&rsquo;t 
know whether this feature is available on earlier versions of Windows.
</p>
<p>In Mac OS X, full-screen mode does not override the Mac Taskbar and 
Dock panels.
Hence it is best <em>not</em> to use <em>Hex</em>&rsquo;s full
screen mode, but instead use the window manager&rsquo;s &ldquo;maximise&rdquo; button
to ensure the graphics window is well-behaved.
</p>
<p>In Linux, full-screen GUI mode with the Gnome window manager
is &ldquo;quirky&rdquo;. Everything works as it should, except that <em>Hex</em>
control panels can&rsquo;t be re-raised over the full-screen window after the
control panel loses mouse focus (i.e. when the cursor leaves the panel). 
You will need to revert to window-manager mode (<code>F</code>) to see everything
again! These problems seem to be due to a strange interaction between
the Gnome window manager and the FLTK interface toolkit.
</p>


<hr size="6">
<a name="Stereo-Displays"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Full-Screen-Mode" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Stereo-Displays-1"></a>
<a name="Stereo-Displays-1"></a>
<h2 class="section">3.11 Stereo Displays</h2>

<a name="index-nVidia-Graphics-Cards"></a>
<a name="index-nVidia-Drivers"></a>

<p>Stereo is known to work on Windows XP and RedHat Linux 8.0 
(kernel version 2.4.18-14, XFree86 version 4.0.0) or later with an nVidia
Quadro4 XGL700 card, and using a recent nVidia driver (1.0.3123 or later).
For best results with <em>Hex</em> and nVidia cards, please use the latest
versions of both driver software, service packs, and operating systems.
The recent nVidia drivers give <em>Hex</em> both stereo-in-a-window and
full-screen stereo on both Windows XP and Linux.
nVidia&rsquo;s 4351 driver (May 2003) is excellent, and is very easy to install.
See section <a href="#nVidia-Stereo-Configuration">nVidia Stereo Configuration</a> for more details. However, if you are
upgrading an nVidia driver on Windows, 
<em>ensure you de-install any earlier drivers first.</em> 
</p>
<a name="index-Stereo-Displays-2"></a>
<a name="index-Stereo-Parallax"></a>
<a name="index-Projection-Control-Panel"></a>
<a name="index-Perspective-Projection-1"></a>
<a name="index-Orthographic-Projection-1"></a>
<a name="index-Far-Plane"></a>
<a name="index-Near-Plane"></a>
<p>The type of graphics visual that <em>Hex</em> uses is determined
when the program first starts up. By default, <em>Hex</em> tries to use
a stereographics window. However, this nearly halves the drawing 
rate compared to non-stereo drawing. So if you don&rsquo;t like/want stereo,
you can force stereo off by using <code>hex -nostereo</code>.
Alternatively, set the HEX_STEREO environment variable to a
false value (i.e. any one of: no, n, false, f, or 0) in your login script.
If a stereo visual has been selected,
the display may be toggled between stereo and mono using the
<em>Stereo Parallax</em> option in the
<em>Projection Control</em> panel 
(or simply by pressing the keyboard <code>S</code> key).
The <em>Parallax</em>
slider may be used to adjust the stereographic effect.
Stereo parallax is the perceived separation
between the left and right images in the viewing plane (monitor screen).
The numerical values in the parallax slider
correspond approximately to screen millimetres on an un-zoomed display.
Positive parallax
values make the scene appear &ldquo;inside&rdquo; the display, whereas negative values
cause the scene to appear through, or in front of, the plane of the display.
A better stereographic effect is achieved when the
<em>Perspective Projection</em> toggle is enabled (by default, an  orthographic
projection is used).
The keyboard <code>P</code> key may also be used to toggle
between perspective and orthographic projection modes.
When perspective is enabled, a checker-board background may also be enabled
to enhance the perspective effect (keyboard <code>B</code>).
Figure 5 shows the antibody/lysozyme scene with perspective and background
modes enabled.
The perspective effect can be
increased by reducing the <em>Far Plane</em> and/or increasing the <em>Near
Plane</em> slider values.
For most people, a small positive parallax and a moderate perspective
gives a pleasing effect without straining the eyes.
</p>
<div class="float">
 <img src="3hfl_perspective.jpg" alt="3hfl_perspective">

<br>
</div><p><strong><strong>Figure 5.</strong> Illustration of the HyHel-5/Lysozyme complex shown 
as contoured Gaussian density surfaces and coloured by chain colour, drawn using
perspective (keyboard P) and background (keyboard B) modes enabled.
</strong>
</p>
<hr size="6">
<a name="Manoeuvring-Molecules"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Stereo-Displays" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Editing-Origins-and-Orientations" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Molecular-Graphics" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Manoeuvring-Molecules"></a>
<a name="Manoeuvring-Molecules-1"></a>
<h1 class="chapter">4. Manoeuvring Molecules</h1>

<p>When <em>Hex</em> reads in a molecule, it uses its all-atom centre of mass
as its centroid. This centroid is used as the local coordinate origin for
docking, and as the point about which any molecular rotations are applied.
However, if one or both molecules are quite large (i.e. too big to 
fit within about a 30 &Aring;ngstrom radius ball), you will probably want to 
change the centroids used for docking. This is because <em>Hex</em>&rsquo;s radial
functions fall off rapidly beyond about 30&Aring; from the origin. Hence,
<strong>unless you move the origin(s) to be closer to the expected 
binding site(s), you are likely to get very poor docking results</strong>. 
If you
don&rsquo;t know or suspect where the binding site is on a very large molecule, 
you should use <em>Hex</em>&rsquo;s <em>macro docking</em> mode for such cases.
See section <a href="#Macro-Docking">Docking Very Large Molecules (Macro Docking)</a>.
</p>
<a name="index-Editing-Molecules"></a>
<p>In addition to rotating, translating, and scaling a molecular graphics scene,
<em>Hex</em>, also allows you to change the <em>relative</em> orientations of a 
pair of molecules. <em>Hex</em> calls this <em>editing</em>, because saving 
&ldquo;manoeuvred&rdquo; structures out to a file causes them to be written using their
transformed coordinates.
Note: <em>Edit Mode</em> is only enabled when you have two molecules loaded. 
<strong>Editing is often a necessary preparatory step before running 
a docking calculation</strong>. 
For example, when you load a pair of molecules, <em>Hex</em> uses some heuristics
to place the ligand near the receptor. However, you may have knowledge about
one or both of the binding sites, and hence you may want to manoeuvre the 
molecules into something that resembles the expected binding orientation.
If you do this, you can then limit the search range of the docking correlation
instead of performing a global search. 
This should give fewer &ldquo;false-positives&rdquo;,
and it will certainly make the docking calculation go faster.
</p>
<p>It should be noted that although you can edit the scene by picking and
moving both the receptor and the ligand orientations, 
the coordinates of the receptor &ldquo;take priority&rdquo; when structures are 
written to an output file. 
In other words, any relative motion is <em>always</em> transferred 
entirely onto the ligand on output. 
The receptor coordinates are always restored to their original values.
</p>

<hr size="6">
<a name="Editing-Origins-and-Orientations"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Editing-Origins-and-Orientations"></a>
<a name="Editing-Origins-and-Orientations-1"></a>
<h2 class="section">4.1 Editing Origins and Orientations</h2>

<a name="index-Orientation-Control-Panel-1"></a>
<a name="index-Euler-Angles"></a>
<p>Molecular centroids may be edited using the <em>Edit Origins</em> button 
(two crosses and a line) on the left-hand border.
When editing orientations,
it may help to activate the <em>Intermolecular Axis</em>, 
in order to see the new and old positions of the centroid(s).
Try changing to <em>Edit Origins</em> mode, and pick a molecule to edit. 
Once the picked molecule is highlighted, dragging <code>Button 1</code> will
move the origin of the selected molecule. 
In <em>Edit Origins</em> mode, buttons 2 and 3 always rotate the entire scene. 
When you revert to the default
<em>Pointer Mode</em>, the new origin will be activated.
Note that the new origin only applies
to the current session. It is not saved when you write an edited molecule
to disc.
You can use virtually all of the other GUI functions while in edit mode, 
although certain operations
(e.g. opening/closing molecules) cause <em>Hex</em> to revert to pointer mode.
</p>
<p>You can also explicitly specify the location of the molecular origins using
the <em>Receptor Origin</em>  and <em>Ligand Origin</em> text entry boxes
in the <em>Orientation Control</em> panel.
For example, following the HyHel-5/lysozyme example, 
you could enter <code>Y-38</code> into the <em>Ligand Origin</em> text box
(and then press the return key) in order to set the ligand origin to the
the CA atom of PHE:38 in the lysozyme Y chain. Similarly, as of 
<em>Hex</em> version 5.0, 
the <em>Receptor Interface</em>  and <em>Ligand Interface</em>  
text entry boxes may be used to specify the residues which should be located
on the positive or negative <em>z</em>-axis, respectively. 
Orienting each protein with respect to the <em>z</em>-axis in this way
is equivalent to specifying initial values for five of the six rigid body 
degrees of freedom (i.e. all but the twist angle). 
</p>
<p>Another way to control the molecular orientations is to use the
<em>Apply to</em> selector in the <em>Orientation Control</em> panel 
to select either the receptor or the ligand, and then to rotate the
selected molecule using the 
Euler angle <em>Alpha, Beta</em> and <em>Gamma</em> sliders. The intermolecular
separation can be modified with the <em>R slider</em>
at the bottom of the main window.
Select <em>Commit</em> when you&rsquo;re done editing, in order to commit the 
new transformations into the molecules.
</p>
<p>Orientation editing can also be performed manually clicking on the
<em>Edit Mode</em> left border button (which looks like a lightning bolt) 
and then by picking and dragging molecules using the mouse.
<a name="index-Hammer-Button-1"></a>
<a name="index-Intermolecular-Axis-1"></a>
<a name="index-Pointer-Mode"></a>
<a name="index-Orientation-Control-Panel-2"></a>
In both editing modes, <em>picking</em> a molecule 
ties its motion to the mouse and to the rotation sliders in the
<em>Orientation Control</em> panel. 
The covalent skeleton of the picked molecule is displayed as 
dashed lines as a 
visual reminder that this molecule has been activated for editing.
Picking the background (i.e. a <code>Button 1</code> 
pick that misses all atoms) de-selects the active molecule
and makes the mouse move the whole scene.
The <em>Hammer</em> button can be used to undo the most recent sequence
of editing movements.
Once you&rsquo;re happy with
your edits, revert to <em>Pointer Mode</em> to commit the new orientations.
You can now save the
scene by going to the <em>File</em> menu and selecting:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Save ... Both
</pre></td></tr></table>



<p>This causes a file selection widget
to appear, prompting you for the name of a file in which to save
both molecules (in the newly edited orientation). You can confirm the new
orientation has been created by loading the new file into another 
<em>Hex</em> session (you may have to rotate the scene to see exactly the 
same view). Alternatively,
you could write each molecule to a separate file using:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Save ... Receptor
</pre></td></tr></table>

<p>and
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Save ... Ligand
</pre></td></tr></table>

<p>You could then open these new files to verify the new orientation.
</p>



<hr size="6">
<a name="Docking-Molecules"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Editing-Origins-and-Orientations" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Rotational-Search" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Manoeuvring-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Docking-Molecules"></a>
<a name="Docking-Molecules-1"></a>
<h1 class="chapter">5. Docking Molecules</h1>


<a name="index-Reference-Orientation"></a>
<a name="index-Receptor-1"></a>
<a name="index-Ligand-1"></a>
<a name="index-Complex-1"></a>
<p>In order to run a docking calculation in <em>Hex</em>, you will need to load 
a <em>receptor</em> and a <em>ligand</em> PDB structure using the 
<em>File</em> pull-down menu. If you want to test the docking algorithm
by docking two separately determined sub-units of a complex for which the 
crystal structure is also available, you can also load the
<em>complex</em> structure which will be used as a <em>reference orientation</em>
to evaluate the accuracy of the docking prediction.
</p>


<a name="index-Hetero-Control-Panel"></a>
<p>Generally, you will
have to remove water molecules and any other <em>hetero</em> molecules prior to
docking. You can do this globally using the <em>Hetero Control</em> menu panel.
If more detailed control is required, you will probably have to edit each
PDB file manually using a text editor. 
It may also be necessary to remove other chains in the
PDB file or to shorten a chain to the domain of interest in docking. For
example, when docking an antigen to an antibody it is usually advisable
to delete all but the Fv fragment of the antibody structure (although the
program has been used to dock a protein G molecule to a complete Fab fragment).
Having edited your PDB files, you should have a <em>receptor</em> and a 
<em>ligand</em> file which contain
only the receptor and ligand molecules, respectively, and (optionally)
a <em>complex</em> file, which contains both molecules in the docked orientation.
When using a <em>complex</em> structure, you should ensure that the chain names
and residue numbers are consistent with 
those of the  <em>receptor</em> and <em>ligand</em>
because <em>Hex</em> uses this information to identify and hence
superpose corresponding pairs of alpha-carbon atoms from each chain 
in order to calculate RMS
deviations between the docked position of the ligand and its position
in the known complex.
If necessary, you can use the 
Linux utility <strong>hex_chain</strong> in <em>Hex</em>&rsquo;s <strong>bin</strong> 
directory to rename a chain in a PDB file.
Similarly, the <strong>hex_delta</strong> utility may be used to
renumber the residues of a given chain.
</p>

<p>Up to three protein structures can be loaded into <em>Hex</em> directly from
the command line. For example, to load three of the PDB files from the 
<strong>examples</strong> directory for the HyHel-5/lysozyme complex, 
you could type at the Linux command prompt:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">hex 3hfl_fv.pdb 3hfl_ly.pdb 3hfl.pdb
</pre></td></tr></table>



<p>If you are studying a particular system and you find you are continually
repeating the same sequence of actions to get things set up,
you may find it to be more convenient to load all three molecules together
using a short macro file, similar to those in the <strong>examples</strong> directory. 
For example, the macro file, <strong>3hfl.mac</strong>, 
can be executed by selecting:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Macros ...  Run ... 3hfl.mac
</pre></td></tr></table>


<p>This macro file contains the following:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example"># 3hfl.mac
# ========
#
close_all
set_colour_file  3hfl.col
open_receptor    3hfl_fv.pdb
open_ligand      3hfl_ly.pdb
open_complex     3hfl.pdb
fit_ligand
</pre></td></tr></table>


<p>When using a <em>complex</em>
reference structure, you should find that on loading the molecules, the
complex is drawn in grey and the complex molecule is superposed onto the
receptor (it is the complex that physically moves here). 
If the ligand molecule originated from an edited <em>complex</em> file, 
then it will also be superposed over the complex because
its coordinates are relative to the same coordinate frame as the receptor
molecule. Otherwise, the ligand can be transformed into the receptor/complex
frame using the <em>Fit Ligand</em> button in the 
<em>Orientation Control</em> panel.
</p>
<p><b>Note.</b> If you wish to superpose one or more proteins onto another,
the above behaviour may be exploited by 
treating the stationary molecule as the &ldquo;receptor,&rdquo;
and by treating each moving molecule as the &ldquo;complex.&rdquo; Saving the
fitted &ldquo;complex&rdquo; to a new PDB file will effectively write it out its
fitted orientation in the coordinate frame of the receptor.
</p>



<hr size="6">
<a name="Rotational-Search"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Docking-Molecules" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Distance-Sub_002dStepping" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Rotational-Search"></a>
<a name="Rotational-Search-1"></a>
<h2 class="section">5.1 Rotational Search</h2>

<a name="index-Docking-Control-Panel-1"></a>

<p>Docking calculations are controlled by the options in the
<em>Docking Control</em> panel. Generally, the docking search proceeds 
by rotating the receptor and ligand about their centroids at each
of a range of intermolecular distances. The receptor and ligand are 
each assigned two Euler rotation angles, and the final rotation is
defined as a twist of the ligand about the intermolecular axis.
The default behaviour is then to perform a full six-dimensional search over 
the full rotational ranges. However, more limited docking searches can be
performed in which the allowed angular and distance ranges may be
constrained by the user. The docking coordinate system is illustrated
in Figure 6 below.
See section <a href="#Manoeuvring-Molecules">Manoeuvring Molecules</a> for details on setting up the
initial docking orientation.
</p>
<div class="float">
 <img src="docking_diagram.jpg" alt="docking_diagram">

</div><p><strong><strong>Figure 6.</strong> Illustration of spherical polar docking 
with respect to the intermolecular axis. 
An initial docking orientation may be defined by specifying which residues
should be located at the local coordinate origin for each molecule, and
by defining &quot;interface residues&quot; which will be located on the
<em>z</em>-axis.  The docking search may be restricted
by defining a &ldquo;range angle&rdquo; for the receptor and/or ligand orientations.
If range angles are defined, then the interface residues will always be
constrained to appear within a spherical cone defined by the corresponding
range angle. This illustration shows two range angles, each of 45 degrees.
</strong>
</p>
<p>The calculation is arranged so that the intermolecular twist angle
search is in the innermost loop of the search. The search around the
twist angle may be accelerated using a 1D FFT. Alternatively, all three
Euler angles assigned to the ligand can be searched together using a 3D FFT.
In the Linux version, all five rotation angles may be searched together
using a 5D FFT. However, this requires at least 1 gigabyte of memory to 
hold the very large 5D rotational grid. 
</p>
<a name="index-FFTW"></a>
<a name="index-MKL"></a>
<a name="index-Kiss-FFT"></a>
<a name="index-cuFFT"></a>
<a name="index-FFT-speeds"></a>
<p><b>Note.</b> The speed of the 3D and 5D FFT calculations depends very much 
on the quality of FFT code, and whether it exploits any hardware 
acceleration. For CPU-based docking, the 
<a href="http://www.intel.com">Intel MKL</a> 
code is about
twice as fast as <a href="http://www.fftw.org">FFTW</a>, which 
is about one and a half times as fast my own multi-dimensional 
FFT code which is partly based on 
<a href="http://sourceforge.net/projects/kissfft">Kiss FFT</a>,
and which doesn&rsquo;t use any assembly or SSE instructions. 
However, only the Linux version of the MKL library is available for free to 
non-profit organisations. So docking is fastest with the Linux version of
<em>Hex</em>. I do not use the FFTW code because I do not agree with its
GPL license conditions. Hence, 3D and 5D docking correlations
on the Windows and Mac versions of <em>Hex</em> is relatively slow. 
</p>
<p>For GPU-based docking, 1D correlations using 
<a href="http://www.nvidia.com/object/cuda_get.html">cuFFT</a>,
on a high-end GeForce or Quadro card
are much faster than on a high-end CPU. Compared to using a single high-end
CPU (3.2GHz i7-965), the same calculation on a high-end GPU (FX-5800) is 
about 45x faster (Ritchie and Venkatraman, manuscript in preparation).
</p>


<a name="index-Steric-Scan"></a>
<a name="index-Final-Search"></a>
<p>There are several controls which specify the resolution, and in particular
the <em>order</em>, N,  of the docking correlation. 
Figure 7, below, shows a screen shot of the Docking Control panel.
The default settings are for the program to perform 
an initial <em>Steric Scan</em>
at N=16, followed by a <em>Final Search</em> at N=25,
using just the steric contribution to the docking energy.
In this mode, about all but the top 30,000 orientations 
are discarded after the <em>Steric Scan</em>. 
The <em>Steric Scan</em> may be toggled off, in which case 
every orientation is evaluated using a steric correlation
(and optionally an electrostatic correlation)
to order N, as given the <em>Final Search</em> slider. 
However, this can significantly increase total docking times.
Using the two-step search with N=16 and N=25 is found to
work well in practically all cases. 
The electrostatic contribution to the docking correlation may be enabled
using the <em>Electrostatics</em> toggle. 
Electrostatics are only ever calculated
in the <em>Final Search</em> phase.
You can get a feeling for how well each correlation order, N,
recognises a given complex by running the program on known complexes and
by observing how highly the correct solution is ranked in each case.
</p>
<div class="float">
 <img src="docking_control.jpg" alt="docking_control">

</div><p><strong><strong>Figure 7.</strong> Screen shot of the Docking Control panel.
</strong>
</p>
<a name="index-CAPRI-Blind-Trial"></a>

<p>Although the default is to use correlations to N=25, in Rounds 3&ndash;5
of the 
<a href="http://capri.ebi.ac.uk/"><strong>CAPRI blind trial</strong></a>, I found that 
better results are obtained using N=30 correlations (see section <a href="#References">References</a>). 
Generally, N=30 is recommended when docking high resolution crystal 
structures for which the conformational change on binding is expected to
be small. N=25 should be used when docking model-built structures or
structures which are expected to be more flexible.
As rule of thumb, I would use N=16 for the scan stage with final scoring
at N=30, and I would use scans using N=20 when scoring with N=30. 
In some cases, a
good solution can be missed with the N=16 scan. It is safer, but obviously
slower always to use N=20 for the initial scan.
</p>



<hr size="6">
<a name="Distance-Sub_002dStepping"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Rotational-Search" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Examples" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Distance-Sub_002dStepping"></a>
<a name="Distance-Sub_002dStepping-1"></a>
<h2 class="section">5.2 Distance Sub-Stepping</h2>

<p>As of version 4.5, <em>Hex</em> performs the high resolution <em>Final Search</em>
correlation using smaller distance incerements than are used for the fast
low resolution <em>Steric Scan</em> phase.  This allows the search space to be
covered more rapidly (coarsely) in the first phase, but more finely in the
final phase. This behaviour is controlled by the
<em>Distance Range</em>, <em>Scan Step</em> and <em>SubSteps</em> parameters in
the <em>Docking Control</em> panel. The default values are
<em>Distance Range</em>=40, <em>Scan Step</em>=0.75, <em>SubSteps</em>=2. This
means that the <em>Steric Scan</em> phase will search over 55 distance
increments of +/- 0.75 &Aring; from the starting separation, 
plus the starting separation itself). These orientations are sorted by 
calculated energy, and a new set of trial orientations are generated for
the top-scoring 10,000&ndash;20,000 orientations using the <em>Scan Step</em> and
<em>SubSteps</em> parameters to construct new distance samples in steps
of +/-(Scan Step)/(Substeps) from the initial orientations. In other words
the default behaviour is essentially to scan the search space at 
1&Aring;  resolution, but to perform the high resolution scoring at 
0.5&Aring; resolution. Setting <em>SubSteps</em>=0 gives the old behaviour of
earlier versions in which a constant distance step is used for both 
resolution levels.
</p>


<hr size="6">
<a name="Docking-Examples"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Distance-Sub_002dStepping" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Disc-Cache" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Docking-Examples"></a>
<a name="Docking-Examples-1"></a>
<h2 class="section">5.3 Docking Examples</h2>

<a name="index-Docking-Control-Panel-2"></a>
<p>OK, lets start docking.
Open the <em>Docking Control</em> panel and select the following 
(mostly default) values from the selection boxes:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Docking ... Correlation Type ...   Shape Only
Controls ... Docking ... FFT Mode ...           3D Fast Lite
Controls ... Docking ... Post Processing ...    None
Controls ... Docking ... Grid Dimension ...     0.6 
Controls ... Docking ... Solutions ...          500
Controls ... Docking ... Receptor Range ...      45
Controls ... Docking ... Receptor Step Size ... 7.5
Controls ... Docking ... Ligand Range ...        45
Controls ... Docking ... Ligand Step Size ...   7.5
Controls ... Docking ... Twist Range ...        360
Controls ... Docking ... Twist Step Size ...    5.5
Controls ... Docking ... Distance Range ...      40
Controls ... Docking ... Scan Step ...         0.75
Controls ... Docking ... Substeps ...             2
Controls ... Docking ... Steric Scan ...         16
Controls ... Docking ... Final Search ...        25
</pre></td></tr></table>


<p>This says that the <em>Steric Scan</em> (N=16) phase of the
docking calculation will be performed at 
(1+40)/0.75=53 intermolecular separations, in +/- steps of
0.75 &Aring; starting from the current distance 
posted in the <em>R</em> slider in the bottom border of the main window.
The <em>Final Search</em> (N=25) phase will be applied to the the highest
scoring scan orientations in steps of 0.75/2 &Aring;, as described above.
The rotational search will use angular increments of about 7.5 degrees in 
each of the two ligand and receptor rotational angles, and in steps
of 5.5 degrees about the twist angle. Internally, the given step sizes
will be adjusted to give computationally convenient numbers of samples.
For example, a step size of 5.5 degrees will actually give 64 samples
(360/64=5.625). Specifying a step size of 7.5 degrees will give 48 or 24 angular
samples in the corresponding FFT grid, or it will select an icosahedral
tesselation with 812 vertices because that is the tesselation order which
will give an average angular distance between neighbouring pairs of vertices of 
about 7.5 degrees.
</p>
<p><b>Note.</b> This method specifying the angular resolution is new in version
5.0. Earlier versions required the user to specify the sampling density 
directly in terms of tesselation orders, which was not very easy to understand!
Please also note that the default search step sizes are often somewhat
finer than the search increments used in previous versions.
</p>
<p>Now start the calculation:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Docking ... Activate
</pre></td></tr></table>


<a name="index-Grid-Dimensions"></a>
<a name="index-Surface-Control-Panel"></a>
<p>You have just started an example docking search restricted to the known 
binding sites, so the calculation should only take a few minutes.
The program will take
a few seconds to calculate the surface skin coefficients and then
proceed to calculate docking correlation scores at each of the specified 
angular and intermolecular increments.
A Cartesian grid is used to sample the molecular skins numerically 
but this grid plays no further role in the calculation once the 
surface skin expansion coefficients have been determined.
Most conventional FFT docking algorithms have to use rather large grids 
(e.g. 1 &Aring;ngstrom cubes) because the grid must accommodate all possible 
translations of the ligand about a stationary receptor.
Here, the grid only needs to contain the larger of the two molecules
so that much finer sampling grids are feasible.
In <em>Hex</em>, a 0.6 &Aring;ngstrom grid seems to work well and is still 
reasonably fast to calculate. The sampling grid size may be varied
using the <em>Grid Dimension</em> selection box.
The calculation of the surface skins used in the docking correlation is
controlled by the parameters in the <em>Surface Control</em> panel. 
The default values do not normally need to be changed.
</p>


<a name="index-Steric-Scan-1"></a>
<a name="index-Twist-Samples"></a>
<a name="index-Final-Search-1"></a>
<a name="index-Docking-Samples"></a>
<p>The search in the above example is
&ldquo;restricted&rdquo; because setting the ligand and receptor <em>range angles</em> to 
45 degrees means that only a small fraction of the total possible rotational 
increments (the <em>Receptor Samples</em> and <em>Ligand Samples</em> values) 
will actually be used for the <em>molecular rotations</em>
of each molecule. 
Essentially, each molecule
is rotated incrementally so that successive tessellation points are rotated
onto the intermolecular axis (the <em>z</em>-axis). 
Thus, in the low resolution <em>Steric Scan</em> phase,
 812 x 812 distinct rotational
orientations are produced, although any orientations that fall outside
the angular range cones are discarded. 
Angular search ranges are illustrated in Figure 6.
At each of the surviving orientations, the
ligand is then rotated (or <em>twisted</em>) 
through 360 degrees about the <em>z</em>-axis 
in 64 steps of about 5.5 degree
increments specified by the <em>Twist Samples</em> value. 
In the high resolution <em>Final Search</em> phase,
the best 10,000 of the above orientations will be 
used to generate up to 25,000 new distinct trial orientations after 
distance sub-sampling (see above) and re-evaluated at
N=25 using 128 samples for each twist angle search.
Note, <em>Hex</em> is hard-wired to use 64 twist steps in the 
initial steric scan phase, because there is no benefit in using finer 
steps in this phase.
</p>
<p>It should be noted that such docking calculations never give a unique 
solution. Rather, <em>Hex</em> sorts the generated
orientations by docking energy and 
prints a summary of the 10,000 highest scoring (lowest energy) orientations.
The best 500 orientations are retained for viewing. 
You can view the solutions using the <em>Soln:</em> slider, or you
can step through the solutions using the keypad or arrow keys:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Home:       Starting orientation
Page/Down:  Next solution
Page/Up:    Previous solution
End:        500'th (or last) solution
</pre></td></tr></table>


<a name="index-Global-Search"></a>
<p>You should find in
the above example that <em>Hex</em> generates several orientations with low RMS
deviations from the correct (starting) orientation. 
However, you might object that
the calculation was biased to find the right answer by restricting the
search to the known binding sites. If so, you could try a global docking
search (about 10^9 trial orientations), using:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Docking ... Receptor Range ...   180
Controls ... Docking ... Ligand Range ...     180
</pre></td></tr></table>


<p>This calculation will take about 30 minutes on a 
single 1GHz Pentium III processor.
This is substantially faster than an equivalent calculation
using a conventional FFT approach.
In this particular
case, <em>Hex</em> should still identify the correct docking orientation
(to within about 1.5 &Aring;ngstrom RMS), but its often useful to use restricted
searches when starting with unbound subunits.
</p>

<a name="index-Orientation-Control-Panel-3"></a>
<p>When viewing docking
solutions, it is helpful to keep either the receptor or the ligand in a fixed
orientation in the display, so that only one molecule moves when stepping
through the docking solutions. The default is to transfer any 
receptor motion onto the ligand, so that the ligand appears to translate and
rotate about a fixed receptor.
This behaviour can be changed using the <em>Orientation Control</em> panel:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Orientation ... Centre ... Ligand
</pre></td></tr></table>

<hr size="6">
<a name="Disc-Cache"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Docking-Examples" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Calculating-Bumps" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Disc-Cache"></a>
<a name="Disc-Cache-1"></a>
<h2 class="section">5.4 Disc Cache</h2>
<a name="index-Translation-Matrices-1"></a>
<p>If you plan to use <em>Hex</em> more than once, enabling a 
<em>disc cache</em> will speed up the translational part of each calculation. 
When disc caching is enabled the (large) translation matrices are calculated 
only when necessary and are saved in the cache directory for re-use. 
Disc caching controls can be found under the <em>File</em> menu.
The default cache directory is <strong>$HEX_ROOT/cache</strong>,
although it is recommended to set a cache directory explicitly using
the HEX_CACHE environment variable.
The cache directory should be specially created for this purpose 
(i.e. it contains nothing else) and a distinct directory should be used 
by each user 
(because there&rsquo;s no synchronisation between multiple invocations of 
<em>Hex</em>).  
With typical usage, the disc cache will occupy up to about 200 Mb 
of disc space. 
</p>


<hr size="6">
<a name="Calculating-Bumps"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Disc-Cache" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Mechanics-Refinement" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Bumps"></a>
<a name="Calculating-Bumps-1"></a>
<h2 class="section">5.5 Calculating Bumps</h2>

<a name="index-Post-Processing-Options"></a>
<a name="index-Clustering-Control-Panel"></a>
<p>Following the basic docking correlation algorithm, candidate docking
orientations may be filtered and refined using one of the
<em>Post Processing</em> options. Post processing is applied to the
top-scoring 1,000 solutions from the correlation search
(or the user-selected number of orientations to retain, if larger than 1,000). 
The simplest
option is to enable a <em>bumps</em> counter, in which the number of
steric clashes between non-bonded pairs of heavy atoms in each solution is
calculated. An option in the <em>Clustering Control</em> panel may then be
used to filter out solutions with a specified number of steric clashes.
</p>

<hr size="6">
<a name="Molecular-Mechanics-Refinement"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Calculating-Bumps" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Clustering-Docking-Results" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Molecular-Mechanics-Refinement"></a>
<a name="Molecular-Mechanics-Refinement-1"></a>
<h2 class="section">5.6 Molecular Mechanics Refinement</h2>

<a name="index-Soft-Potentials"></a>
<a name="index-False-Positives"></a>
<p>In addition to the bumps counter, a single (rigid body) 
molecular mechanics energy may be calculated for each docking solution 
(<em>MM Energies</em>), 
or a Newton-like energy miminisation (<em>MM Minimisation</em>) can be applied 
to each docking solution. 
These energies are calculated using &ldquo;soft&rdquo; Lennard-Jones and hydrogen
bond potentials, adapted from the OPLS force-field parameters, along
with an explicit charge-charge electrostatic contribution.
When docking complexes where conformational changes are known to be
small, this gives an effective way to prune many &ldquo;false-positive&rdquo;
orientations and to enhance the energy of the &ldquo;right answer&rdquo;. However,
this rigid-body refinement procedure should not be used if conformational 
changes are expected to be large because (despite using soft potentials) 
it tends to
eject ligands with incorrect conformations from the binding site.
This part of the program is still &ldquo;under development&rdquo;. 
</p>

<hr size="6">
<a name="Clustering-Docking-Results"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Molecular-Mechanics-Refinement" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Saving-Docking-Results" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Clustering-Docking-Results"></a>
<a name="Clustering-Docking-Results-1"></a>
<h2 class="section">5.7 Clustering Docking Results</h2>

<p>Because <em>Hex</em> uses essentially a 
brute-force search approach to the docking problem,
it is advisable to <em>over-sample</em>
 the search space rather than to risk missing
a good solution by under-sampling the space. However, this can cause 
multiple similar but incorrect orientations (false-positives) to push good
solutions down the list. By default, <em>Hex</em> 
uses a simple clustering algorithm
to group spatially similar docking orientations. Each docking solution is
first ordered by energy, and the lowest energy solution is made the
seed orientation for the first cluster. The list is then searched 
down to a given depth for 
other similar orientations whose main-chain alpha-Carbon RMS deviation is
within a given threshold (default 3&Aring; RMS) of the seed orientation, and
these orientations are then assigned to the first cluster. The process is
then repeated starting from the next lowest unassigned orientation, until
all solutions have been assigned to a cluster.
The <em>Cluster Window</em> parameter may be used to control the search depth
when looking for cluster members. 
Because clustering uses a simple but inefficient algorithm 
(rather like a &ldquo;bubble-sort&rdquo;), it is advisable use this parameter 
to limit the search depth if the number of saved solutions is large. 
</p>


<a name="index-Clustering-Control-Panel-1"></a>
<p>The <code>Controls ... Clustering</code> control panel may be used to control the
type of clustering and clustering parameters used, and how 
cluster membership affects the displayed solutions. By default, only the
lowest energy member of each cluster is displayed or written to an output
PDB file (<em>Display Clusters = Best</em>). 
Alternatively, you can choose to
view (or output) all solutions (<em>Display Clusters = All</em>). 
Similarly, solutions
may be viewed in cluster order (best to worst) or in order of calculated
energy using the <em>Sort Solutions</em> selection option.
The RMS threshold for cluster membership may be changed using the
<em>RMS Threshold</em> slider. 
Displayed solutions may also be selected by
the number of steric clashes, or bumps, detected. Any solution containing
a number of bumps which is less than or equal to the current 
<em>Bumps Threshold</em> (default 0) is a candidate for cluster membership. 
Solutions with a greater number of bumps may still be viewed or output,
but these always appear after any non-bumping solutions in the sorted
list of solutions (i.e. have a larger cluster rank).
</p>
<p>The default docking search range and clustering parameters are normally
sufficient to generate good coverage of the search space, and to 
distinguish different but similar solutions. But you may wish to experiment.
Clustering is re-calculated whenever a clustering parameter is changed,
without having to re-run the docking search.
Here, &ldquo;good coverage&rdquo;
means that for each docking orientation, there is about a 99% chance of 
generating at least one other solution within 3&Aring; RMS of the first one
(unpublished experiments). 
Deciding whether or not two similar but different orientations ought to 
fall within the same cluster is somewhat subjective. The 3&Aring; threshold
seems to work well for large-ish ligands (roughly 100 residues), but
should probably be reduced for smaller ligands. In the CAPRI blind docking
experiment, I tend to use a large threshold of, say, 5&Aring; to try to
improve the chances of finding at least one &ldquo;medium accuracy&rdquo; prediction
within the first 10 predictions submitted). But this doesn&rsquo;t always help!
</p>



<hr size="6">
<a name="Saving-Docking-Results"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Clustering-Docking-Results" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Multiple-Structures" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Saving-Docking-Results"></a>
<a name="index-Docking-Results"></a>
<a name="Saving-Docking-Results-1"></a>
<h2 class="section">5.8 Saving Docking Results</h2>


<a name="index-Docking-Summary-File"></a>
<a name="index-Docking-Transformations-File"></a>
<a name="index-Saving-Docking-Results-1"></a>
<p>There are two main ways in which docking results may be saved to disc. 
The most compact method is to write
a <em>docking summary file</em> using the <code>File ... Save ... Transform</code>
menu item. This writes a file that lists the docking energies for all
predicted orientations, along with the rotation and translation parameters
which should be applied to the ligand coordinates in the original ligand
PDB file in order to produce the corresponding docking orientation in the
coordinate frame of the recptor PDB file. The actual file format is descibed
in the first few &ldquo;comment&rdquo; lines of the output file.
A similar but more detailed 
output file format is given by <code>File ... Save ... Matrix</code>. This
lists the transformations in a more verbose format using keywords to
label each piece of data.
For rigid body docking, both formats are quite compact,
but you will need to write your own code to
process the docking summary file. 
Its not possible to read these files
back into <em>Hex</em>; they are purely for output to external software.
</p>

<p>The second, less compact output method is to write a separate PDB file for
each docking orientation.
The current docking orientation can be written to a single PDB file by 
selecting:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Save ... Both
</pre></td></tr></table>

<p>Alternatively, a range of docking solutions may be saved using:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Save ... Range
</pre></td></tr></table>

<a name="index-Saving-Docking-Ranges"></a>
<p><a name="Save-Docking"></a>
This opens a new <em>Save Docking Range</em> panel in which a range
of docking solutions may be specified, along with a 
a file name pattern to be used for the output files.
The default is to save the first 100 solutions to 
separate PDB files of the form <strong>dock001.pdb</strong>, etc., in the current 
directory. 
In both the PDB range and the summary output modes, either a
list of <em>raw solutions</em> (sorted by energy) are written, or a range of 
clustered orientations are written, depending on the current clustering 
option. In other words, the sequence of solutions written out is the
same sequence that you would see when stepping through the orientations
using the <code>Page/Down</code> and <code>Page/Up</code> or arrow keys.
</p>


<hr size="6">
<a name="Docking-Multiple-Structures"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Saving-Docking-Results" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Macro-Docking" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Docking-Multiple-Structures"></a>
<a name="Docking-Multiple-Structures-1"></a>
<h2 class="section">5.9 Docking Multiple Structures</h2>

<a name="index-NMR-Docking"></a>
<a name="index-NMR-Models"></a>
<p>Docking multiple structures (e.g. NMR structures) with <em>Hex</em> is easy.
Just put all your ligand structures (assumed to be in a
common coordinate frame) into a single PDB file, and use
the PDB convention of MODEL/ENDMDL to distinguish each structure.
<em>Hex</em> will run its docking search for each ligand in turn. Similarly,
multiple receptor model structures may be loaded and docked, giving an
<em>M by N</em> docking (may take a while though!). The actual structures
to be docked may be selected using the <em>Receptor Docking</em>
and <em>Ligand Docking</em> selectors in the <code>Controls ... Solid Models</code>
control panel (default all v&rsquo;s all). Individual structures may be
selected for viewing using the <em>Receptor Display</em> and 
<em>Ligand Display</em> selectors in the same control panel.
This panel also contains some
experimental controls for <em>cloning</em> and <em>editing</em> different models
(where each clone is treated as a new NMR-type model structure 
to be docked, with each docking run each starting from the edited pose).
</p>
<p><em>Hex</em> can also read a list of ligands from a multi-molecule
SDF file. However, docking results are always written out in PDB format.
</p>


<hr size="6">
<a name="Macro-Docking"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Docking-Multiple-Structures" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Additional-Docking-Parameters" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Macro-Docking"></a>
<a name="index-Docking-Large-Molecules"></a>
<a name="Docking-Very-Large-Molecules-_0028Macro-Docking_0029"></a>
<h2 class="section">5.10 Docking Very Large Molecules (Macro Docking)</h2>


<a name="index-CAPRI-Blind-Trial-1"></a>
<a name="index-Macro_002dSampling-Control-Panel"></a>

<p>The radial shape functions used in <em>Hex</em> decay exponentially beyond
about 35 &Aring;ngstroms from the chosen origin. This means that the shapes
of proteins much larger than this are not represented well.
One example of such a large protein is the viral-surface  
hemagglutinin molecule presented in the CASP II and CAPRI 
protein docking challenges. In order to dock such large molecules,
<em>Hex</em> assigns
multiple local coordinate systems to the larger molecule (assumed to
be the <em>receptor</em>) and docks the ligand around each local coordinate 
frame on the receptor. Setting up
the receptor coordinate systems is controlled by the
<code>Controls ... Macro-Sampling</code> control panel. 
</p>


<a name="index-Harmonics-Control-Panel"></a>
<p>The general approach is
to construct a very low resolution spherical harmonic surface representation
of the receptor (say, setting L=5 in the <em>Harmonics Control</em> panel), 
and then to cover this surface with a large number of spheres. These spheres
are placed using the surface normals of each triangular patch used to
draw the harmonic surface. 
The spheres are then iteratively <em>culled</em>, where
the sphere with the greatest overlap with its neighbours is culled at each
iteration. This is repeated until some pre-set number of spheres (default 25)
remain - these should still cover the surface, but be reasonably 
well-dispersed. The positions of these spheres are then used to generate
initial docking orientations for the ligand over the receptor, and an 
appropriate local coordinate frame is derived from the geometry.
For multi-domain receptors, each sphere can be assigned to a source chain, 
and spheres that do not belong to a given chain may optionally be excluded.
The gives a simple (but imperfect) way to limit the docking search to one
of several symmetry-related domains 
(as in the hemagglutinin case, for example).
</p>


<a name="index-Docking-Control-Panel-3"></a>
<p>In order to set up such a calculation for an antibody/antigen system
(where the antibody&rsquo;s binding site is known), it is a good idea to 
orient the ligand (antibody) such that the binding site faces the
receptor - i.e. the principal intermolecular axis passes near the 
antibody hypervariable loops. This can be achieved by manually 
<em>editing</em> the antibody into position. 
Then, the macro-sampling algorithm will
place the ligand (antibody) at each of the generated starting orientations
and in each orientation the antibody will be transformed to face the 
receptor. Using the main <em>Docking Control</em> panel, a search range
of 45 degrees should be set for both the ligand and receptor so that
the docking search will gyrate each molecule about the local intermolecular
coordinate system yet still keep each molecule roughly facing the other 
as defined by the starting pose.
</p>

<hr size="6">
<a name="Additional-Docking-Parameters"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Macro-Docking" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Molecular-Matching" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Additional-Docking-Parameters"></a>
<a name="Additional-Docking-Parameters-1"></a>
<h2 class="section">5.11 Additional Docking Parameters</h2>


<p>There exist a
few further parameters which control the docking calculation but which
are not accessible from the GUI. These parameters must be set using 
keywords in a macro file.
For example, the steric overlap penalty factor has a default value of 11. 
This can be adjusted by running the example macro <strong>factor_q.mac</strong>:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">#
# steric_q.mac - change steric penalty factor Q (default=11).
#
# (this parameter can't be changed from the GUI yet).
#
factor_q 11
</pre></td></tr></table>


<p>Similarly, another example macro file is <strong>factor_g.mac</strong>:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">#
# factor_g.mac - change electrostatic scale factor (def=700 J/mol).
#
# (this parameter can't be changed from the GUI yet).
factor_g 700
</pre></td></tr></table>



<hr size="6">
<a name="Molecular-Matching"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Additional-Docking-Parameters" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Molecular-Matching"></a>
<a name="Molecular-Matching-1"></a>
<h2 class="section">5.12 Molecular Matching</h2>


<a name="index-Matching-Control-Panel"></a>
<a name="index-Superposing-Molecules"></a>
<p><em>Hex</em> superposes or <em>matches</em> pairs of molecular 
structures using the same 3D
density representation as for docking. Superposition is very much
like docking, except now the search is for maximum similarity
rather than maximum complementarity.
Superposition calculations
can only be performed once you have opened a 
<em>receptor</em> and a <em>ligand</em> molecule. 
If you want to superpose a pair of molecules that originate from
the same PDB file, you will have to manually edit the molecules 
into separate PDB-format files.
Superposition calculations are controlled by the parameters in the
<em>Matching Control</em> panel.
To calculate a superposition
using the default settings (which are usually reasonable), use: 
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Matching ... Activate
</pre></td></tr></table>

<p><em>Hex</em> will take a few seconds to calculate the steric density functions
for each molecule. It will then systematically search
over the range of angular and intermolecular distance samples given in the
<em>Matching Control</em> panel. However, unlike docking, <em>Hex</em> first
translates the ligand&rsquo;s centre of mass to coincide with that of the
receptor. If the molecules are similar, this essentially reduces the search 
space to a rotational search (to a first approximation). Hence, the default
distance range and angular samples are set to much smaller values than are
used in a typical docking correlation search. Similarly, good global overlaps
can be found using much lower-order correlations than needed for docking.
Therefore, its usually sensible to leave all the superposition search 
parameters at their default values. Nonetheless, the defaults generally
give a search which significantly over-samples the search space, and only
one or two distinct superpositions are likely to remain after clustering.
</p>
<p>By default, superpositions are calculated using only 
the van der Waals density representation of each molecule. However, this
can be changed with the <em>Search Mode</em> selector to match shapes
using both the interior (van der Waals) and exterior (surface skin) density
functions. This doubles the computational cost, but it can be useful when 
matching a small molecular fragment onto a surface patch of a larger
molecule, because the skin-matching component ensures surface matches are
favoured over trivial inclusion matches of the fragment within the body
of the larger molecule.
</p>
<a name="index-Carbo-Index"></a>
<a name="index-Shape-Matching-Scores"></a>

<p>The shape matching algorithm re-uses much of the docking code, 
and the <em>Matching Control</em> panel works in almost exactly the same 
way as for docking (see section <a href="#Docking-Examples">Docking Examples</a>). 
One noticeable difference, however, is that similarity scores are
calculated using a Carbo-like overlap score. This is essentially a 
normalised overlap volume, scaled onto [-1000,0] for compatibility
with the existing clustering, ranking, and summary output code used by the
docking module. In other words, <em>Hex</em> always thinks &ldquo;negative is good.&rdquo;
</p>


<hr size="6">
<a name="Miscellaneous"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Molecular-Matching" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Macros" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Docking-Molecules" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="Miscellaneous-1"></a>
<h1 class="chapter">6. Miscellaneous</h1>


<hr size="6">
<a name="Macros"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Miscellaneous" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Hetero-Atoms" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Macros"></a>
<a name="Macros-2"></a>
<h2 class="section">6.1 Macros</h2>


<p>In <em>Hex</em>, macros (or scripts)
are command files which contain a sequence of instructions equivalent to
the commands that you might issue from the graphical user interface. To
create a macro file, go to the <em>Macros</em> menu and select <em>Learn</em>. 
All subsequent operations that you perform (including opening files, 
editing molecules, changing colours, etc.) will be recorded. 
When you have completed the sequence of commands that you want to record, 
select the <em>Save</em> command in the <em>Macros</em> menu. 
This will produce a small window, prompting you for a file name in
which to save the command sequence. The convention is to use a file name
of the form <strong>filename.mac</strong>, although you could adopt your own convention.
</p>


<p>Here is an example.
Lets open an antibody structure, colour it according to the residue identities
of the framework and hypervariable loop regions, and position the molecule
(using the mouse) in a pleasing orientation. The command sequence might be:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">Macros ...  Learn                                  # start recording
File ...  Open ... Receptor ... 3hfl.pdb           # existing file
Controls ... Molecule ... Colour File ... 3hfl.col # existing file
Controls ... Hetero ... Display
Solvent ... solvent display off
Lock                                               # lock home view
Macros ... Save ... test.mac                       # save commands
</pre></td></tr></table>


<p>Printing the file <strong>test.mac</strong> should show a command sequence similar to:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">open_receptor   3hfl.pdb
set_colour_file 3hfl.col
display_solvent 0
button_active   3
cursor_rotation -0.0161 0.0233
button_active   0
button_active   2
cursor_rotation 0.0121 -0.0295
button_active   0
lock_home_view  1
</pre></td></tr></table>


<p>You can now re-execute these commands using:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">File ... Close ... All
Macros ... Run ... test.mac
</pre></td></tr></table>


<p>If you make a mistake while recording a macro and want to 
cancel the current recording, select:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">Macros ...  Forget
</pre></td></tr></table>


<p>Note that running a
macro <em>does not</em> update the state of any toggles or any numerical values
in the graphical user interface. Thus the state of the display and the
values shown in the GUI controls may become inconsistent after
running a macro. Nonetheless, macros are useful for running canned sequences
of commands and for running batch jobs. Since a macro file may invoke another
macro, it is quite easy to set up a sequence of docking calculations, for
example, using the same docking parameters on a number of trial complexes.
</p>


<p>Here is a more complicated example using several macro files, starting with 
the top level macro, <strong>dock_ptc.mac</strong>, which is provided in the 
<strong>examples</strong> directory:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">#-------------------------------------------------------------------
#
# dock_ptc.mac
#
docking_receptor_samples  492
docking_ligand_samples    492
docking_alpha_samples     128

receptor_range_angle      180
ligand_range_angle         30
twist_range_angle         360

r12_range                  21
r12_step                 0.75
grid_size                0.6

docking_main_scan          16
docking_main_search        25

run_macro ptc.mac

run_macro randomise_ligand.mac

activate_docking

#-------------------------------------------------------------------
#
# ptc.mac - loads unbound protein subunits in standard orientation
#

close_all

open_receptor  2ptn_e.pdb
open_ligand    4pti_i.pdb
open_complex   2ptc.pdb

fit_ligand

#-------------------------------------------------------------------
#
# randomise_ligand.mac
#

moving_thing  1      # (edit 0=receptor, 1=ligand, 2=complex)
randomise_molecule   # (i.e. randomise the ligand orientation)
commit_view          # (commit changes)
moving_thing -1      # (done editing)
#-------------------------------------------------------------------
</pre></td></tr></table>


<a name="index-Batch-Docking-Example"></a>
<p>One could run this
macro sequence interactively, and then use the 
<code>Page/Up</code> and <code>Page/Down</code> keys to step
through displayed solutions. However, when testing the docking algorithm,
or when performing multiple docking runs, its probably easier
(and of more interest) to examine 
the calculated energies and ranks of each predicted docking orientation.
Since these values are printed as the calculations proceed,
its often convenient to run the whole thing in batch and to capture
the results in a log file.
For example, at the unix prompt, type:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">hex &lt;dock_ptc.mac &gt;dock_ptc.log  # <em>Hex</em> goes into the background
ps -ef | grep hex                # check background process running
</pre></td></tr></table>

<p>The various docking parameters are described in more detail in 
<a href="#Docking-Molecules">Docking Molecules</a>.
</p>
<p>Please be aware that 
<b>very little checking is done when reading a macro file</b>. 
Hence you should always use the GUI,
via <code>Macros ... Learn</code> and <code>Macros ... Save</code>, 
to find legal values for the macro commands. Using values which are not 
available in the GUI may crash the program or produce meaningless results.
</p>


<hr size="6">
<a name="Hetero-Atoms"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Macros" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Hardcopy-Output" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Hetero-Atoms"></a>
<a name="Hetero-Atoms-1"></a>
<h2 class="section">6.2 Hetero Atoms</h2>


<a name="index-Hetero-Control-Panel-1"></a>
<p>When <em>Hex</em> loads a protein
or DNA structure, it tries to draw all atoms including
any water molecules or other <em>hetero</em> atoms (such as ions and other small
molecules) that it finds in the PDB file. 
The <em>Hetero Control</em> panel allows a certain degree of control over how 
such atoms are treated. 
The <em>Hetero Control</em> panel is activated using:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Hetero
</pre></td></tr></table>


<p>This panel allows you
to modify the treatment of solvent and hetero atoms:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Display Solvent (toggle)
Enable Solvent  (toggle)
Display Hetero  (toggle)
Enable Hetero   (toggle)
</pre></td></tr></table>


<p>When <em>Enable Solvent</em> is selected, all water molecules are 
treated as part of the structure when calculating dot surfaces or spherical 
harmonic surfaces, for example. Toggling this option excludes solvent from 
the surface calculation. The <em>Display Solvent</em> toggle may be used
to eliminate all waters from the graphics display (but remember, they
still exist (if enabled), as far as  any calculations are concerned).
Enabled solvent atoms
are displayed in light blue: disabled atoms are displayed in grey.
The <em>Hetero</em> toggles have a similar function.  
Similarly, the <em> Enable ARG/LYS</em> toggle allows the
status of any arginine and lysine side chain atoms to be changed.
Disabling these side chains during a predictive docking run generally
reduces the sensitivity of the docking correlation, but is often highly
beneficial in the following molecular mechanics refinement step: This part of
the program is still under development.
</p>


<p>The <em>Solvent Colour</em> button activates a simple colour chart which may be 
used to select an alternative colour with which to display enabled solvent 
atoms.
</p>


<a name="index-Solvent-Culling"></a>
<p>The <em>Cull Solvent</em> button calculates which water molecules are 
completely buried within a protein - and <em>disables</em> those that are not. 
The culling calculation uses the dot surface algorithm to determine the 
accessibility of each water molecule. 
The van der Waals surface of each atom in the structure
is covered with dots: those dots which are occluded by neighbouring atoms
are deleted in a recursive search, so that any atom with no surviving dots
must be completely inaccessible to a probe sphere. Any water molecule
that has surviving surface dots must, therefore, be accessible to the probe
and is <em>culled</em>: i.e. removed from the surface atom list. The algorithm 
is repeated (automatically) until no more waters can be removed. Usually this
takes from one to four cycles, depending on how many shells of solvent
surround the molecule.
</p>


<p>As an alternative to this automatic calculation, it is also possible to 
manually select water molecules (or any other atoms) to be culled. 
Select <em>ID</em> picking mode, and pick those atoms you wish to cull.
Pressing the <em>Toggle Picked IDs</em> button (in <em>Hetero Control</em>) will 
toggle the status of any picked atoms. 
</p>


<p>Finally, using the <em>Apply to</em> selector, you can choose to apply these 
operations to all molecules (the default) or only to a selected molecule.
If more sophisticated atom selections are required, it will probably be 
necessary to edit PDB files manually, before loading them into the program.
</p>


<hr size="6">
<a name="Hardcopy-Output"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Hetero-Atoms" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Compressed-Files" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Hardcopy-Output"></a>
<a name="Hardcopy-Output-1"></a>
<h2 class="section">6.3 Hardcopy Output</h2>


<a name="index-Pipe-Output"></a>
<a name="index-GRASP"></a>
<a name="index-Molscript"></a>
<p><em>Hex</em> supports JPEG, PNG, and Postscript hardcopy output. 
Just compose your scene
and then use the appropriate <code>File ... Print</code> option to write the
scene to an image file. The <code>File ... Print</code> menu also contains a
special <em>Pipe</em> option. This causes the molecular coordinates for the
current scene to be piped (in the sense of a Unix pipe) to a script in
the <strong>$HEX_ROOT/pipes</strong> directory, which can be used to load the
current structures directly into an external program such as GRASP or 
Molscript. The <strong>$HEX_ROOT/pipes</strong> directory contains example scripts
to run these programs.
</p>


<hr size="6">
<a name="Compressed-Files"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Hardcopy-Output" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Environment-Variables" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Compressed-Files"></a>
<a name="Compressed-Files-1"></a>
<h2 class="section">6.4 Compressed Files</h2>


<a name="index-gzip"></a>
<a name="index-bzip2"></a>
<a name="index-Gzipped-Files"></a>
<a name="index-Bzipped-Files"></a>
<p><em>Hex</em> has full support for gzip (.gz) and bzip2 (.bz/.bz2)
file compression formats. If you specify an output file name with one
of these file name extensions, 
you will automatically get the corresponding type
of compressed output. Similarly, if any input file has one of these
extensions, <em>Hex</em> will automatically decompress the file as it
reads it. 
If you have installed a copy of the PDB (e.g. from the 
PDB CDROM disc set) on your hard drive, you can use <em>Hex</em> to
view the structure files without first having to decompress them.
Just point <em>Hex</em>&rsquo;s file selection widget at the top level of 
your PDB directory and start visual browsing!
</p>



<hr size="6">
<a name="Environment-Variables"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Compressed-Files" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Environment-Variables"></a>
<a name="Environment-Variables-1"></a>
<h2 class="section">6.5 Environment Variables</h2>


<p><em>Hex</em> uses several environment variables to locate directories and
data files. Most of these are optional, as described in the table below. 
The non-optional 
environment variables are defined automatically if you use the
self-installing bundles or the
<strong>hex_configure.bin</strong> script to install <em>Hex</em>.
</p>
<table>
<tr><td width="18%"><strong>Name</strong></td><td width="10%"><strong>Status</strong></td><td width="28%"><strong>Default</strong></td><td width="44%"><strong>Description</strong></td></tr>
<tr><td width="18%">HEX_CACHE</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/cache</td><td width="44%">locates a cache directory for the translation 
     matrices: <strong>highly recommended</strong></td></tr>
<tr><td width="18%">HEX_COLOURS</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/examples</td><td width="44%">locates a directory of <em>Hex</em> colour files</td></tr>
<tr><td width="18%">HEX_CPUS</td><td width="10%">optional</td><td width="28%">1</td><td width="44%">the default number of CPUs to use</td></tr>
<tr><td width="18%">HEX_GPUS</td><td width="10%">optional</td><td width="28%">1</td><td width="44%">the default number of GPUs to use</td></tr>
<tr><td width="18%">HEX_FIRST_GPU</td><td width="10%">optional</td><td width="28%">0</td><td width="44%">defines the first GPU to use for computations</td></tr>
<tr><td width="18%">HEX_DATA</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/data</td><td width="44%">locates non-standard <em>Hex</em> data files</td></tr>
<tr><td width="18%">HEX_LOG</td><td width="10%">optional</td><td width="28%">hex.log</td><td width="44%">defines the name of the current log file</td></tr>
<tr><td width="18%">HEX_MACROS</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/examples</td><td width="44%">locates a directory of <em>Hex</em> macro files</td></tr>
<tr><td width="18%">HEX_PDB</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/examples</td><td width="44%">locates a directory of PDB files</td></tr>
<tr><td width="18%">HEX_PIPE</td><td width="10%">optional</td><td width="28%">$HEX_ROOT/pipes</td><td width="44%">locates a directory of <em>Hex</em> pipe scripts</td></tr>
<tr><td width="18%">HEX_MESSAGES</td><td width="10%">optional</td><td width="28%">yes</td><td width="44%">write messages to separate output window</td></tr>
<tr><td width="18%">HEX_ROOT</td><td width="10%">required</td><td width="28%">none</td><td width="44%">locates the main <em>Hex</em> installation directory 
      (e.g. /usr/local/hex)</td></tr>
<tr><td width="18%">HEX_STARTUP</td><td width="10%">optional</td><td width="28%">none</td><td width="44%">locates a start-up macro script 
     (e.g. $HEX_ROOT/data/startup_v4.mac)</td></tr>
<tr><td width="18%">HEX_STEREO</td><td width="10%">optional</td><td width="28%">yes</td><td width="44%">defines whether a stereographic visual is requested by default</td></tr>
<tr><td width="18%">HEX_VERSION</td><td width="10%">optional</td><td width="28%">6a</td><td width="44%">the default version of <em>Hex</em> to use</td></tr>
</table>

<p>Additionally, you should ensure that your <strong>PATH</strong> contains the
<strong>$HEX_ROOT/bin</strong> directory. On systems with multiple GPUs, it
may be desirable to reserve some GPUs for display or other compute 
purposes. For example on a system with 3 GPUs, you could force Hex
to use only the last GPU by setting HEX_FIRST_GPU=2 and HEX_GPUS=1.
</p>

<hr size="6">
<a name="Command-Line-Options"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Environment-Variables" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Loading-Three-Structures-Together" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Command-Line-Options"></a>
<a name="Command-Line-Options-1"></a>
<h2 class="section">6.6 Command Line Options</h2>

<p>The general command-line format to invoke <em>Hex</em> is
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex [-options] [pdbfile(s)]
</pre></td></tr></table>


<p>where <code>options</code> may be a combination of:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">-v#           (where # is a version number/string - e.g. 4f, 5a)
-ncpu n       (where n is the number of CPUs to use)
-ngpu n       (where n is the number of GPUs to use)
-nice n       (runs at lower priority: n a &quot;nice&quot; value)
-cuda         (request program with CUDA support; the default)
-nocuda       (request program with no CUDA support)
-kill         (try to kill any stuck Hex jobs)
-g            (run within a debugger such as gdb)
-debug        (print debugging messages; another -debug adds more)
-noexec       (don't put a batch run in the background)
-gui          (force GUI mode even when stdin is not a terminal)
-stereo       (request a stereographic window)
-s            (same as -stereo)
-nostereo     (request a non-stereographic window)
-nos          (same as -nostereo)
-messages     (write messages to a separate window - the default)
-nomessages   (write messages to standard output, not separate window)
-stdout       (same as -nomessages)
-log logfile  (or -l, write log of entire session to file)
-l   logfile  (same as -log)
-pdb pdbfiles (load up to 3 PDB files to be displayed)
-browse       (same as -pdb, but forces interactive mode for browser)
-noligand     (treat two PDB files as receptor + complex)
-2            (short form of -noligand: superposes two molecules)
-pos          (position ligand near receptor - the default)
-nopos        (don't mess with ligand coordinates)
-e macfile(s) (execute up to 3 macro files)
</pre></td></tr></table>

<hr size="6">
<a name="Loading-Three-Structures-Together"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Command-Line-Options" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Loading-and-Superposing-Two-Structures" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.1 Loading Three Structures Together</h3>

<p>Load three PDB files which contain a <em>receptor</em>,
a <em>ligand</em>, and an existing <em>complex</em> structure (to be
used to calculate RMS deviations of the docking predictions), and execute a
macro file of docking commands, writing the results to a named log file:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">hex r.pdb l.pdb c.pdb -e docking_job.mac -l docking_job.log
</pre></td></tr></table>

<hr size="6">
<a name="Loading-and-Superposing-Two-Structures"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Loading-Three-Structures-Together" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-with-Macro-File-And-Log-File" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.2 Loading and Superposing Two Structures</h3>

<p>Load a <em>receptor</em> and a <em>complex</em> (i.e. no <em>ligand</em>),
and superpose the complex onto the receptor:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex -2 r.pdb c.pdb
</pre></td></tr></table>

<hr size="6">
<a name="Docking-with-Macro-File-And-Log-File"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Loading-and-Superposing-Two-Structures" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Docking-with-File-I_002fO-Redirection" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.3 Docking with Macro File And Log File</h3>

<p>Run a docking calculation (assumed to be
fully specified in a macro file), 
using 2 CPUs:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex -ncpu 2 -e job.mac -l job.log
</pre></td></tr></table>

<hr size="6">
<a name="Docking-with-File-I_002fO-Redirection"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Docking-with-Macro-File-And-Log-File" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Serialising-Several-Batch-Jobs" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.4 Docking with File I/O Redirection</h3>

<p>Run the previous example using I/O redirection:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex -ncpu 2 &lt;job.mac &gt;job.log
</pre></td></tr></table>

<hr size="6">
<a name="Serialising-Several-Batch-Jobs"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Docking-with-File-I_002fO-Redirection" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Using-Hex-as-a-Web-Browser" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.5 Serialising Several Batch Jobs</h3>

<p>If both <em>stdin</em> and <em>stdout</em> are not terminal devices, 
<em>Hex</em> automatically puts itself into the background. When running a Unix
script with a series of different jobs, this behaviour should be suppressed
using the <code>-noexec</code> option:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex -ncpu 2 -noexec &lt;job1.mac &gt;job1.log  # job2 starts after job1
hex -ncpu 2 -noexec &lt;job2.mac &gt;job2.log
</pre></td></tr></table>


<hr size="6">
<a name="Using-Hex-as-a-Web-Browser"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Serialising-Several-Batch-Jobs" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Examples-Directory" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Command-Line-Options" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h3 class="subsection">6.6.6 Using <em>Hex</em> as a Web Browser</h3>

<p>On Unix systems, <em>Hex</em> may be used as a web browser helper application 
for PDB files by specifying that <em>Hex</em> should be invoked
when the browser receives a MIME type that corresponds to a PDB file
(e.g. chemical/x-pdb). 
See the <strong>hex_browser</strong> file in the
<strong>$HEX_ROOT/bin</strong> directory for details. If anyone implements something
similar for MS-Windows, please let me know!
</p>

<hr size="6">
<a name="Examples-Directory"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Using-Hex-as-a-Web-Browser" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Bugs-and-Known-Limitations" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Examples-Directory"></a>
<a name="Examples-Directory-1"></a>
<h2 class="section">6.7 Examples Directory</h2>


<p>The <strong>examples</strong> directory, under the main <strong>hex</strong> directory,
contains PDB data files of examples of two protein complexes: 
HyHel-5/lysozyme and BPTI/Inhibitor. 
It also contains short <em>macro</em> files to load the subunits
of these complexes. 
Unless you have changed the values of the HEX_PDB and HEX_MACROS environment 
variables, <em>Hex</em> will look in the <strong>examples</strong> directory by default.
Select:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Macros ...  Run
</pre></td></tr></table>


<p>and then select one
of [<strong>3hfl.mac</strong>, <strong>2ptc.mac</strong>], or [<strong>hfl.mac</strong>, <strong>ptc.mac</strong>].
This will open the files that contain the atom
coordinates of either the crystallographically determined complex
(first pair) or the coordinates of the separately determined protein
subunits (second pair), superposed on the corresponding complex 
structure. In both cases the structure of the complex should
appear as a thin grey skeleton.
</p>


<p>The <strong>examples</strong> directory
also contains sample colour files, <strong>*.col</strong>, and a few other macro files,
<strong>*.mac</strong>, which you might wish to examine. 
In particular, the <strong>startup.mac</strong> file explains how to select an initial
colour scheme, for example. It also shows how to switch off the initial logo 
display if you prefer to start with a blank screen, and how to change
the type of browser (Netscape, by default) used to display this User Manual 
on-line.
</p>
<p>Please note, the example docking scripts <strong>dock*.mac</strong> will only
work correctly if <em>Hex</em> can find the relevant PDB files and related
macros.
Once you&rsquo;ve defined the HEX_PDB and HEX_MACRO environment variables,
to point to your own directories, 
you should copy the appropriate files from the <strong>examples</strong> directory 
into these new directories if you still want to use these scripts.
Optionally, you can also download a more extensive set of docking examples
from the <em>Hex</em> download page.
A much more extensive set of docking test cases can be obtained from
<a href="http://zlab.bu.edu/zdock/benchmark.shtml"><em>Zhiping Weng group&rsquo;s Docking Benchmark</em></a>.
</p>



<hr size="6">
<a name="Bugs-and-Known-Limitations"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Examples-Directory" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Macros-Bug" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Bugs"></a>
<a name="Bugs-and-Known-Limitations-1"></a>
<h2 class="section">6.8 Bugs and Known Limitations</h2>


<p>Please report any bugs/suggestions to the author (see section <a href="#Contacts">Contacts</a>).
Described below are some known limitations, some of which might never be
fixed.
</p>
<hr size="6">
<a name="Macros-Bug"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Bugs-and-Known-Limitations" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Non_002dStandard-Residues" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Bugs-and-Known-Limitations" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Macros-Bug"></a>
<a name="Macros-1"></a>
<h3 class="subsection">6.8.1 Macros</h3>

<p>When <em>Hex</em> runs a macro,
it doesn&rsquo;t update the state of any toggles or any numerical values in
the graphical user interface. Thus the state of the display and the values
shown in the user interface controls may become inconsistent after running
a macro.
</p>


<hr size="6">
<a name="Non_002dStandard-Residues"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Macros-Bug" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Postscript-Fonts" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Bugs-and-Known-Limitations" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Non_002dStandard-Residues-Bug"></a>
<a name="Non_002dStandard-Residues-1"></a>
<h3 class="subsection">6.8.2 Non-Standard Residues</h3>

<p>Non-standard or substituted amino acid residues may not be drawn properly, 
because <em>Hex</em> requires an entry in the <strong>atom_type.dat</strong>
and <strong>atom_bonds.dat</strong> files for each type of atom and bond, respectively.
However, <em>Hex</em> makes a reasonable attempt to draw the bonds of 
unrecognised molecules (typically small ligands).
The  sequence <code>File ... Open ... Dataset</code> is intended to allow
user-specific atom and bond data files to be loaded, but this isn&rsquo;t properly
implemented yet.
</p>

<hr size="6">
<a name="Postscript-Fonts"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Non_002dStandard-Residues" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#The-Test-Functions" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Bugs-and-Known-Limitations" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Postscript-Fonts-Bug"></a>
<a name="Postscript-Fonts-1"></a>
<h3 class="subsection">6.8.3 Postscript Fonts</h3>

<p>The <em>Graphics Font</em> selected from the font menu panel is not propagated 
to Postscript. All printed text always appears in Times-Roman.
</p>


<hr size="6">
<a name="The-Test-Functions"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Postscript-Fonts" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Technical-Information" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Test-Functions"></a>
<a name="The-Test-Functions-1"></a>
<h2 class="section">6.9 The Test Functions</h2>

<a name="index-Icosahedral-Tessellation"></a>
<p>Surface sampling using an 
<em>icosahedral tessellation</em> and surface parametrisation using 
<em>spherical harmonic</em> functions are both central to the shape 
similarity and complementarity calculations in <em>Hex</em>.
In <em>Test Mode</em>, you can view the spherical harmonic basis functions and 
the icosahedral tessellations used in the calculations. 
Go to the <em>File</em> menu and select:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Open ... Test
</pre></td></tr></table>


<a name="index-Harmonic-Surface-Control-Panel-1"></a>
<p>You now should see a spherical grid plotted as a white mesh. 
Each grid cell has approximately the same
area. Formerly, this grid was used as a look-up table to 
map an arbitrary spherical coordinate 
<em>(theta,phi)</em>
onto the nearest vertex of an icosahedral 
tessellation, but its no longer used in the current version.
You can change the number of grid cells using the <em>Mesh Order</em> slider in
the <em>Harmonic Surface</em> control panel. Select:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Graphics ... Harmonic Surface ... Mesh Order ... 10
</pre></td></tr></table>


<p>The <em>Mesh Order</em> slider controls
the number of vertices to be used in an icosahedral tessellation, by
specifying the number of subdivisions made along each edge of a regular 
icosahedron. (It also indirectly controls the number of cells in the
corresponding spherical grid).
</p>


<p>You can view the icosahedral
tessellation by selecting the L=0 spherical harmonic function:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Controls ... Harmonics ... Order ... 0
</pre></td></tr></table>


<p>and by clicking the <em>Enable Harmonics</em> button. You may need to adjust 
the zoom factor (bottom left slider) to get a better view.
The <em>Harmonics</em> button (which looks like a balloon: its
supposed to represent the L=2, M=0 harmonic function) 
on the right-hand border of the main window also acts as 
a toggle equivalent to the <em>Enable Harmonics</em> button. 
</p>


<a name="index-Spherical-Harmonic-Functions"></a>
<p>You can now display arbitrary spherical harmonic functions using the 
<em>Order (L)</em> and <em>Degree (M)</em> sliders. 
Each function is plotted by evaluating the function at the
angular coordinates of each icosahedral tessellation vertex and by setting
the radial distance for each angular sample to that of the function value.
You can control the number of sample points with the <em>Mesh Order</em> slider
(above) and you can control the display style of the functions using the
<em>Display Mode</em> selector of the <em>Harmonic Surface</em> control panel. Try:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">Graphics ... Harmonic Surface ... Display Mode ... Polyframe
</pre></td></tr></table>


<p>Note that when you
rotate or translate the scene using the mouse buttons, the display reverts
to wire-frame rendering until you release the button. This is usually faster
than re-rendering polygons during the motion. The <em>Solid Motion</em> button
(right-hand border, last button) can be used to toggle this behaviour.
</p>


<a name="index-Harmonics-Control-Panel-1"></a>
<a name="index-Spherical-Harmonic-Rotation"></a>
<p>The three sliders in the <em>Harmonics Control</em> panel labelled 
<em>Alpha</em>, <em>Beta</em> and <em>Gamma</em> are used to test Euler angle 
rotations of the spherical harmonics. Changing these
values will cause the rotated function
<em>R(alpha,beta,gamma)(Y[l,m](theta,phi))</em>
to be calculated and displayed.
Visually, this should be identical to drawing the unrotated figure and 
then manually
rotating it with the mouse by 
<em>R(gamma)</em> 
about the <em>z</em>-axis, 
<em>R(beta)</em> 
about the <em>y</em>-axis, 
followed by 
<em>R(alpha)</em> 
about the <em>z</em>-axis. 
This is an important visual test because
ensuring that the spherical harmonic functions can be rotated 
computationally is fundamental to <em>Hex</em>&rsquo;s shape superposition and 
docking calculations. 
</p>


<p>Finally, using spherical harmonics to parametrise the surface shape of a 
cube provides a further (possibly perverse) test of the method. 
You could try:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">File ...  Open ... cube
</pre></td></tr></table>


<a name="index-Spherical-Harmonic-cube"></a>
<p>and then experiment with different display styles and surface resolutions. 
Using L=16 gives a remarkably good <em>spherical harmonic cube</em>. 
In addition to rotating the cube manually,
it can be rotated by rotating the harmonic basis functions (above) or,
equivalently, by rotating the shape expansion coefficients by setting the
<em>Rotate Coefficients</em> toggle in <em>Harmonics Control</em>.
</p>

<hr size="6">
<a name="Technical-Information"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#The-Test-Functions" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Contacts" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Technical-Information"></a>
<a name="Technical-Information-1"></a>
<h2 class="section">6.10 Technical Information</h2>

<p>In order to run <em>Hex</em>,
your system should have at least 64Mb main memory but preferably more (128
Mb or greater), and about 4 MB of disc space for the program and data files.
The speed of the docking calculations can be increased by allocating an
optional <em>disc cache</em> directory of up to about 100 MB.
For molecular docking and graphics applications, I&rsquo;d recommend buying
the highest-spec Linux machine you can afford, but be sure to select a 
graphics card that has Linux support.
I personally own a Pentium III Xeon system with a Quadro4 XGL700 
graphics card (currently running Redhat 9, and Fedora 4 and 8). Its great.
</p>

<a name="index-Intel-Linux-Version"></a>
<a name="index-Intel-Linux-Version-1"></a>
<p>The latest Linux version was built on Pentium III Xeon system running
Fedora Core 8 (kernel 2.6.23, X.org 7.1, gcc 4.1).
Versions are also available for 
Fedora Core 4 (kernel 2.6.21, X.org 6.8.2, gcc 4.0),
RedHat 9.0 (kernel 2.4.20-8, XFree86 4.3.0, gcc 3.2.2), and
RedHat 8.0 (kernel 2.4.18-14, XFree86, 4.2.0, gcc 3.2-7).
Earlier versions of <em>Hex</em> are still available for some older versions
of RedHat. 
</p>
<a name="index-PowerPC-Version"></a>
<a name="index-Mac-OS-X-Version"></a>
<a name="index-Intel-Mac-Version"></a>
<p>As of <em>Hex</em> 4.5, PowerPC executables are now available for Mac OS X 
(Tiger 10.4), and Linux (Fedora Core 8). These were built on a G4 PowerBook,
but should work on other PowerPC systems. An version for the Intel Mac will
be made available soon.
</p>

<a name="index-MicroSoft-Windows-Version"></a>
<a name="index-MS_002dWindows"></a>
<a name="index-Windows-XP"></a>

<p>The Windows version was built on a Pentium III 686 system running Windows XP, 
and using the <a href="http://www.mingw.org/"><strong>MinGW</strong></a> development 
environment. As far as I know, the Windows executable works on all
recent Windows systems.
</p>
<a name="index-SGI-version"></a>
<a name="index-Silicon-Graphics-Version"></a>
<a name="index-Sun-version"></a>
<p>Due to lack of available hardware and rapidly falling demand, executables 
for SGI/IRIX and Sun/sparc are now only available up to version 4.5. Sorry.
</p>


<hr size="6">
<a name="Contacts"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Technical-Information" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Acknowledgements" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Contacts"></a>
<a name="Contacts-1"></a>
<h2 class="section">6.11 Contacts</h2>

<a name="index-Author"></a>
<p>Author: <a href="http://www.loria.fr/~ritchied/"><strong>Dave Ritchie</strong></a>.
Before e-mailing me with a problem or question, <strong>please first
check the FAQs</strong> (see section <a href="#FAQs">Frequently Asked Questions</a>). 
I always try to respond to problems, and I welcome suggestions
for future enhancements. However, if you do have a question, suggestion, 
or if you think you&rsquo;ve found a bug, <em>please</em> check the FAQs first 
(see section <a href="#FAQs">Frequently Asked Questions</a>), and check 
<a href="http://www.loria.fr/~ritchied/hex/"><em>Hex&rsquo;s Home Page</em></a>
to make sure you&rsquo;re using the lastest version. 
Please remember that <em>Hex</em>
is essentially a research program and it is provided &ldquo;as is&rdquo;, without any
guarantee of updates or user-support.
</p>


<a name="index-Acknowledgements"></a>
<hr size="6">
<a name="Acknowledgements"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Contacts" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#References" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">6.12 Acknowledgements</h2>

<a name="index-BBSRC"></a>
<a name="index-FLTK"></a>
<a name="index-The-Gimp"></a>
<a name="index-GMP"></a>
<a name="index-gzip-1"></a>
<a name="index-bzip2-1"></a>
<a name="index-Kiss-FFT-1"></a>
<a name="index-Mesa"></a>
<a name="index-MinGW"></a>
<a name="index-MKL-1"></a>
<a name="index-Stride-1"></a>
<a name="index-TexInfo"></a>
<a name="index-zlib"></a>


<p>Much of <em>Hex</em> was developed during projects (1996-2000) funded by the 
<a href="http://www.bbsrc.ac.uk/"><strong>BBSRC</strong></a>. <em>Hex</em> was written mostly
in C, but also uses some C++ for the GUI. 
The graphical user interface was built with 
<a href="http://www.fltk.org/"><strong>FLTK</strong></a>. 
The <em>Hex</em> logo and icons were created with 
<a href="http://www.gimp.org/"><strong>The Gimp</strong></a>.
The translation matrices used in <em>Hex</em> are calculated 
using the <a href="http://www.swox.com/gmp/"><strong>GMP library</strong></a>.
The 3D and 5D FFTs were implemented on Linux using the
<a href="http://www.intel.com">Intel MKL</a> library,
and on other platforms using 
<a href="http://sourceforge.net/projects/kissfft"><strong>Kiss FFT</strong></a>,
Copyright (c) 2003-2004, Mark Borgerding.
Support for compressed files is thanks to 
<a href="http://www.gzip.org/zlib/"><strong>zlib</strong></a>, 
<a href="http://www.gzip.org/"><strong>gzip</strong></a>, and
<a href="http://sources.redhat.com/bzip2"><strong>bzip2</strong></a>.
The initial Linux version of <em>Hex</em> would have been impossible without the
<a href="http://www.mesa3d.org/"><strong>Mesa</strong></a> graphics library.
The Windows version of <em>Hex</em> was built entirely under the
<a href="http://www.mingw.org/"><strong>MinGW</strong></a> environment 
(<em>Hex</em> has never been anywhere near a Microsoft compiler).
This manual was produced using 
<a href="http://www.gnu.org/software/texinfo/"><strong>texinfo</strong></a>.
</p>
<p><em>Hex</em> uses the
<a href="ftp://ftp.ebi.ac.uk/pub/software/unix/stride"><strong>Stride</strong></a> program
to assign protein secondary structures for cartoon displays. 
See section <a href="#Stride-Licence">Stride Licence</a> if you have any doubts about whether you are entitled
to use Stride.
</p>
<a name="index-People"></a>

<p>Vishwesh Venkatraman helped to re-write the CPU and GPU correlation calculations 
to use multi-threading.
</p>
<p>The following people for making significant
programming contributions to <em>Hex</em> as part of their Honours or MSc projects:
<strong>Ryan Cairns BSc</strong> (FLTK port),
<strong>Ewan Lyle BSc</strong> (ribbon cartoons),
<strong>Ioannis Zaferopoulos BSc</strong> (solid models), 
<strong>Russell Hamilton MSc</strong> (macro docking displays), 
<strong>Guillaume Valadon BSc</strong> (surface colouring).
</p>
<p><strong>Gary MacIndoe</strong> wrote a nice web interface:
<a href="http://www.loria.fr/~ritchied/hex_server/">http://www.loria.fr/~ritchied/hex_server/</a>. 
This is useful if you wish to try out <em>Hex</em> without having
to install it. However, for best results, please download the program 
from  <a href="http://www.loria.fr/~ritchied/hex/"><strong>Hex</strong></a>, 
and run it locally.
</p>



<hr size="6">
<a name="References"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Acknowledgements" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Citing-Hex" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-References"></a>
<a name="References-1"></a>
<h2 class="section">6.13 References</h2>


<ul>
<li><a name="index-GPU-Paper"></a>
</li><li> <em>Fast FFT Protein-Protein Docking on Graphics Processors</em>,
      D.W. Ritchie, and V. Venkatraman (2010), 
<em>Manuscript in preparation</em>.
<br>
<a name="index-5D-FFT-Paper"></a>
</li><li> <em>Accelerating Protein-Protein Docking Correlations Using A 
            Six-Dimensional Analytic FFT Generating Function</em>,
      D.W. Ritchie, D. Kozakov, and S. Vajda (2008), 
<em>Bioinformatics</em> <strong>24</strong>(17), 1865-1873.
<br>
<a name="index-Docking-Review-Paper"></a>
</li><li> <em>Recent Progress and Future Directions in Protein-Protein Docking</em>,
      D.W. Ritchie, (2008), 
<em>Curr. Prot. Pep. Sci.</em> <strong>9</strong>(1), 1-15.
<br>
<a href="http://www.loria.fr/~ritchied/papers/ritchie_cpps_2008.pdf">http://www.loria.fr/~ritchied/papers/ritchie_cpps_2008.pdf</a>
<br>
<a name="index-HTVS-Paper"></a>
</li><li> <em>Toward High Throughput 3D Virtual Screening using Spherical Harmonic 
            Molecular Surface Representations</em>,
      L. Mavridis and D.W. Ritchie, (2007), 
<em>J. Chem. Inf. Model.</em> <strong>47</strong>(5), 1787-1796.
<br>
<a href="http://pubs.acs.org/cgi-bin/abstract.cgi/jcisd8/2007/47/i05/abs/ci7001507.html">http://pubs.acs.org/cgi-bin/abstract.cgi/jcisd8/2007/47/i05/abs/ci7001507.html</a>
<br>
<a name="index-Translation-Matrices-Paper"></a>
</li><li> <em>High Order Analytic Translation Matrix Elements for Real Space Six-Dimensional
            Polar Fourier Correlations</em>,
      D.W. Ritchie (2005),
<em>J. Appl. Cryst.</em> <strong>38</strong>, 808-818.
<a href="http://scripts.iucr.org/cgi-bin/paper?ea5040">http://scripts.iucr.org/cgi-bin/paper?ea5040</a>
<br>
<a name="index-CAPRI-Paper"></a>
</li><li> <em>Docking Essential Dynamics Eigenstructures</em>,
      D. Mustard and D.W. Ritchie (2005),
<em>PROTEINS: Struct. Funct. Bioinf.</em> <strong>60</strong>(2), 269-274.
<br>
<a href="http://www.loria.fr/~ritchied/papers/ritchie_gaeta_2005.pdf">http://www.loria.fr/~ritchied/papers/ritchie_gaeta_2005.pdf</a>

</li><li> <em>Evaluation of Protein Docking Predictions Using <em>Hex</em> 3.1 in CAPRI 
      Rounds 1 and 2</em>, D.W. Ritchie (2003),
<em>PROTEINS: Struct. Funct. Genet.</em> <strong>52</strong>, 98-106.
<br>
<a href="http://www.loria.fr/~ritchied/papers/ritchie_capri_paper.pdf">http://www.loria.fr/~ritchied/papers/ritchie_capri_paper.pdf</a>

</li><li> <em>Protein Docking Using Spherical Polar Fourier Correlations</em>,
D.W. Ritchie and G.J.L. Kemp (2000), 
<em>PROTEINS: Struct. Funct. Genet.</em> <strong>39</strong>, 178-194.
<br>
<a href="http://www.loria.fr/~ritchied/papers/docking.pdf">http://www.loria.fr/~ritchied/papers/docking.pdf</a>

</li><li> <em>Fast Computation, Rotation, and Comparison of Low Resolution
Spherical Harmonic Molecular Surfaces</em>,
D.W. Ritchie and G.J.L. Kemp (1999), 
<em>J. Comp. Chem.</em> <strong>20</strong>, 383-395.
<br>
<a href="http://www.loria.fr/~ritchied/papers/matching.pdf">http://www.loria.fr/~ritchied/papers/matching.pdf</a>

</li><li> <em>Parametric Protein Shape Recognition</em>, D.W. Ritchie (1998),
PhD Thesis, University of Aberdeen.
<br>
<a href="http://www.loria.fr/~ritchied/papers/dwr_thesis.pdf">http://www.loria.fr/~ritchied/papers/dwr_thesis.pdf</a>


</li><li> <em>Knowledge-Based Protein Secondary Structure Assignment</em>, 
D. Frishman and P. Argos (1995),
<em>PROTEINS: Struct. Funct. Genet.</em> <strong>23</strong>, 566-579.
<br>
<a href="ftp://ftp.ebi.ac.uk/pub/software/unix/stride">ftp://ftp.ebi.ac.uk/pub/software/unix/stride</a>
</li></ul>


<hr size="6">
<a name="Citing-Hex"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#References" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Citing-Hex"></a>
<a name="Citing-Hex-1"></a>
<h2 class="section">6.14 Citing Hex</h2>

<p>When citing <em>Hex</em> in a publication, please cite whichever of the
above articles seems most appropriate, and
please also mention <em>Hex&rsquo;s Home Page</em>:
</p>
<table><tr><td>&nbsp;</td><td><pre class="display">   <a href="http://www.loria.fr/~ritchied/hex/">http://www.loria.fr/~ritchied/hex/</a>
</pre></td></tr></table>





<hr size="6">
<a name="Licences"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Citing-Hex" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Hex-Licence" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Miscellaneous" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Licences"></a>
<a name="Licences-1"></a>
<h1 class="appendix">A. Licences</h1>

<hr size="6">
<a name="Hex-Licence"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Licences" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Stride-Licence" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Licences" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Hex-Licence"></a>
<a name="Hex-Licence-1"></a>
<h2 class="section">A.1 Hex Licence</h2>

<p>This is a legal agreement between you, the LICENSEE, and the LICENSOR for
use of the computer program <em>Hex 8.X.X</em>, where &quot;8.X.X&quot; is a version
number. This licence does not apply to versions of Hex with version numbers
less than &quot;6.0&quot;.
</p>
<p>By installing copying or otherwise using <em>Hex 8.X.X</em> you agree to be bound by
the terms of this licence agreement. If you do not agree to the terms of this
licence agreement the licensor is unwilling to license the software to you.
In such circumstances, you may not use or copy the software, and you should
return the unused product within 7 days.
</p>
<p>1. Definitions
</p>
<p>(a) LICENCE means the terms and conditions contained herein.
<br>
(b) LICENSOR means 
the Institut National de Recherche en Informatique et en Automatique (INRIA),
of Domaine de Voluceau, Rocquencourt, Le Chesnay, France,
as owner of the software comprising the computer program <em>Hex 8.X.X</em>.
<br>
(c) SITE means a single
computer or internal computer network operated by the LICENSEE at the premises
of the LICENSEE specified when <em>Hex 8.X.X</em> was ordered or installed.
</p>
<p>2. Grant of Licence
</p>
<p>(a) The LICENSOR grants
the LICENSEE a non-transferable and non-exclusive licence to use <em>Hex 8.X.X</em>
in connection with computers only on the SITE to which the LICENCE applies.
<br>
(b) The LICENSEE will
not, under any circumstances, reverse engineer, decompile or disassemble
<em>Hex 8.X.X</em>.
</p>
<p>3. Licence Fee
</p>
<p>(a) In consideration
of the LICENCE granted pursuant to Section 2, the LICENSEE shall pay to
the LICENSOR the applicable LICENCE FEE within 30 days of receipt of an
appropriate invoice prepared by the LICENSOR.
<br>
(b) For Academic and
Governmental SITES, the LICENCE FEE is waived.
</p>
<p>4. Transfer of Licence
</p>
<p>(a) The LICENSEE shall
not in any way sublicense, rent, assign or transfer this LICENCE or the
computer program <em>Hex 8.X.X</em>.
</p>
<p>5. Copyright
</p>
<p>(a) The LICENSOR owns
the copyright and all and any other intellectual and industrial property
rights in <em>Hex 8.X.X</em> and its associated documentation. You must therefore treat
<em>Hex 8.X.X</em> and its associated documentation like any other copyrighted material
(e.g. a book).
<br>
(b) <em>Hex 8.X.X</em> and the
associated documentation is protected by United Kingdom copyright laws,
international treaty provisions, and all other applicable national laws.
<br>
(c) The LICENSEE acknowledges
that it does not and shall not acquire any title, copyright or any other
intellectual property rights in <em>Hex 8.X.X</em> or the associated documentation.
</p>
<p>6. Limited Warranty and Disclaimer of Warranty
</p>
<p>(a) The LICENSEE acknowledges
that software in general is not error-free and that the existence of such
errors shall not constitute a breach of this LICENCE.
<br>
(b) Save as expressly
provided under this LICENCE, no warranty or conditions express or implied,
statutory or otherwise as to condition, quality, performance, merchantability
or fitness for purpose is given or assumed by the LICENSOR in respect of
<em>Hex 8.X.X</em> and 
its associated documentation and all such warranties and conditions
are hereby excluded save to the extent that this exclusion is prohibited
by law.
</p>
<p>7. Limitations of Liability
</p>
<p>(a) Except to the extent
that such liability may not lawfully be excluded, the LICENSOR shall not
be liable for damages arising out of or in conjunction with the LICENCE
of <em>Hex 8.X.X</em> or its use or performance.
<br>
(b) In no event shall
the LICENSOR be liable for any loss of profits, business interruption,
loss of business information or other special consequential damages, even
if the LICENSOR has been advised of the possibility of such damages, or
for any claim against the LICENSEE by any other party.
</p>
<p>8. Termination
</p>
<p>(a) This LICENCE agreement
is effective from the date of purchase or installation, whichever is earlier,
of the computer program <em>Hex 8.X.X</em>.
<br>
(b) The LICENSEE can
terminate the LICENCE at any time by notifying the LICENSOR of its intention
in writing.
<br>
(c) The LICENSOR can
terminate the LICENCE, after providing the LICENSEE with 14 days prior
notice and an opportunity to remedy the situation, if the LICENSEE fails
to pay any amount due to the LICENSOR.
<br>
(d) The LICENCE shall
terminate 30 days after written notice by either party in the event of
any material or persistent breach by the other party of any of the terms
of this LICENCE which is either incapable of remedy or remains unremedied
at the end of the said period of 30 days.
<br>
(e) Upon termination
of this LICENCE, the LICENSEE shall (i) discontinue use of <em>Hex 8.X.X</em> (ii) deliver
to the LICENSOR all the software furnished by the LICENSEE pursuant hereto
together with all copies thereof and (iii) erase or destroy any of <em>Hex 8.X.X</em>
contained in the computer memory or data storage systems under the control
of the LICENSEE.
</p>
<p>9. Entire Agreement
</p>
<p>(a) This agreement
constitutes the complete and exclusive statement of the agreement between
the LICENSEE and the LICENSOR and supercedes all other oral or written
proposals, prior agreements and other prior communications between the
parties, concerning the subject matter of this agreement.
</p>
<p>10. General
</p>
<p>(a) This agreement
shall be governed by and construed in accordance with the laws of France,
and the parties prorogate the exclusive jurisdiction of the French Courts.
<br>
(b) No representation
or promise relating to, and no amendment of, this agreement shall be binding
unless it is in writing and signed by both parties.
<br>
(c) The terms and
conditions of this agreement shall previal notwithstanding any variance
with the terms and conditions of any order submitted by the LICENSEE.
<br>
(d) Except for the
LICENSEE&rsquo;s obligation to pay the LICENSOR, neither party shall be liable
for any failure to perform due to cause beyond reasonable control.
<br>
(e) No waiver by a
party of any breach of any provision of this agreement shall constitute
a waiver of any other breach of that or any other provision of this agreement.
<br>
(f) In the event that
any of the provisions of this LICENCE are held to be invalid or unenforceable,
this agreement shall be construed without such provisions.
</p>

<hr size="6">
<a name="Stride-Licence"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Hex-Licence" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Kiss-FFT-Licence" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Licences" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Stride-Licence"></a>
<a name="Stride-Licence-1"></a>
<h2 class="section">A.2 Stride Licence</h2>

<p>The following text is reproduced from the Stride Documentation:
</p>
<p><em>&ldquo;All rights reserved, 
whether the whole or part of the program is concerned.
Permission to use, copy and modify this software and its documentation is 
granted for academic use provided that:</em>
</p>
<p><em>(i) this copyright notice appears in all copies of the software and
related documentation;</em>
</p>
<p><em>(ii) the reference given below (Frishman and Argos, 1995) must be cited in
any publication of scientific results based in part or completely on the
use of this program;</em>
</p>
<p><em>(iii) bugs will be reported to the authors.</em>
</p>
<p><em>The use of the software in commercial activities 
is not allowed without a 
prior written commercial license agreement.&quot;</em>
</p>

<hr size="6">
<a name="Kiss-FFT-Licence"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Stride-Licence" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Licences" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Licences" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Kiss-FFT-Licence"></a>
<a name="Kiss-FFT-Licence-1"></a>
<h2 class="section">A.3 Kiss FFT Licence</h2>

<p>The following text is reproduced from the Kiss FFT source code:
</p>
<p><em>&ldquo;Copyright (c) 2003-2004, Mark Borgerding</em>
</p>
<p><em>All rights reserved.</em>
</p>
<p><em>Redistribution and use in source and binary forms, with or without 
modification, are permitted provided that the following conditions are met:</em>
</p>
<p><em>* Redistributions of source code must retain the above copyright notice, 
this list of conditions and the following disclaimer.</em>
</p>
<p><em>* Redistributions in binary form must reproduce the above copyright notice, 
this list of conditions and the following disclaimer in the documentation and/or 
other materials provided with the distribution.</em>
</p>
<p><em>* Neither the author nor the names of any contributors may be used to endorse 
or promote products derived from this software without specific prior written permission.</em>
</p>
<p><em>THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS &quot;AS IS&quot; AND ANY 
EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF 
MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL 
THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, 
SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, 
PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; 
OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, 
WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, 
EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.&rdquo;</em>
</p>

<hr size="6">
<a name="Installation-Guide"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Kiss-FFT-Licence" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Microsoft-Windows-Installation" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Licences" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Installation-Guide"></a>
<a name="Installation-Guide-1"></a>
<h1 class="appendix">B. Installation Guide</h1>


<hr size="6">
<a name="Microsoft-Windows-Installation"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Installation-Guide" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Mac-OS-X-Installation" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">B.1 Microsoft Windows Installation</h2>

<p>For Microsoft Windows,
just download and run the self-installer program <strong>hex-6a-win.exe</strong>. 
This adds a <em>Hex</em> icon to the Programs menu and adds the environment
variable HEX_ROOT to the registry. That&rsquo;s all you should need to get started!
However, there are several other environment variables which you can set 
to customise <em>Hex</em>&rsquo;s behaviour. These have the same names as the 
Linux version described below. In particular you may wish to define 
the HEX_STEREO and HEX_CACHE variables.
</p>
<p>By default, <em>Hex</em> will use a stereographics window if a suitable graphics
card is detected. However, this gives a slower drawing rate than non-stereo 
mode, even when viewing a non-stereo (zero parallax) scene. 
Setting HEX_STEREO=no causes
a non-stereographic window to be selected at start-up time, 
and hence should give faster graphics performance. 
During a docking calculation, <em>Hex</em> will write translation matrix 
files to the <strong>$HEX_ROOT\cache</strong> directory. 
However, you can define HEX_CACHE to point to another directory 
(possibly on another disc with more space) to over-ride the default.
</p>
<p>In Windows XP, environment variables are defined using the 
<code>Advanced</code> option in the <em>System</em> control panel.
</p>
<hr size="6">
<a name="Mac-OS-X-Installation"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Microsoft-Windows-Installation" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Linux-Installation" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">B.2 Mac OS X Installation</h2>


<p>For Mac OS X, just download and double-click the 
<strong>hex-6a-osx.pkg</strong> package file. This will install <em>Hex</em> as
an application bundle in the <code>/Applications</code> directory. Double-clicking
the <em>Hex</em> icon will run the program. The bundle automatically assigns
the correct value of the HEX_ROOT environment variable. 
You may wish to edit the <strong>hex_launch</strong> script within the bundle to
assign non-default values to the other environment variables mentioned
in this section.
</p>
<hr size="6">
<a name="Linux-Installation"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Mac-OS-X-Installation" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">B.3 Linux Installation</h2>

<p>You will need to download the installation script
<strong>hex_setup.bin</strong>, and either the
<strong>hex-6a-lnx.tgz</strong> tar file for an Intel PC
or 
<strong>hex-6a-ppc.tgz</strong> for Linux on a PowerPC machine 
(i.e. <em>not</em> running Mac OS X).
The distribution files should be installed using the shell script
<strong>hex_setup.bin</strong> by typing:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">sh hex_setup.bin
</pre></td></tr></table>


<p>This script can normally be trusted to install <em>Hex</em> correctly, 
even if you 
have already installed an earlier version of the program. The script
will add any necessary environment variables to your default shell
environment.
In cases where a central file server is used to serve files to different 
types of workstations, several different executables may be installed into 
a single <strong>hex</strong> installation directory. 
Note that in previous versions of <em>Hex</em>, the installation script
was called <strong>hex_install.bin</strong>. 
If you prefer to install the program manually, you should
create a <strong>hex</strong> directory, <code>cd</code> to that directory, 
and then uncompress the distribution files into it. 
For example, assuming you are installing the usual Linux PC version, use:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">zcat hex-6a-lnx.tgz  | tar xvpf -
</pre></td></tr></table>

<a name="index-Environment-Variables-1"></a>
<p>In order to make <em>Hex</em> work properly, your login script 
(<strong>.login</strong> or <strong>.profile</strong>, or whatever)
should set the environment variable HEX_ROOT to the <strong>hex</strong> installation
directory. You should also add <em>Hex</em>&rsquo;s <strong>bin</strong> directory to your 
PATH so that (a) you can run <em>Hex</em>, and (b) <em>Hex</em> can find the 
files it needs under HEX_ROOT.  
These two steps are normally performed by the <strong>hex_setup.bin</strong> script.
Optionally, you can also specify directories 
that specify where <em>Hex</em> should start looking for colour files, 
macro files, and PDB files,  and you can specify the location of a
temporary cache directory (all except the <strong>hex</strong> installation directories 
are optional and must be created manually). You can also specify the default
number of CPUs to use on a multi-processor system.
These parameters are described in <a href="#Environment-Variables">Environment Variables</a>.
</p>


<p>Here&rsquo;s a C-shell example (I have these lines in my <strong>.cshrc</strong> file):
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">setenv HEX_ROOT     /home/ritchied/hex
setenv HEX_PDB      .
setenv HEX_MACROS   .
setenv HEX_COLOUR   .
setenv HEX_DATA     /home/ritchied/hex/data
setenv HEX_CACHE    /tmp/ritchied/hex_cache
setenv HEX_STARTUP  /home/ritchied/hex/examples/my_startup_v5.mac
setenv HEX_CPUS     8
setenv HEX_GPUS     2
setenv HEX_MESSAGES yes
setenv HEX_STEREO   yes

setenv PATH ${PATH}:${HEX_ROOT}/bin
</pre></td></tr></table>


<p>For Bourne and Korn-shell users, the corresponding commands 
(in the <strong>.profile</strong> file) would be:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">set HEX_ROOT =     /home/ritchied/hex
set HEX_PDB =      .
set HEX_MACROS =   .
set HEX_COLOUR =   .
set HEX_DATA =     /home/ritchied/hex/data
set HEX_CACHE =    /tmp/ritchied/hex_cache
set HEX_STARTUP =  /home/ritchied/hex/examples/my_startup_v5.mac
set HEX_CPUS =     8
set HEX_GPUS =     2
set HEX_MESSAGES = yes
set HEX_STEREO =   yes

set PATH = ${PATH}:${HEX_ROOT}/bin
export PATH HEX_CACHE HEX_STARTUP HEX_CPUS HEX_STEREO
export HEX_ROOT HEX_PDB HEX_MACROS HEX_COLOUR HEX_DATA 
</pre></td></tr></table>

<p>In addition to the README and LICENCE files, 
the <strong>hex</strong> directory should contain the following 7 directories:
</p>

<table><tr><td>&nbsp;</td><td><pre class="example">bin    exe    doc    data    examples    pipes   cache
</pre></td></tr></table>

<a name="index-Pipes-Directory"></a>
<a name="index-Examples-Directory-1"></a>
<a name="index-Data-Directory"></a>

<p>Typing <code>hex</code> at the Linux command prompt should locate the 
script <strong>hex</strong> 
in the <strong>bin</strong> directory which
then launches the appropriate executable from the <strong>exe</strong> directory. 
As of version 6.0, there are normally two separate executables for each type of
system: one build with CUDA and one built without CUDA. By default, the launch 
script will run the CUDA executable if it has been installed. You can give the
option -nocuda on the command line to select explicitly the non-CUDA executable.
</p>

<p>The <strong>examples</strong> directory contains several
PDB files and example <em>Hex</em> macro files. 
Unless you set the above environment variables,
<em>Hex</em> will first look in the <strong>examples</strong> directory for any macro 
and PDB files you want to use. 
The <strong>data</strong> directory
contains the runtime data files necessary to colour and charge protein
and DNA molecules, for example.
The <strong>pipes</strong> directory contains Linux scripts which allow the molecular 
structures in the current <em>Hex</em> scene to be written directly to external
programs such as GRASP or Molscript. The <strong>cache</strong> directory is the
default location in which <em>Hex</em> will store its translation matrix files
for docking runs. You may wish to use a different location for the cache
by defining the HEX_CACHE environment variable to point to an alternative
directory.
</p>



<hr size="6">
<a name="Feature-History"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Linux-Installation" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e3" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Installation-Guide" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-Feature-History"></a>
<a name="Feature-History-1"></a>
<h1 class="appendix">C. Feature History</h1>

<p><em>Hex</em> has been on-going work for around 12 years, 
although most of the basic functionality was implemented 
in the first three years.
This section summarises the development history.
There are no entries here for versions 4.3 and 4.4 - these were 
internal development versions and were never released.
</p>

<hr size="6">
<a name="New-Features-in-Hex-Version-6_002e3"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Feature-History" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e1" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.1 New Features in <em>Hex</em> Version 6.3</h2>

<p>This is a bug-fix release from version 6.1.
</p>
<ul>
<li> Fixed some memory leaks in the docking code.
</li><li> The CUDA version now comes with the run-time libraries (CUDA 3.1) included.
      This should fix some compatibility issues between Hex and different versions
      of the CUDA run-time libraries, especially between CUDA versions 2.3 and 3.0.
</li><li> The GUI has been built using FLTK version 1.3.
</li></ul>

<hr size="6">
<a name="New-Features-in-Hex-Version-6_002e1"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e3" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e0" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.2 New Features in <em>Hex</em> Version 6.1</h2>

<ul>
<li> Macro-docking now works correctly again.
</li><li> Docking multi-model PDB files caused a memory fault in version 6.0.
      This has been fixed.
</li><li> In the refinement stage of a docking calculation, re-sampling distance
      sub-steps has now been turned off as the default behavour.
      The default correlation scan order has been changed from 16 to 18
      (this happened some time ago, but I forgot to document it!).
</li><li> The distance heuristic for setting up the initial inter-molecular
      separation has been improved. This should normally give better starting
      orientations, but it may change the results in some cases because it 
      may cause a different range of orientations to be sampled.
</li></ul>

<hr size="6">
<a name="New-Features-in-Hex-Version-6_002e0"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e1" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-5_002e1" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.3 New Features in <em>Hex</em> Version 6.0</h2>

<ul>
<li> The docking correlation code has been rewritten to be thread-safe, and
      to be able to use as many CPUs as are available.
</li><li> The 1D and 3D calculations have been ported to the run on Nvidia GPUs
      using CUDA. 
</li><li> Docking calculations may now simultaneously use as may GPUs and CPUs as
      are available.
</li><li> Switching between full-screen and window-manager modes finally works 
      properly again.
</li><li> Some old &quot;FAQs&quot; have been deleted from the manual (this means anything
      before about 2005, or Redhat 9). 
</li><li> Some command-line arguments have been changed to allow more control 
      over the number of CPUS and GPUs to use. Similarly, two new
      environment variables (HEX_GPUS and HEX_FIRST_GPU) have been added to 
      allow close control over which GPUs to use during a calculation.
</li></ul>


<hr size="6">
<a name="New-Features-in-Hex-Version-5_002e1"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-6_002e0" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-5_002e0" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.4 New Features in <em>Hex</em> Version 5.1</h2>

<ul>
<li> 5D and 3D FFT grids are scaled to match user-defined search range angles.
      This allows localised 5D FFTs to fit in much less memory, and requires
      less post-filtering.
</li><li> Added &quot;Box&quot; remarks to PDB docking output to help set up MD refinements.
</li></ul>


<hr size="6">
<a name="New-Features-in-Hex-Version-5_002e0"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-5_002e1" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e5" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.5 New Features in <em>Hex</em> Version 5.0</h2>

<ul>
<li> 3D and 5D FFT docking correlations have been implemented. 
</li><li> Linux version uses Intel MKL code for better FFT docking performance.
</li><li> Simpler set-up of docking search parameters.
</li><li> Added the ability to read SDF files as well as PDB files.
</li><li> Added the ability to recognise and draw spherical harmonic surfaces and
      surface properties calculated by the ParaSurf program 
      from Cepos Insilico Ltd.
</li><li> Simpler single-file installation tar file for Linux.
</li><li> Executables for Silicon Graphics and SunOS are no longer provided.
</li><li> Molecular surfaces may now be drawn in a new alpha-blending 
      (transparency) mode.
</li><li> Added the ability to set up initial docking orientations by specifying
      named interface residues.
</li><li> Added screen-shots and illustrations to the User Manual.
</li><li> Various minor bug-fixes.
</li></ul>



<hr size="6">
<a name="New-Features-in-Version-4_002e5"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Hex-Version-5_002e0" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e2" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.6 New Features in Version 4.5</h2>

<p><em>Hex</em> Version 4.5.
</p>
<ul>
<li> Executables now avalable for Mac OSX and Linux PowerPC machines.
</li><li> Added small distance sub-stepping for high-order correlations for
      better coverage of search space.
</li><li> The low-level correlation code is accelerated using a real fast 
      Hartley transform (FHT).
</li><li> The number of solutions that are retained after a docking run is 
      now user-definable (formerly hard-coded at 512).
</li><li> Added a user-definable cluster window size (useful when the list
      of docking solutions is large).
</li><li> Added support for one-button mouse/pointer using <em>Ctrl</em> and 
      <em>Meta</em> keys.
</li><li> Fixed Missing Ligand HETATM bug.
</li><li> Fixed various minor bugs.
</li><li> Added new environment variable HEX_MESSAGES to control use of
      text message window.

</li></ul>


<hr size="6">
<a name="New-Features-in-Version-4_002e2"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e5" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e1" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.7 New Features in Version 4.2</h2>

<p><em>Hex</em> Version 4.2.
</p>
<ul>
<li> The coordinate origin for multiple model structures is now 
      taken from the first model (not the average of all models).
</li><li> Fixed bug for PDB files containing one structure with one MODEL keyword.
</li><li> Fixed silly bug in picking (ID) mode.
</li><li> Text output to Message Window now much faster.
</li><li> There are several undocumented features currently under development.
</li></ul>


<hr size="6">
<a name="New-Features-in-Version-4_002e1"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e2" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e0" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.8 New Features in Version 4.1</h2>

<p><em>Hex</em> 4.1 is a bug-fix release. 
</p>
<ul>
<li> Double-clicking a directory in the file browser no longer crashes 
      the program.
</li><li> Cartoons are now drawn correctly for chains with blank labels.
</li><li> Fixed start-up crash up on some SGI systems (JPEG lib version fault).
</li></ul>


<hr size="6">
<a name="New-Features-in-Version-4_002e0"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e1" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e4" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.9 New Features in Version 4.0</h2>

<ul>
<li> Microsoft-Windows version now available, thanks to...
</li><li> New Graphical User Interface (GUI) - 
      made with <a href="http://www.fltk.org/"><strong>FLTK</strong></a>.
</li><li> New Message Window for text output and session logging.
</li><li> New Progress Panel for docking calculations.
</li><li> New Cartoon display mode (see section <a href="#Cartoons">Cartoons</a>).
</li><li> Shape matching now uses a Carbo-like similarity index (see section <a href="#Molecular-Matching">Molecular Matching</a>).
</li><li> GUI controls now available in full-screen mode (see section <a href="#Full-Screen-Mode">Full Screen Mode</a>).
</li><li> GUI controls and graphics window now stay responsive during docking(!).
</li><li> SGI mips3 version dropped - 
      the O2 (mips4) is now the minimum SGI platform for <em>Hex</em>.
</li></ul>


<hr size="6">
<a name="New-Features-in-Version-3_002e4"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-4_002e0" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e3" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.10 New Features in Version 3.4</h2>

<ul>
<li> Improved stereo support 
      - works with nVidia&rsquo;s 3123 and later drivers
      (see section <a href="#Stereo-Displays">Stereo Displays</a>).
<a name="index-gzip-2"></a>
<a name="index-bzip2-2"></a>
</li><li> New compressed file support
     - can read/write gzip (.gz) and bzip (.bz/bz2) files
      (see section <a href="#Compressed-Files">Compressed Files</a>).
</li><li> New 3D density superposition algorithm - 
      (see section <a href="#Molecular-Matching">Molecular Matching</a>).
<a name="index-TexInfo-1"></a>
</li><li> New User Manual -
      made with <a href="http://www.gnu.org/software/texinfo/"><em>texinfo</em></a>.
</li></ul>


<hr size="6">
<a name="New-Features-in-Version-3_002e3"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e4" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e2" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.11 New Features in Version 3.3</h2>

<p><em>Hex</em> 3.3 is a bug-fix release. Running <em>Hex</em> in batch mode
(or when standard input was not a terminal device) was broken.
The program would quickly crash with a <small>SIGSEG</small> error.  Fixed now.
</p>


<hr size="6">
<a name="New-Features-in-Version-3_002e2"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e3" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e1" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.12 New Features in Version 3.2</h2>

<p><em>Hex</em> 3.2 contains mostly minor bugfixes. However,
stereo graphics is now available on Linux systems that have
suitable hardware support:
</p>
<ul>
<li> New stereo display option - type <code>hex -stereo</code>
      to request stereo (see section <a href="#Stereo-Displays">Stereo Displays</a>).
</li></ul>



<hr size="6">
<a name="New-Features-in-Version-3_002e1"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e2" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e0" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.13 New Features in Version 3.1</h2>

<p><em>Hex</em> 3.1 represents a fairly major upgrade relative to
<em>Hex</em> 3.0, as follows:
</p>

<ul>
<li> New <em>Full Screen Display</em> option
     - type <code>F</code> to toggle (see section <a href="#Full-Screen-Mode">Full Screen Mode</a>).
</li><li> New <em>Animation Mode</em> to cycle over docked
     orientations or multi-model structures (see section <a href="#Animation">Animation</a>).
</li><li> New <em>Save Docking</em> option to write
     docking transformations to a disc file (see <a href="#Save-Docking">Save Docking</a>).
</li><li> New <em>Solid Surface</em> display modes to draw
     electrostatic and steric density functions (see section <a href="#Solid-Surfaces">Solid Surfaces</a>).
</li><li> New <em>Radial Filtering</em> modes for docking scans:
     up to 25% speed-up.
</li><li> New <em>Radial Cutoff</em> option for viewing
     large molecules (see <a href="#Radial-Cutoff-Distance">Radial Cutoff Distance</a>).
</li><li> New command <code>hex -kill</code> to clean up stuck
     processes (see section <a href="#Command-Line-Options">Command Line Options</a>).
</li><li> New <em>dynamically linked Linux executables</em>
     for hardware graphics support (see section <a href="#Technical-Information">Technical Information</a>).
</li><li> Improved multi-tasking 
     - up to 20% speed-up on dual processor systems.
</li><li> Improved accuracy for translation matrices - some performance penalty
     when creating disc cache.
</li><li> Correlation limit increased to N=32 (for large memory machines) -
     not generally recommended.
</li><li> Improved surface contouring
     - slightly slower calculation,
     but smoother surfaces and faster graphics.
</li><li> Linux RedHat 7.2 version compiled with Intel icc compiler
     - up to 10% speed-up.
</li><li> Linux RedHat 7.3 version available (also compiled with icc).
</li><li> Removed limit of 65,535 atoms per molecule.
</li><li> Fixed obscure bug with model numbers in multi-model dockings.
</li><li> Fixed bug where mis-named gamma hydrogens were silently charged
     as mercury (Hg).
</li></ul>

<p>Note, there is still technically a PDB limit of 99,999 atoms per structure
(Fortran I5 format), but <em>Hex</em> will try to read files that break this
restriction. As of version 3.1, the models in multi-model structures are now
numbered from 1, or use the actual model number given in the PDB MODEL record.
</p>


<hr size="6">
<a name="New-Features-in-Version-3_002e0"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e1" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h2 class="section">C.14 New Features in Version 3.0</h2>

<p><em>Hex</em> 3.0 contains several new features and a number of improvements
compared to <em>Hex</em> 2.4. These are mostly concerned with the docking
algorithm and the graphics displays, although several bugs reported by
users have also been fixed. The main new features are:
</p>                                                                                
<ul>
<li> Docking multiple &rdquo;models&rdquo; (e.g. from NMR structures)
</li><li> Docking very large molecules (as in the CASP2/CAPRI docking challenges)
</li><li> Clustering docking solutions
</li><li> Bumps filter for docking
</li><li> Molecular mechanics refinement of docking orientations
      (still under development)
</li><li> Docking twist search now accelerated by FFT
</li><li> Multi-file output of docking results
</li><li> New molecular surface calculation
</li><li> Solid Models: VDW spheres, licorice, tubes, etc.
</li><li> Lighting, with movable light
</li><li> Screen-shot output to JPEG, PNG, and Postscript files
</li><li> Special &rdquo;pipe&rdquo; option (e.g. to pass a scene to Molscript or Grasp)
</li><li> Better Linux support
</li><li> Improved command-line options
</li><li> Automated installation script: <strong>hex_configure.bin</strong>
</li></ul>

<p>Existing <em>Hex</em> 2.4 users: 
Please note that <em>some of the parameter names
used in macro files have changed from version 2.4</em>. Therefore,
<em>some of your existing macro files may no longer work as expected</em>.
This was unavoidable, given the many changes in the new version.
Apologies for any inconvenience.
</p>


<hr size="6">
<a name="nVidia-CUDA-Configuration"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#New-Features-in-Version-3_002e0" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-Stereo-Configuration" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Feature-History" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#nVidia-Stereo-Configuration" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-nVidia-CUDA-Configuration"></a>
<a name="nVidia-CUDA-Configuration-1"></a>
<h1 class="appendix">D. nVidia CUDA Configuration</h1>

<a name="index-cuFFT-1"></a>
<p>In order to use one or more Nvidia GPU devices in a docking calculation, 
it is necessary to download a version of Hex wich was compiled with CUDA
support, and to install a CUDA-enabled device driver 
and the cuFFT run-time library, which is part of the CUDA developer Toolkit
(<a href="http://www.nvidia.com/object/cuda_get.html">http://www.nvidia.com/object/cuda_get.html</a>, Get CUDA). 
</p>
<p>On Windows systems, Hex plus CUDA should &quot;just work&quot;. 
Some Linux systems (e.g. Ubuntu) provide a convenient way to download the 
Nvidia driver as a &quot;proprietary hardware driver&quot;. However, I recommend 
that you do NOT use this feature as it could cause problems when downloading the 
developer Toolkit. I recommend downloading both driver and toolkit directly from 
the Nvidia web site. However, if you upgrade the Linux kernel, you will need to
re-install the Nvidia drivers.
</p>
<p>In order to let the Hex executable find the CUDA run-time libraries, you will
need either to set the LD_LIBRARY_PATH environment variable to point to the
folder containing the CUDA libraries, or you will need to edit the system
file <strong>/etc/ld.so.conf</strong> to mention the location of the CUDA libraries 
if they were not in a well-known place such as <strong>/usr/lib</strong>. 
On my 64-bit workstation, I installed CUDA in <strong>/usr/local/cuda</strong>. 
I then edited <strong>/etc/ld.so.conf</strong> to make it look like:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">include /etc/ld.so.conf.d/*.conf
#
# make CUDA run-time stuff globally visible
#
/usr/local/cuda/lib64
</pre></td></tr></table>

<p>The <strong>/usr/bin/ldd</strong> tool make be used to check that all run-time 
libraries are visible to the Hex executable. Here is the result on my system:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">jordan$ ldd /home/ritchied/hex/exe/hex6-cuda.x64
	linux-vdso.so.1 =&gt;  (0x00007fff013bf000)
	libpthread.so.0 =&gt; /lib/libpthread.so.0 (0x00007f6b30e49000)
	libcufft.so.2 =&gt; /usr/local/cuda/lib64/libcufft.so.2 (0x00007f6b303f4000)
	libcudart.so.2 =&gt; /usr/local/cuda/lib64/libcudart.so.2 (0x00007f6b301b4000)
	libcuda.so.1 =&gt; /usr/lib/libcuda.so.1 (0x00007f6b2c532000)
	libGLU.so.1 =&gt; /usr/lib/libGLU.so.1 (0x00007f6b2c2c1000)
	libGL.so.1 =&gt; /usr/lib/libGL.so.1 (0x00007f6b2c0c2000)
	libXft.so.2 =&gt; /usr/lib/libXft.so.2 (0x00007f6b2bead000)
	libXrender.so.1 =&gt; /usr/lib/libXrender.so.1 (0x00007f6b2bca3000)
	libfontconfig.so.1 =&gt; /usr/lib/libfontconfig.so.1 (0x00007f6b2ba71000)
	libexpat.so.1 =&gt; /lib/libexpat.so.1 (0x00007f6b2b848000)
	libfreetype.so.6 =&gt; /usr/lib/libfreetype.so.6 (0x00007f6b2b5c3000)
	libXext.so.6 =&gt; /usr/lib/libXext.so.6 (0x00007f6b2b3b1000)
	libX11.so.6 =&gt; /usr/lib/libX11.so.6 (0x00007f6b2b07b000)
	libm.so.6 =&gt; /lib/libm.so.6 (0x00007f6b2adf7000)
	libgcc_s.so.1 =&gt; /lib/libgcc_s.so.1 (0x00007f6b2abe0000)
	libc.so.6 =&gt; /lib/libc.so.6 (0x00007f6b2a871000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f6b31065000)
	libdl.so.2 =&gt; /lib/libdl.so.2 (0x00007f6b2a66d000)
	libstdc++.so.6 =&gt; /usr/lib/libstdc++.so.6 (0x00007f6b2a35d000)
	librt.so.1 =&gt; /lib/librt.so.1 (0x00007f6b2a155000)
	libz.so.1 =&gt; /lib/libz.so.1 (0x00007f6b29f3e000)
	libGLcore.so.1 =&gt; /usr/lib/libGLcore.so.1 (0x00007f6b282e7000)
	libnvidia-tls.so.1 =&gt; /usr/lib/tls/libnvidia-tls.so.1 (0x00007f6b3115b000)
	libXau.so.6 =&gt; /usr/lib/libXau.so.6 (0x00007f6b280e4000)
	libxcb.so.1 =&gt; /usr/lib/libxcb.so.1 (0x00007f6b27ec8000)
	libXdmcp.so.6 =&gt; /usr/lib/libXdmcp.so.6 (0x00007f6b27cc3000)
</pre></td></tr></table>






<hr size="6">
<a name="nVidia-Stereo-Configuration"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#FAQs" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#nVidia-CUDA-Configuration" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#FAQs" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-nVidia-Stereo-Configuration"></a>
<a name="nVidia-Stereo-Configuration-1"></a>
<h1 class="appendix">E. nVidia Stereo Configuration</h1>

<p>For OpenGL stereo-in-a-window and full-screen stereo, you will need a 
Quadro FX-1800 graphics card or higher
(<a href="http://www.nvidia.com/object/quadro_pro_graphics_boards.html">http://www.nvidia.com/object/quadro_pro_graphics_boards.html</a>, Nvidia Quadro stereo).
For stereo on a flat panel display, you will need a device with a refresh rate
of 100Hz or higher. Nvidia recommend the Samsung SyncMaster 2233RZ or the
Viewsonic VX2265 
(<a href="http://www.nvidia.com/object/3D_Vision_Requirements.html">http://www.nvidia.com/object/3D_Vision_Requirements.html</a>, flat panel stereo).
If you have the right hardware and Windows Vista or Windows 7, stereographics should &quot;just work&quot;.
</p>
<p>On Linux, configuring your nVidia graphics card for stereo should be straight-forward. 
Just download the latest driver from 
<a href="http://www.nvidia.com/view.asp?PAGE=linux">http://www.nvidia.com/view.asp?PAGE=linux</a>, and follow the 
instructions. However, as mentioned above,
<strong>make sure to de-install any earlier nVidia drivers first.</strong>
</p>
<p>Here&rsquo;s what the Device Section of my <strong>/etc/X11/xorg.conf</strong>
file looks like after editing nVidia&rsquo;s supplied template:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">#
#  Stereo is not compatible with the compositing feature: 
#
Section &quot;Extensions&quot;
    Option         &quot;Composite&quot;         &quot;Disable&quot;
EndSection
#
#  If you have multiple graphics cards, use /usr/bin/lspci to find 
#  the PCI bus ID of the specific device to use
#
Section &quot;Device&quot;
   Identifier             &quot;NV AGP&quot;
   VendorName             &quot;nvidia&quot;
   Driver                 &quot;nvidia&quot;
   BusID                  &quot;PCI:1:0:0&quot;
#  Option &quot;Stereo&quot;        &quot;0&quot;    # 0=off 
   Option &quot;Stereo&quot;        &quot;1&quot;    # 1=DCC with cable splitter
#  Option &quot;Stereo&quot;        &quot;3&quot;    # 3=3-pin DIN connector
#  Option &quot;Stereo&quot;        &quot;10&quot;   # 10=USB IR emitter for 3D Vision kit.
   Option &quot;metamodes&quot; &quot;DFP: 1680x1050_120 +0+0; DFP: 1680x1050 +0+0&quot;
   Option &quot;NoLogo&quot;        &quot;true&quot;
   Option &quot;Overlay&quot;       &quot;false&quot;
   Option &quot;UseEdidFreqs&quot;  &quot;true&quot;
   Option &quot;NvAgp&quot;         &quot;1&quot; # 0=off 1=nvidia, 2=AGPGART, 3=try 2, 1
EndSection
</pre></td></tr></table>

<a name="index-DCC-Stereo-Mode"></a>
<p>I verified these settings (and that stereo was enabled) by running
the <strong>glxinfo</strong> program.
Although my old XGL700 has an on-board stereo DIN (3-pin) socket, I
couldn&rsquo;t get a signal off this with the 3123/4191 drivers.
Currently, stereo synching only seems to
work in DCC (Data Control Channel) mode.
This requires a monitor cable &ldquo;splitter&rdquo;,
to which the stereo glasses are connected. Such &ldquo;splitters&rdquo; often come
included with stereo glasses kits.
For best performance, use nVidia&rsquo;s
AGP driver (part of NVidia&rsquo;s driver distribution), and NOT the
Linux AGP driver (/dev/agpgart, etc.). Also, use a kernel built with the
MTRR option (memory type-range registers) if your machine supports this.
</p>
<a name="index-Nvidia-3D-Vision"></a>
<p>Provided you have suitable hardware, stereographics on flat panel displays
using Nvidia&rsquo;s recent 3D Vision kit should work directly on Windows Vista and Windows 7. 
For Linux, you will need to download and install a recent version
(195.36.15  = 19 march 2010, or later)
of the driver
(<a href="http://www.nvidia.com/Download/index5.aspx?lang=en-us">Nvidia drivers</a>,
and set <strong>Option &quot;Stereo&quot; &quot;10&quot;</strong> in the <strong>/etc/X11/xorg.conf</strong> file.
In order to let the nvidia driver detect the USB infra-red emitter, it is necessary
to map USB devices to a pseudo-file system (which is not done automatically in
Ubuntu Linux). Here is an extract from my <strong>/etc/fstab</strong> file:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example"># /etc/fstab: static file system information.
#
#  mount the usb device pseudo file system
#
none    /proc/bus/usb   usbfs defaults      0        0
</pre></td></tr></table>

<p>Here is a reduced extract from 
<strong>/var/log/Xorg.0.log</strong> on my 64-bit FX-5800 workstation.
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">(**) NVIDIA(0): Option &quot;Stereo&quot; &quot;10&quot;
(**) NVIDIA(0): Option &quot;TwinView&quot; &quot;0&quot;
(**) NVIDIA(0): Option &quot;MetaModes&quot; &quot;1680x1050_120 +0+0; 1680x1050 +0+0&quot;
...
(**) NVIDIA(0): USB IR emitter stereo requested
(**) NVIDIA(0): Enabling RENDER acceleration
(II) NVIDIA(0): NVIDIA GPU Quadro FX 5800 (GT200GL) at PCI:2:0:0 (GPU-0)
(--) NVIDIA(0): Memory: 4194304 kBytes
(--) NVIDIA(0): VideoBIOS: 62.00.3a.00.03
(II) NVIDIA(0): Detected PCI Express Link width: 16X
(--) NVIDIA(0): Interlaced video modes are supported on this GPU
(--) NVIDIA(0): Connected display device(s) on Quadro FX 5800 at PCI:2:0:0:
(--) NVIDIA(0):     Samsung SyncMaster (DFP-1)
(--) NVIDIA(0): Samsung SyncMaster (DFP-1): 330.0 MHz maximum pixel clock
(--) NVIDIA(0): Samsung SyncMaster (DFP-1): Internal Dual Link TMDS
(II) NVIDIA(0): Validated modes:
(II) NVIDIA(0):     &quot;DFP:1680x1050_120+0+0&quot;
(II) NVIDIA(0):     &quot;DFP:1680x1050+0+0&quot;
(II) NVIDIA(0): Virtual screen size determined to be 1680 x 1050
(--) NVIDIA(0): DPI set to (88, 88); computed from &quot;UseEdidDpi&quot; X config
(--) NVIDIA(0):     option
(==) NVIDIA(0): Disabling 32-bit ARGB GLX visuals.
...
(II) NVIDIA(0): USB IR emitter - Copyright (c) 2009 NVIDIA Corp
(II) NVIDIA(0):     stereo controller
(II) NVIDIA(0): Initialized GPU GART.
(II) NVIDIA(0): Setting mode &quot;DFP:1680x1050_120+0+0&quot;
(II) Loading extension NV-GLX
(II) NVIDIA(0): Initialized OpenGL Acceleration
</pre></td></tr></table>

<p>However, for 3D vision to work on many Quadro models, it is necessary to connect
the IR emitter to the GPU using a mini-DIN-to-1/8-inch connector to get the
synchronisation signal for the glasses. This cable was missing in my 3D Vision
shipment (ordered in France), but the Nvidia support team quickly mailed me the
necessary part.
</p>


<hr size="6">
<a name="FAQs"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#nVidia-Stereo-Configuration" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Next section in reading order"> &gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#nVidia-Stereo-Configuration" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Next chapter"> &gt;&gt; </a>]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<a name="index-FAQs"></a>
<a name="Frequently-Asked-Questions"></a>
<h1 class="appendix">F. Frequently Asked Questions</h1>

<p>Actually, the questions listed below aren&rsquo;t so frequently-asked at all. 
As of January 2010, the program has been downloaded by about 15,000 people.
I have had almost no problems/complaints since version 5.1. But here are
the main points people have raised...
</p>

<a name="index-GPU-precision"></a>
<a name="index-CPU-precision"></a>
<ul>
<li> <strong>Why do some of the clusters change between the first and second runs of <em>Hex</em>,
              and why do the CPU and GPU results show similar differences?</strong>
</li></ul>

<p>The differences you see are mainly due to loss of arithmetic precision. The translation matrices in the cache are first calculated in extended precision using the GMP library. They are then converted to double precision for the CPU FFT calculation, and saved on disc as single precision. So the results from the first run will differ from those of the second run. This does not affect the calculated energy of each individual docking orientation very much but it can change the membership of the clusters, as you have already observed.
</p>
<p>I made the decision to use single precision for the cache about 10 years ago! Maybe I should change this now, but it will not make a big difference to the overall results.
</p>
<p>You will see similar fluctuations when comparing CPU and GPU results because the GPU FFT calculations are done in single precision, whereas the CPU calculations are done in double precision. This effect was tablulated as supplementary material in our Bioinformatics paper.
</p>
<p>On the other hand, since the docking calculations use discrete sampling without minimisation, the result will also depend on the starting coordinates. Moving one or both proteins, using a different FFT scheme, or changing the FFT sampling density will all cause similar (probably larger) fluctuations in the clusters. I am not sure, but assuming all else is equal, this might explain why you see different results for 1BRS on your machine compared to our server.
</p>
<p>Personally, I like to treat the log of the rank of each solution as a relative confidence level, bearing in mind that of course they could all be wrong!
</p>





<a name="index-Stride-2"></a>
<ul>
<li><a name="index-Broken-Ribbons"></a>
</li><li> <strong>Why are ribbon cartoons sometimes not drawn properly?</strong>
</li></ul>

<p>Hex uses the Stride program to calculate secondary structure element types.
Stride needs at least 5 residues to work with. Hence it cannot assign 
secordary structures shorter than this, and Hex will therefore draw a red
tube (which normally represents a loop) in such cases. Additionally, if a
PDB file contains two chains with the same chain label, Stride will skip
the second chain, and only the first chain will be drawn correctly in
cartoon mode. This can happen quite often if one uses Hex to view
previously calculated complexes, for example.
</p>

<ul>
<li><a name="index-Source-Code-FAQ"></a>
</li><li> <strong>I noticed you supply only binary executables. Is there any
chance of getting a copy of the source code?</strong>
</li></ul>

<a name="index-VMD"></a>
<a name="index-MolMol"></a>
<a name="index-Laguerre-Polynomials"></a>
<p>In general, the answer is no. At least not yet. <em>Hex</em> represents many 
years of work and it implements some ideas that have not been published yet. 
I would also like to 
keep open the possibility of potential future commercialisation, which might 
be difficult if the source is freely available. If your main interest in
the source is just to poke around to see how it works, there are other
OpenGL-based molecular graphics packages which do provide source that you 
could take a look at instead 
(e.g. <a href="http://www.ks.uiuc.edu/Research/vmd/">VMD</a>, 
<a href="http://www.mol.biol.ethz.ch/wuthrich/software/molmol/">MolMol</a>). 
If you&rsquo;re more interested in the 
computational parts (spherical harmonics, Laguerre polynomials, etc.), these 
should be quite easy for any reasonably competent C or Fortran programmer to 
code up anyway. In terms of lines of code, this represents only a very
small proportion of the program.  
</p>

<p>On the other hand, I can see that putting it all together to make a new docking
program (or other application) is not trivial. I am considering making all
the polar Fourier code available as a dynamic runtime library (DLL or .so). 
Please let me know if you are interested in this idea...
</p>
<ul>
<li><a name="index-Linux-Install-FAQ"></a>
</li><li> <strong>I downloaded the Linux version of Hex, followed all
the instructions, but still nothing works. What do I try next?</strong>
</li></ul>

<p>I&rsquo;m not aware of any recent Intel Linux systems it doesn&rsquo;t run on. However,
I can&rsquo;t guarantee that <em>Hex</em> will run on every Linux system, because a lot
depends on where the run-time libraries are located (eg. /usr/lib/libc.so,
etc.), although I&rsquo;ve tried to minimise such dependencies.
</p>
<p>There are really two main reasons why <em>Hex</em> might not work. 
Assuming you have downloaded an appropriate executable, the problem is
probably due to incorrect file permissions or the executable search path.
If nothing happens when you type <code>hex</code>, its probably a problem with the
HEX_ROOT and PATH variables (check the README file). If these look OK,
change directory to the hex/exe directory and type <code>./hex6a.i586</code>
to start the program without the aid of the
front-end startup script - it should give you a graphics
window with the <em>Hex</em> logo on a black background. 
The 5 part of the filename
corresponds to version 5.0 - if you have a different version of the program,
the executables will have slightly different names.
Anyway, if that works, the
problem is definitely with HEX_ROOT or PATH - in which case, check that
HEX_ROOT is set to the directory you installed hex in (i.e. /your/stuff/hex
without anything after the hex bit): also check that the PATH variable
includes $HEX_ROOT/bin - i.e. /your/stuff/hex/bin in amongst the other
colon-separated bits of the PATH (sorry if this is all obvious,
but different people have different levels of knowledge). You should
be able to type <code>which hex</code> and get back <strong>/your/stuff/hex/bin</strong>
if the PATH is properly set up.
</p>
<p>If running stand-alone ./hex5.i586 doesn&rsquo;t work, the next thing to try
is the <code>ldd</code> utility.  This shows the dependency list of any shared
libraries that a program was linked against:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">% ldd hex5d.i586
</pre></td></tr></table>

<p>The output on my Linux system looks like this:
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">linux-gate.so.1 =&gt;  (0x00b36000)
libGLU.so.1 =&gt; /usr/X11R6/lib/libGLU.so.1 (0x028f3000)
libGL.so.1 =&gt; /usr/lib/libGL.so.1 (0x00a7d000)
libXext.so.6 =&gt; /usr/X11R6/lib/libXext.so.6 (0x00d6a000)
libX11.so.6 =&gt; /usr/X11R6/lib/libX11.so.6 (0x00c94000)
libpthread.so.0 =&gt; /lib/libpthread.so.0 (0x00d7b000)
libstdc++.so.6 =&gt; /usr/lib/libstdc++.so.6 (0x008e7000)
libm.so.6 =&gt; /lib/libm.so.6 (0x00c53000)
libgcc_s.so.1 =&gt; /lib/libgcc_s.so.1 (0x008db000)
libc.so.6 =&gt; /lib/libc.so.6 (0x00112000)
libGLcore.so.1 =&gt; /usr/lib/libGLcore.so.1 (0x00d8d000)
libnvidia-tls.so.1 =&gt; /usr/lib/tls/libnvidia-tls.so.1 (0x00b23000)
libdl.so.2 =&gt; /lib/libdl.so.2 (0x00c79000)
/lib/ld-linux.so.2 (0x00b05000)
</pre></td></tr></table>

<p>If there are any blanks in the ldd output 
or if the version numbers of the libraries in
your /lib directory are hopelessly below the numbers here, then this is
probably the problem. You would then need to try installing a version 
that was built for a more similar vintage of your operating system,
or (better) you should update your entire operating system.
</p>
<p>Otherwise, you&rsquo;re out of luck for the moment - but I&rsquo;d appreciate it if
you could send me some details of your system (e.g. Linux vendor &amp; release,
output from <code>uname -a</code>, etc.) and particularly the
exact message you get after typing ./hex5.i586 - I&rsquo;ll try to figure out
what needs to be done and if possible make up a new Linux release to fix it.
</p>


<ul>
<li><a name="index-connect-error"></a>
<a name="index-bind-error"></a>
<a name="index-startup-crash"></a>
</li><li> <strong>When I try to start Hex, it prints some brief messages
about &ldquo;connect&rdquo; or &ldquo;bind&rdquo; and then just seems to get stuck -
it never shows the graphics window?
What should I do?</strong>
</li></ul>

<p>By default, <em>Hex</em> writes all its information and warning messages to
a dedicated text output window. 
This window is is controlled by a separate subprocess 
(so you will normally see <em>two</em> <em>Hex</em> processes running in
your system&rsquo;s process or task monitor tool), and Unix-style
sockets are used to communicate between the two processes. The error messages 
referring to &ldquo;bind&rdquo; or &ldquo;connect&rdquo; indicate there was a problem creating
the message window subprocess. 
This could happen if networking is disabled on your machine (unlikely), 
if there is a problem with the machine&rsquo;s hostname (possible), or
on very slow machines (likely, due to time-outs). 
<em>Hex</em> always assumes that the standard name <em>localhost</em> 
(IP address 127.0.0.1) exists, and that networking is enabled to the
extent that <code>ping localhost</code> works. 
On Unix/Linux machines, you can stop <em>Hex</em>  
from trying to create a message window by typing
<code>hex -nomessages</code>, 
or by setting the environment variable HEX_MESSAGES=no, in which case 
<em>Hex</em> will print all its messages to the terminal window. 
Windows XP does not use terminal windows, and <em>Hex</em> relies on the
message window process to show all text messages.
</p>


<ul>
<li><a name="index-Linux-I586_002fI686-FAQ"></a>
</li><li> <strong>I noticed you only provide an i586 executable for 
Linux. I have an i686/i786 system, so could you supply an executable for my
machine. It might run faster?</strong>
</li></ul>

<p>I compared the docking performance of <em>Hex</em> compiled for a 686 processor
on my 686 system with a 585-compiled version, and there was hardly any difference. 
Docking performance depends almost entirely on the efficiency of a few
floating point operations, and the larger 686 instruction set doesn&rsquo;t 
make floating point operations go any faster. Having said that, compiling with 
Intel&rsquo;s 
<a href="http://www.intel.com/software/products/compilers/">icc</a>
gives
about a 10% speed-up over GNU&rsquo;s 
<a href="http://gcc.gnu.org/">gcc-2.96</a>,
and the release versions
of <em>Hex</em> are compiled with icc whenever possible (sorry, GNU).
</p>
<ul>
<li><a name="index-Non_002dStandard-Residues-FAQ"></a>
</li><li> <strong>I want to display and dock a protein with non-standard
residues. What do I do?</strong>
</li></ul>

<p>The $HEX_ROOT/data directory contains several files 
(<strong>residue_types.dat, atom_bonds.dat atom_names.dat, atom_types.dat</strong>)
which define the atom connectivity, charges, and names of non-standard residues. You will
need to define new entries in these files to define your new residues.
Probably the easiest approach is to identify the most similar existing 
amino acid
and to copy/rename/edit the entries. You may need to calculate/guess 
some partial charges for the new atoms.
</p>
<ul>
<li><a name="index-Refine-Docking-FAQ"></a>
</li><li> <strong>How can I refine the results from a docking correlation?</strong>
</li></ul>

<p>It is possible to further refine results from the docking correlation
in <em>Hex</em> version 3.
This &ldquo;refinement&rdquo; applies a soft molecular-mechanics energy evaluation
and/or minimisation to the top 1000 orientations. However, if the amount
of conformational change on binding is large, this refinement could
eliminate what was a good solution. On the other hand, if the conformational
change is small, then the calculation should lower the energy of a good
orientation and significantly raise the energy of other orientations.
I think its hard to assess how good the predicted orientations might be,
and which of them might correspond to the actual solution. It is certainly
hard to devise a scoring function which can reliably eliminate 
&ldquo;false-positives&rdquo; (Camacho &amp; Vajda, Curr Op Struct Biol, 2002 12:36-40).
Camacho &amp; Vajda, and others, have proposed using a combination of
algorithms, and to look for consensus orientations, but I have doubts.
Personally, I think the surest approach is to use whatever experimental
or bioinformatics evidence (e.g. correlated mutations) might be available
to help suggest the location of the binding sites, and then to use a
docking algorithm to create some feasible models which can subsequently
be challenged by lab-based experiments.
</p>
<ul>
<li><a name="index-Electrostatic-Correlations-FAQ"></a>
</li><li> <strong>How can I perform a purely electrostatic correlation?</strong>
</li></ul>

<p>Its not possible to do a pure electrostatic correlation in <em>Hex</em>. 
However, it should be possible to increase the weight of the
electrostatic correlation in a shape + electrostatic calculation.
For example, if you create a small <em>macro file</em> containing
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">factor_g 7000
</pre></td></tr></table>

<p>and then run this macro using the <code>Macros ... Run</code> menu, all subsequent
docking calculations will have electrostatic energies scaled up by
a factor of 10. Thus electrostatics should dominate. Generally, I
would say that shape complementarity is more important that electrostatics, 
but if you know electrostatics are important for a given system, then this 
may be worth trying.
</p>
<ul>
<li><a name="index-Electrostatic-Correlations-FAQ-1"></a>
</li><li> <strong>When I enable electrostatics, I only get one docking
solution. What is going on?</strong>
</li></ul>

<p>Its likely that your structures contain unrecognised residues, 
and/or unrecognised
atoms. <em>Hex</em> assigns a default charge and radius to any unrecognised
atoms, normally based on the first letter of each atom name. These defaults are 
usually reasonable, but if a structure has a large number of unrecognised 
hydrogens, <em>Hex</em> might assign 0.5e to each hydrogen to give a highly positively
charge protein. A pair of such proteins will have a very unfavourable electrostatic
interaction, and docking such a pair will give a total of zero low energy
orientations. In order to avoid having zero-length arrays within the program, 
<em>Hex</em> forces at least one solution to be reported.
</p>
<ul>
<li><a name="index-Commercial-Licence-FAQ"></a>
</li><li> <strong>What is the licensing fee for commercial user?</strong>
</li></ul>

<p>There is currently no licence fee for commercial users. But this does not mean
the software is &ldquo;free&rdquo;. When you download or otherwise obtain a copy 
of <em>Hex</em> 
you are agreeing to be bound by the <em>Hex</em> Licence Agreement.  
You are also asked to obtain prior agreement with the author if the 
program is to be used for profit.
This might seem to be unnecessarily complicated, but the intention is 
to make the software available and to publicise the algorithms used, 
without compromising the possibility of potential future commercialisation. 
These days, universities like to 
protect their assets, and I want to protect my assetc ;-).
</p>

<ul>
<li><a name="index-Reference-Energy-FAQ"></a>
</li><li> <strong>What is the reference energy for electrostatic calculations
in Hex, and in thermodynamic terms are you working with G or U?</strong>
</li></ul>

<p>In <em>Hex</em>, the reference (zero) energy point is for two proteins at
infinite separation. i.e. zero interaction. Negative scores are
favourable, positive unfavourable 
(i.e. the molecules interpenetrate, or have poor surface contacts).
The electrostatic energy <em>Hex</em> calculates corresponds to the
classical internal energy (U) of a system of point charges. The
shape complementarity score, or &ldquo;hydrophobic energy&rdquo;, is scaled to 
comparable units (but forced to have more negative energies). 
This pseudo energy is also closer to U than G, but could easily be in error
by 100&rsquo;s of KJ/mol compared to experimental measurements.
I have compared calculated <em>Hex</em> energies of complexes with equilibrium constants
(i.e. ignoring the difference between U and G due to entropy)
for some antibody/antigen complexes and found no correlation. So, the calculated
energies are pretty meaningless on an absolute scale.
But I&rsquo;m sure this is also true of most (all?) other macromolecular docking 
software.
</p>
<ul>
<li><a name="index-Docking-Parameters-FAQ"></a>
</li><li> <strong>When I change some of the parameters in the docking control 
panel, such as the intermolecular distance, I get different docking results. 
What&rsquo;s going on?</strong>
</li></ul>

<p>The first few controls in the Docking Control panel define the search 
<a name="index-Docking-Control-Panel-4"></a>
increments that <em>Hex</em> will use during a docking search. 
If you increase the number of
steps (i.e. use more and smaller steps) then the rank of the 
&ldquo;right&rdquo; orientation
is likely to be reduced because <em>Hex</em> will find more low-energy 
false-positive
orientations during its search. This effect can largely be alleviated by 
ensuring you <em>cluster</em> the generated solutions using the 
Clustering Control panel.  In general, its better to over-sample the search 
space and then cluster the
solutions rather than risk missing good solutions by under-sampling in order to
make the program run faster. 
</p>
<p>The default value of the Distance Range in the Docking Control panel may be too 
small for large protein-protein complexes. Hence, if you find that increasing
the distance range gives a completely different set of docking orientations, its
likely that you were seriously under-sampling the search space in the earlier
run. In other words, if the distance range is too small, the program is not 
going to explore all possible surface-surface contacts as one molecule is 
rotated against the other. 
</p>
<p><em>Hex</em> samples radial distances &ldquo;symmetrically&rdquo; 
from the starting orientation according to (Starting Distance) +/- 
(Distance Range)/2 in equal steps of +/-(Step Size). 
One way to work out what distance range to specify for a global search is to 
sketch each protein as a plane cross-section in such a way that the plane chosen
for each protein includes its centroid, and the largest and smallest radial rays
from the centroid to the surface. Then, set the initial inter-centroid distance
(the starting distance) to the average of the sums of the smallest and largest
radii. For example, if the smallest radii sum to 28A, and the largest radii sum 
to 46A, you should set the starting separation to (28+46)/2=37A, and the
Distance Range to 2*(46-37)=18A. Using the default Step Size (0.75A), 
<em>Hex</em> will then explore 18/0.75=25 trial distances (i.e. 12 inward and 12 
outward distance steps, plus 1 for the initial distance). 
</p>
<p>In practice, its often
not possible to calculate the radii exactly, and hence set an optimal initial
intermolecular separation. An alternative approach is to enable the Radial Envelope
Filter, and to set a large Distance Range. When the radial filter is enabled,
<em>Hex</em> will use low-resolution spherical harmonic surface approximations of 
each surface to select which intermolecular distances to evaluate at each 
angular pose of each molecule. 
However, because these surface representations are 
inexact (they assume the proteins are globular, or star-like), there is a 
possibility that some orientations that should be evaluated might be skipped.
</p>
<ul>
<li><a name="index-No-Docking-Solutions-FAQ"></a>
</li><li> <strong>I loaded a receptor and a ligand into Hex,
ran a default docking calculation, but nothing seemed to happen. At
the end of the run, the ligand hadn&rsquo;t moved! What&rsquo;s going on?</strong>
</li></ul>

<p><em>Hex</em> should always place the ligand near the receptor in tightly
fitting orientations. You should get a ranked list of such orientations
at the end of each run, and you should be able to scroll through 
these interactively using the Page/Down and Page/Up (and Home
and End) keys on the keyboard. Whether any of these orientientations
actually represent the right answer, is of course, a completely separate
matter!!
In any case, if you are not seeing this, it is
probably because you need to place the two molecules quite
close together before starting the calculation. <em>Hex</em> searches
over the range of intermolecular separations and angular orientations as
specified in the Docking Control panel. If the molecules are
initially too far apart, <em>Hex</em> will not search over (relatively)
small separations unless you specify a large distance range
in the docking search. But this will add significantly to the
execution time. If you know the location of the ligand binding
site, you should manually move the ligand to be close to the
binding site (this can be done in <em>Edit Molecule</em> mode), and
then restrict the receptor rotation angle range to a small value,
say 45 degrees. If you know (or have a hunch about) the binding
mode of the ligand, you can also use Edit Molecule mode to 
orient the binding surface of the ligand towards the receptor
binding site, and then also restrict the ligand rotation angle
similarly. When pre-positioning the molecules, it may help to
enable the inter-molecular axis display (the double-headed arrow
on the right-hand border). This shows the current <em>z</em>-axis as a
white line, about which the docking is performed. The end-points 
of this line are the molecular origins (which can also be changed
using <em>Edit Origin</em> mode). Please also look at the answer given
for the previous FAQ.
</p>
<ul>
<li><a name="index-Internal-Energies"></a>
</li><li> <strong>What does &quot;Etot&quot; mean and how do I find the internal 
              energies of the proteins?</strong>
</li></ul>
                                                                                
<p>Etot is the total calculated interaction energy of the system.
<em>Hex</em> doesn&rsquo;t calculate internal energies, and so
doesn&rsquo;t report separate individual molecular energies.
</p>                                                                                
<ul>
<li><a name="index-Reloading-Docking-Conformations"></a>
</li><li> <strong>If I wish to reload my docked conformations and
visualise them sequentially then how can I do that? </strong>
</li></ul>
                                                                                

<p>If you select <code>File ... Save ... Range</code> from the <em>File</em> menu, 
you can write 
out each prediction of the complex to a separate PDB file - these can 
then be read back into <em>Hex</em> in subsequent sessions. 
However, these are then treated as a &quot;single&quot; molecule in <em>Hex</em>, 
rather than the seprate receptor + ligand that you started with. 
Alternatively, you can select a docking prediction, and then use 
<code>File ... Save ... Receptor</code> and <code>File ... Save ... Ligand</code>
 to save the separate 
components of the current orientation. These can then be 
read back together from the Unix command line: 
</p>
<table><tr><td>&nbsp;</td><td><pre class="example">hex receptor.pdb ligand.pdb 
</pre></td></tr></table>

<p>Unfortunately, its not possible to reload together all the predictions 
from a previous session. You would need to rerun
the calculation from the beginning to get that behaviour.
</p>
<ul>
<li><a name="index-Beowulf-Parallelisation"></a>
<a name="index-MPI"></a>
<a name="index-PVM"></a>
</li><li> <strong>Can Hex use Beowulf-style parallelisation such as 
MPI or PVM?</strong>
</li></ul>

<p>No, <em>Hex</em> can exploit only the processors ot it can see on the motherboard.
Since the basic calculation now takes only a matter of seconds, there is little
point in using multiple nodes for a single pair-wise docking calculation. If
you wish to do multiple pair-wise docking, I would recommend using Unix scripts
to spread the calculation over multiple nodes on a cluster and to collect the
results.
</p>
<ul>
<li><a name="index-Papers-Citing-Hex"></a>
</li><li> <strong>Can you recommend any papers that cite Hex? 
I would like to learn more about how other people use it.</strong>
</li></ul>

<p>Quite a few journal articles cite <em>Hex</em>. However, they generally
do not describe in detail how it was used, and so would probably not
help someone to get started. If you do wish to look at articles that 
use <em>Hex</em>, you would need to use a bibliography service to search 
for papers that cite Ritchie &amp; Kemp. 
</p>
<ul>
<li><a name="index-Missing-Ligand-_0028HETATM-bug_0029"></a>
<a name="index-Cant-See-the-Ligand"></a>
<a name="index-HETATM-record-bug"></a>
</li><li> <strong>I have both my protein and ligand in PDB files. The protein is 
seen on the screen but upon loading the ligand, I can&rsquo;t see it?</strong>
</li></ul>

<p>This bug should have been fixed in <em>Hex</em> version 4.5. If you
cannot upgrade to that version (or later), 
you probably need to edit the ligand PDB file 
to change &quot;HETATM&quot; to &quot;ATOM  &quot; (6 characters). 
Prior to version 4.5, <em>Hex</em> would
not display a molecule that contains only HETATM data. 
</p>
<ul>
<li><a name="index-Bumps-_003d-_002d1-_003f"></a>
<a name="index-Hbonds-_003d-_002d1-_003f"></a>
</li><li> <strong>What does &quot;Bumps = -1&quot; mean in the output table?
              Similarly, why do I always get Hbonds = -1?</strong>
</li></ul>

<p>The &quot;-1&quot; means nothing was calculated. You will always see this value
unless you enable <em>Backbone Bumps</em> in the <em>Postprocessing</em> 
selector in the <em>Docking Control</em> panel.
In that case, each orientation will be checked for steric clashes
between backbone atoms (the column will show the total count of
bumps). You can then use the <em>Clustering</em> control panel to put
solutions with more than a given number of bumps to the end of
the list, if you wish. Requesting molecular mechanics (MM) 
energies also causes backbone bumps checking to be turned on.
Similarly, &quot;Hbonds = -1&quot; means no hydrogen bonds were calculated.
You need to enable the <em>Show H-bonds</em> toggle in the
<em>Molecule Control</em> panel.
</p>









<hr size="6">
<a name="Subject-Index"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#FAQs" title="Previous section in reading order"> &lt; </a>]</td>
<td valign="middle" align="left">[ &gt; ]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#FAQs" title="Beginning of this chapter or previous chapter"> &lt;&lt; </a>]</td>
<td valign="middle" align="left">[<a href="#Introduction" title="Up section"> Up </a>]</td>
<td valign="middle" align="left">[ &gt;&gt; ]</td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left"> &nbsp; </td>
<td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h1 class="unnumbered">Subject Index</h1>
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a href="#Subject-Index_cp_symbol-1" class="summary-letter"><b>5</b></a>
 &nbsp; 
<br>
<a href="#Subject-Index_cp_letter-A" class="summary-letter"><b>A</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-B" class="summary-letter"><b>B</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-C" class="summary-letter"><b>C</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-D" class="summary-letter"><b>D</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-E" class="summary-letter"><b>E</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-F" class="summary-letter"><b>F</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-G" class="summary-letter"><b>G</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-H" class="summary-letter"><b>H</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-I" class="summary-letter"><b>I</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-K" class="summary-letter"><b>K</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-L" class="summary-letter"><b>L</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-M" class="summary-letter"><b>M</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-N" class="summary-letter"><b>N</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-O" class="summary-letter"><b>O</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-P" class="summary-letter"><b>P</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-R" class="summary-letter"><b>R</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-S" class="summary-letter"><b>S</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-T" class="summary-letter"><b>T</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-V" class="summary-letter"><b>V</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-W" class="summary-letter"><b>W</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-Z" class="summary-letter"><b>Z</b></a>
 &nbsp; 
</td></tr></table>
<table border="0" class="index-cp">
<tr><td></td><th align="left">Index Entry</th><th align="left"> Section</th></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_symbol-1">5</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-5D-FFT-Paper">5D FFT Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-A">A</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Acknowledgements">Acknowledgements</a></td><td valign="top"><a href="#Contacts">6.11 Contacts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Additional-Docking-Parameters">Additional Docking Parameters</a></td><td valign="top"><a href="#Additional-Docking-Parameters">5.11 Additional Docking Parameters</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Animation">Animation</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Animation-Control-Panel">Animation Control Panel</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Arrow-Keys">Arrow Keys</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Author">Author</a></td><td valign="top"><a href="#Contacts">6.11 Contacts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Axes_002c-Axis">Axes, Axis</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-B">B</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-B-Key">B Key</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Basic-Display-Styles">Basic Display Styles</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Batch-Docking-Example">Batch Docking Example</a></td><td valign="top"><a href="#Macros">6.1 Macros</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-BBSRC">BBSRC</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Beowulf-Parallelisation">Beowulf Parallelisation</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bind-error">bind error</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Biological-Symmetry">Biological Symmetry</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Bond-Widths">Bond Widths</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Broken-Ribbons">Broken Ribbons</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Bugs">Bugs</a></td><td valign="top"><a href="#Bugs-and-Known-Limitations">6.8 Bugs and Known Limitations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Bumps">Bumps</a></td><td valign="top"><a href="#Calculating-Bumps">5.5 Calculating Bumps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Bumps-_003d-_002d1-_003f">Bumps = -1 ?</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Buttons">Buttons</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Buttons-1">Buttons</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bzip2">bzip2</a></td><td valign="top"><a href="#Compressed-Files">6.4 Compressed Files</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bzip2-1">bzip2</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-bzip2-2">bzip2</a></td><td valign="top"><a href="#New-Features-in-Version-3_002e4">C.10 New Features in Version 3.4</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Bzipped-Files">Bzipped Files</a></td><td valign="top"><a href="#Compressed-Files">6.4 Compressed Files</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-C">C</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Cant-See-the-Ligand">Cant See the Ligand</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CAPRI-Blind-Trial">CAPRI Blind Trial</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CAPRI-Blind-Trial-1">CAPRI Blind Trial</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CAPRI-Paper">CAPRI Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Carbo-Index">Carbo Index</a></td><td valign="top"><a href="#Molecular-Matching">5.12 Molecular Matching</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Cartoon-Button">Cartoon Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Cartoon-Control-Panel">Cartoon Control Panel</a></td><td valign="top"><a href="#Cartoons">3.3 Cartoons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Cartoons">Cartoons</a></td><td valign="top"><a href="#Cartoons">3.3 Cartoons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Cavities">Cavities</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Changing-the-Scene-Origin">Changing the Scene Origin</a></td><td valign="top"><a href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Charge-Density">Charge Density</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Checkerboard">Checkerboard</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Citing-Hex">Citing Hex</a></td><td valign="top"><a href="#Citing-Hex">6.14 Citing Hex</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clipping">Clipping</a></td><td valign="top"><a href="#Clipping-the-Scene">3.8 Clipping the Scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clipping-Button">Clipping Button</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clipping-Clipping-the-Scene">Clipping Clipping the Scene</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clipping-Slider">Clipping Slider</a></td><td valign="top"><a href="#Clipping-the-Scene">3.8 Clipping the Scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clustering-Control-Panel">Clustering Control Panel</a></td><td valign="top"><a href="#Calculating-Bumps">5.5 Calculating Bumps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clustering-Control-Panel-1">Clustering Control Panel</a></td><td valign="top"><a href="#Clustering-Docking-Results">5.7 Clustering Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Clustering-Docking-Results">Clustering Docking Results</a></td><td valign="top"><a href="#Clustering-Docking-Results">5.7 Clustering Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Colour-Files">Colour Files</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Colour-Ramp">Colour Ramp</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Colour-Schemes">Colour Schemes</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Colour-Schemes-1">Colour Schemes</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Command-Line-Options">Command Line Options</a></td><td valign="top"><a href="#Command-Line-Options">6.6 Command Line Options</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Commercial-Licence-FAQ">Commercial Licence FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Complex">Complex</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Complex-1">Complex</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Compressed-Files">Compressed Files</a></td><td valign="top"><a href="#Compressed-Files">6.4 Compressed Files</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-connect-error">connect error</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Contacts">Contacts</a></td><td valign="top"><a href="#Contacts">6.11 Contacts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Control_002fC">Control/C</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Control_002fQ">Control/Q</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Correlation">Correlation</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-CPU-precision">CPU precision</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Crystal-Symmetry">Crystal Symmetry</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cuFFT">cuFFT</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-cuFFT-1">cuFFT</a></td><td valign="top"><a href="#nVidia-CUDA-Configuration">D. nVidia CUDA Configuration</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-D">D</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Data-Directory">Data Directory</a></td><td valign="top"><a href="#Linux-Installation">B.3 Linux Installation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-DCC-Stereo-Mode">DCC Stereo Mode</a></td><td valign="top"><a href="#nVidia-Stereo-Configuration">E. nVidia Stereo Configuration</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Disc-Cache">Disc Cache</a></td><td valign="top"><a href="#Disc-Cache">5.4 Disc Cache</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Display-Styles">Display Styles</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Distance-Sub_002dStepping">Distance Sub-Stepping</a></td><td valign="top"><a href="#Distance-Sub_002dStepping">5.2 Distance Sub-Stepping</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Control-Panel">Docking Control Panel</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Control-Panel-1">Docking Control Panel</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Control-Panel-2">Docking Control Panel</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Control-Panel-3">Docking Control Panel</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Control-Panel-4">Docking Control Panel</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Examples">Docking Examples</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Large-Molecules">Docking Large Molecules</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Molecules">Docking Molecules</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Motion">Docking Motion</a></td><td valign="top"><a href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Multiple-Structures">Docking Multiple Structures</a></td><td valign="top"><a href="#Docking-Multiple-Structures">5.9 Docking Multiple Structures</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Parameters-FAQ">Docking Parameters FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Results">Docking Results</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Review-Paper">Docking Review Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Samples">Docking Samples</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Summary-File">Docking Summary File</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Docking-Transformations-File">Docking Transformations File</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Dot-Surfaces">Dot Surfaces</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Dot-Surfaces-1">Dot Surfaces</a></td><td valign="top"><a href="#Dot-Surfaces">3.6 Dot Surfaces</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-E">E</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Editing">Editing</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Editing-Molecules">Editing Molecules</a></td><td valign="top"><a href="#Manoeuvring-Molecules">4. Manoeuvring Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Editing-Origins-and-Orientations">Editing Origins and Orientations</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Electrostatic-Correlations-FAQ">Electrostatic Correlations FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Electrostatic-Correlations-FAQ-1">Electrostatic Correlations FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Electrostatic-Potential">Electrostatic Potential</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Electrostatics">Electrostatics</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Environment-Variables">Environment Variables</a></td><td valign="top"><a href="#Environment-Variables">6.5 Environment Variables</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Environment-Variables-1">Environment Variables</a></td><td valign="top"><a href="#Linux-Installation">B.3 Linux Installation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Escape">Escape</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Euler-Angles">Euler Angles</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Examples-Directory">Examples Directory</a></td><td valign="top"><a href="#Examples-Directory">6.7 Examples Directory</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Examples-Directory-1">Examples Directory</a></td><td valign="top"><a href="#Linux-Installation">B.3 Linux Installation</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-F">F</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-F-Key">F Key</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-False-Positives">False Positives</a></td><td valign="top"><a href="#Molecular-Mechanics-Refinement">5.6 Molecular Mechanics Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-FAQs">FAQs</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Far-Plane">Far Plane</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Fast-Fourier-Transform">Fast Fourier Transform</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Feature-History">Feature History</a></td><td valign="top"><a href="#Feature-History">C. Feature History</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-FFT">FFT</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-FFT-speeds">FFT speeds</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-FFTW">FFTW</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-File-Selection-Panel">File Selection Panel</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Final-Search">Final Search</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Final-Search-1">Final Search</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-FLTK">FLTK</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Fog">Fog</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Fonts">Fonts</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Frame-Rate">Frame Rate</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Full-Screen-Mode">Full Screen Mode</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-G">G</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-G-Key">G Key</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Gaussian-Densities">Gaussian Densities</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Gaussian-Surfaces">Gaussian Surfaces</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Gaussian-Surfaces-1">Gaussian Surfaces</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Getting-Started">Getting Started</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Global-Search">Global Search</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GMP">GMP</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GPU-Paper">GPU Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GPU-precision">GPU precision</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Graphics-Menu">Graphics Menu</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GRASP">GRASP</a></td><td valign="top"><a href="#Hardcopy-Output">6.3 Hardcopy Output</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Grid-Dimensions">Grid Dimensions</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GUI">GUI</a></td><td valign="top"><a href="#How-to-Read-this-Manual">1.3 How to Read this Manual</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-GUI-Control-Toggle">GUI Control Toggle</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-gzip">gzip</a></td><td valign="top"><a href="#Compressed-Files">6.4 Compressed Files</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-gzip-1">gzip</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-gzip-2">gzip</a></td><td valign="top"><a href="#New-Features-in-Version-3_002e4">C.10 New Features in Version 3.4</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Gzipped-Files">Gzipped Files</a></td><td valign="top"><a href="#Compressed-Files">6.4 Compressed Files</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-H">H</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hammer-Button">Hammer Button</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hammer-Button-1">Hammer Button</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hardcopy-Output">Hardcopy Output</a></td><td valign="top"><a href="#Hardcopy-Output">6.3 Hardcopy Output</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Harmonic-Surface-Control-Panel">Harmonic Surface Control Panel</a></td><td valign="top"><a href="#Spherical-Harmonic-Surfaces">3.5 Spherical Harmonic Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Harmonic-Surface-Control-Panel-1">Harmonic Surface Control Panel</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Harmonics-Button">Harmonics Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Harmonics-Control-Panel">Harmonics Control Panel</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Harmonics-Control-Panel-1">Harmonics Control Panel</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hbonds-_003d-_002d1-_003f">Hbonds = -1 ?</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-HETATM-record-bug">HETATM record bug</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hetero-Atoms">Hetero Atoms</a></td><td valign="top"><a href="#Hetero-Atoms">6.2 Hetero Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hetero-Control-Panel">Hetero Control Panel</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hetero-Control-Panel-1">Hetero Control Panel</a></td><td valign="top"><a href="#Hetero-Atoms">6.2 Hetero Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hex-Licence">Hex Licence</a></td><td valign="top"><a href="#Hex-Licence">A.1 Hex Licence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hex_002c-Dictionary-Definition"><em>Hex</em>, Dictionary Definition</a></td><td valign="top"><a href="#Introduction">1. Introduction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Home-Position">Home Position</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Home-View">Home View</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-HTVS-Paper">HTVS Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Hydrogen-Atoms">Hydrogen Atoms</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-I">I</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Icosahedral-Tessellation">Icosahedral Tessellation</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-ID-Button">ID Button</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Impossible-Things">Impossible Things</a></td><td valign="top"><a href="#Introduction">1. Introduction</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Installation-Guide">Installation Guide</a></td><td valign="top"><a href="#Installation-Guide">B. Installation Guide</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intel-Linux-Version">Intel Linux Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intel-Linux-Version-1">Intel Linux Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intel-Mac-Version">Intel Mac Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intermolecular-Axis">Intermolecular Axis</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intermolecular-Axis-1">Intermolecular Axis</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Intermolecular-Distance">Intermolecular Distance</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Internal-Energies">Internal Energies</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-K">K</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Keyboard-Keys">Keyboard Keys</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Keyboard-Keys-1">Keyboard Keys</a></td><td valign="top"><a href="#Clipping-the-Scene">3.8 Clipping the Scene</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Killing-Jobs">Killing Jobs</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Kiss-FFT">Kiss FFT</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Kiss-FFT-1">Kiss FFT</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Kiss-FFT-Licence">Kiss FFT Licence</a></td><td valign="top"><a href="#Kiss-FFT-Licence">A.3 Kiss FFT Licence</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-L">L</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Laguerre-Polynomials">Laguerre Polynomials</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Licences">Licences</a></td><td valign="top"><a href="#Licences">A. Licences</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Ligand">Ligand</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Ligand-1">Ligand</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Ligand-Models">Ligand Models</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Linux-I586_002fI686-FAQ">Linux I586/I686 FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Linux-Install-FAQ">Linux Install FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-M">M</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Mac-OS-X-Version">Mac OS X Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Macro-Docking">Macro Docking</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Macro_002dSampling-Control-Panel">Macro-Sampling Control Panel</a></td><td valign="top"><a href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Macros">Macros</a></td><td valign="top"><a href="#Macros">6.1 Macros</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Macros-Bug">Macros Bug</a></td><td valign="top"><a href="#Macros-Bug">6.8.1 Macros</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Manoeuvring-Molecules">Manoeuvring Molecules</a></td><td valign="top"><a href="#Manoeuvring-Molecules">4. Manoeuvring Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Marching-Tetragons">Marching Tetragons</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Matching-Control-Panel">Matching Control Panel</a></td><td valign="top"><a href="#Molecular-Matching">5.12 Molecular Matching</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Menu-Buttons">Menu Buttons</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Menu-Buttons-1">Menu Buttons</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Mesa">Mesa</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MicroSoft-Windows-Version">MicroSoft Windows Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MinGW">MinGW</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Missing-Ligand-_0028HETATM-bug_0029">Missing Ligand (HETATM bug)</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MKL">MKL</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MKL-1">MKL</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecular-Centroids">Molecular Centroids</a></td><td valign="top"><a href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecular-Graphics">Molecular Graphics</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecular-Matching">Molecular Matching</a></td><td valign="top"><a href="#Molecular-Matching">5.12 Molecular Matching</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecular-Mechanics-Refinement">Molecular Mechanics Refinement</a></td><td valign="top"><a href="#Molecular-Mechanics-Refinement">5.6 Molecular Mechanics Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecular-Similarity">Molecular Similarity</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecule-Control-Panel">Molecule Control Panel</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecule-Control-Panel-1">Molecule Control Panel</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molecule-Control-Panel-2">Molecule Control Panel</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MolMol">MolMol</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Molscript">Molscript</a></td><td valign="top"><a href="#Hardcopy-Output">6.3 Hardcopy Output</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Mouse-Buttons">Mouse Buttons</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Movie-Type">Movie Type</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MPI">MPI</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-MS_002dWindows">MS-Windows</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-N">N</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Near-Plane">Near Plane</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-NMR-Docking">NMR Docking</a></td><td valign="top"><a href="#Docking-Multiple-Structures">5.9 Docking Multiple Structures</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-NMR-Models">NMR Models</a></td><td valign="top"><a href="#Docking-Multiple-Structures">5.9 Docking Multiple Structures</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-No-Docking-Solutions-FAQ">No Docking Solutions FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Non_002dStandard-Residues-Bug">Non-Standard Residues Bug</a></td><td valign="top"><a href="#Non_002dStandard-Residues">6.8.2 Non-Standard Residues</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Non_002dStandard-Residues-FAQ">Non-Standard Residues FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Nvidia-3D-Vision">Nvidia 3D Vision</a></td><td valign="top"><a href="#nVidia-Stereo-Configuration">E. nVidia Stereo Configuration</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nVidia-CUDA-Configuration">nVidia CUDA Configuration</a></td><td valign="top"><a href="#nVidia-CUDA-Configuration">D. nVidia CUDA Configuration</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nVidia-Drivers">nVidia Drivers</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nVidia-Graphics-Cards">nVidia Graphics Cards</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-nVidia-Stereo-Configuration">nVidia Stereo Configuration</a></td><td valign="top"><a href="#nVidia-Stereo-Configuration">E. nVidia Stereo Configuration</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-O">O</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Control-Panel">Orientation Control Panel</a></td><td valign="top"><a href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Control-Panel-1">Orientation Control Panel</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Control-Panel-2">Orientation Control Panel</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Control-Panel-3">Orientation Control Panel</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orientation-Editing">Orientation Editing</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Origin-Editing">Origin Editing</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Origin-Editing-Button">Origin Editing Button</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orthographic-Projection">Orthographic Projection</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Orthographic-Projection-1">Orthographic Projection</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-P">P</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-P-Key">P Key</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Page_002fDown">Page/Down</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Page_002fUp">Page/Up</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Papers-Citing-Hex">Papers Citing Hex</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Parametric-Functions">Parametric Functions</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-PDB">PDB</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-People">People</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Perspective-Projection">Perspective Projection</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Perspective-Projection-1">Perspective Projection</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Picking">Picking</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Pipe-Output">Pipe Output</a></td><td valign="top"><a href="#Hardcopy-Output">6.3 Hardcopy Output</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Pipes-Directory">Pipes Directory</a></td><td valign="top"><a href="#Linux-Installation">B.3 Linux Installation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Pointer-Mode">Pointer Mode</a></td><td valign="top"><a href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Pointer-Mode-Button">Pointer Mode Button</a></td><td valign="top"><a href="#Function-Buttons">2.2 Function Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Polar-Hydrogens">Polar Hydrogens</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Post-Processing-Options">Post Processing Options</a></td><td valign="top"><a href="#Calculating-Bumps">5.5 Calculating Bumps</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Postscript-Fonts-Bug">Postscript Fonts Bug</a></td><td valign="top"><a href="#Postscript-Fonts">6.8.3 Postscript Fonts</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-PowerPC-Version">PowerPC Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Projection-Control-Panel">Projection Control Panel</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-PVM">PVM</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-R">R</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-R-Slider">R Slider</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Radial-Cutoff">Radial Cutoff</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Receptor">Receptor</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Receptor-1">Receptor</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Receptor-Models">Receptor Models</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Reference-Energy-FAQ">Reference Energy FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Reference-Orientation">Reference Orientation</a></td><td valign="top"><a href="#Docking-Molecules">5. Docking Molecules</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-References">References</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Refine-Docking-FAQ">Refine Docking FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Relative-Permittivity">Relative Permittivity</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Reloading-Docking-Conformations">Reloading Docking Conformations</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Ribbon-Cartoons">Ribbon Cartoons</a></td><td valign="top"><a href="#Cartoons">3.3 Cartoons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Rotational-Search">Rotational Search</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-S">S</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-S-Key">S Key</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Saving-Docking-Ranges">Saving Docking Ranges</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Saving-Docking-Results">Saving Docking Results</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Saving-Docking-Results-1">Saving Docking Results</a></td><td valign="top"><a href="#Saving-Docking-Results">5.8 Saving Docking Results</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Scene-Centre">Scene Centre</a></td><td valign="top"><a href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-SGI-version">SGI version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Shape-Matching-Scores">Shape Matching Scores</a></td><td valign="top"><a href="#Molecular-Matching">5.12 Molecular Matching</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Short_002dCut-Buttons">Short-Cut Buttons</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Sidechains-Button">Sidechains Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Sidechains-Button-1">Sidechains Button</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Sigma-Surface">Sigma Surface</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Silicon-Graphics-Version">Silicon Graphics Version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Slider">Slider</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Soft-Potentials">Soft Potentials</a></td><td valign="top"><a href="#Molecular-Mechanics-Refinement">5.6 Molecular Mechanics Refinement</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Model-Button">Solid Model Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Model-Control-Panel">Solid Model Control Panel</a></td><td valign="top"><a href="#Solid-Models">3.2 Solid Models</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Models">Solid Models</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Models-1">Solid Models</a></td><td valign="top"><a href="#Solid-Models">3.2 Solid Models</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Motion-Button">Solid Motion Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Surface-Button">Solid Surface Button</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Surfaces">Solid Surfaces</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solid-Surfaces-1">Solid Surfaces</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Solvent-Culling">Solvent Culling</a></td><td valign="top"><a href="#Hetero-Atoms">6.2 Hetero Atoms</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Source-Code-FAQ">Source Code FAQ</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-cube">Spherical Harmonic cube</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Functions">Spherical Harmonic Functions</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Rotation">Spherical Harmonic Rotation</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Surfaces">Spherical Harmonic Surfaces</a></td><td valign="top"><a href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Surfaces-1">Spherical Harmonic Surfaces</a></td><td valign="top"><a href="#Molecular-Graphics">3. Molecular Graphics</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Surfaces-2">Spherical Harmonic Surfaces</a></td><td valign="top"><a href="#Spherical-Harmonic-Surfaces">3.5 Spherical Harmonic Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spherical-Harmonic-Surfaces-3">Spherical Harmonic Surfaces</a></td><td valign="top"><a href="#Spherical-Harmonic-Surfaces">3.5 Spherical Harmonic Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Spin-Angle">Spin Angle</a></td><td valign="top"><a href="#Animation">3.9 Animation</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-startup-crash">startup crash</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stereo-Displays">Stereo Displays</a></td><td valign="top"><a href="#Keyboard-Keys">2.3 Keyboard Keys</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stereo-Displays-1">Stereo Displays</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stereo-Displays-2">Stereo Displays</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stereo-Parallax">Stereo Parallax</a></td><td valign="top"><a href="#Stereo-Displays">3.11 Stereo Displays</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Steric-Scan">Steric Scan</a></td><td valign="top"><a href="#Rotational-Search">5.1 Rotational Search</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Steric-Scan-1">Steric Scan</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stride">Stride</a></td><td valign="top"><a href="#Cartoons">3.3 Cartoons</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stride-1">Stride</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stride-2">Stride</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Stride-Licence">Stride Licence</a></td><td valign="top"><a href="#Stride-Licence">A.2 Stride Licence</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Sun-version">Sun version</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Superposing-Molecules">Superposing Molecules</a></td><td valign="top"><a href="#Molecular-Matching">5.12 Molecular Matching</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Surface-Control-Panel">Surface Control Panel</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Surface-Skins">Surface Skins</a></td><td valign="top"><a href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Symmetry-Types">Symmetry Types</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-T">T</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Tau-Surface">Tau Surface</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Technical-Information">Technical Information</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Tessellation-Mesh-Order">Tessellation Mesh Order</a></td><td valign="top"><a href="#Spherical-Harmonic-Surfaces">3.5 Spherical Harmonic Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Test-Functions">Test Functions</a></td><td valign="top"><a href="#The-Test-Functions">6.9 The Test Functions</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-TexInfo">TexInfo</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-TexInfo-1">TexInfo</a></td><td valign="top"><a href="#New-Features-in-Version-3_002e4">C.10 New Features in Version 3.4</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-The-Gimp">The Gimp</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Translation-Matrices">Translation Matrices</a></td><td valign="top"><a href="#Solid-Surfaces">3.4 Solid Surfaces</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Translation-Matrices-1">Translation Matrices</a></td><td valign="top"><a href="#Disc-Cache">5.4 Disc Cache</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Translation-Matrices-Paper">Translation Matrices Paper</a></td><td valign="top"><a href="#References">6.13 References</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Twist-Samples">Twist Samples</a></td><td valign="top"><a href="#Docking-Examples">5.3 Docking Examples</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-V">V</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-van-der-Waals-Spheres">van der Waals Spheres</a></td><td valign="top"><a href="#Solid-Models">3.2 Solid Models</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Viewing-Large-Molecules">Viewing Large Molecules</a></td><td valign="top"><a href="#Basic-Display-Styles">3.1 Basic Display Styles</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-VMD">VMD</a></td><td valign="top"><a href="#FAQs">F. Frequently Asked Questions</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-W">W</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-Windows-XP">Windows XP</a></td><td valign="top"><a href="#Technical-Information">6.10 Technical Information</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
<tr><th><a name="Subject-Index_cp_letter-Z">Z</a></th><td></td><td></td></tr>
<tr><td></td><td valign="top"><a href="#index-zlib">zlib</a></td><td valign="top"><a href="#Acknowledgements">6.12 Acknowledgements</a></td></tr>
<tr><td></td><td valign="top"><a href="#index-Zoom">Zoom</a></td><td valign="top"><a href="#Getting-Started">2. Getting Started</a></td></tr>
<tr><td colspan="3"> <hr></td></tr>
</table>
<table><tr><th valign="top">Jump to: &nbsp; </th><td><a href="#Subject-Index_cp_symbol-1" class="summary-letter"><b>5</b></a>
 &nbsp; 
<br>
<a href="#Subject-Index_cp_letter-A" class="summary-letter"><b>A</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-B" class="summary-letter"><b>B</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-C" class="summary-letter"><b>C</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-D" class="summary-letter"><b>D</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-E" class="summary-letter"><b>E</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-F" class="summary-letter"><b>F</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-G" class="summary-letter"><b>G</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-H" class="summary-letter"><b>H</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-I" class="summary-letter"><b>I</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-K" class="summary-letter"><b>K</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-L" class="summary-letter"><b>L</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-M" class="summary-letter"><b>M</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-N" class="summary-letter"><b>N</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-O" class="summary-letter"><b>O</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-P" class="summary-letter"><b>P</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-R" class="summary-letter"><b>R</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-S" class="summary-letter"><b>S</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-T" class="summary-letter"><b>T</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-V" class="summary-letter"><b>V</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-W" class="summary-letter"><b>W</b></a>
 &nbsp; 
<a href="#Subject-Index_cp_letter-Z" class="summary-letter"><b>Z</b></a>
 &nbsp; 
</td></tr></table>

<hr size="6">
<a name="SEC_Contents"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h1>Table of Contents</h1>
<div class="contents">

<ul class="toc">
  <li><a name="toc-Introduction" href="#Introduction">1. Introduction</a>
  <ul class="toc">
    <li><a name="toc-What-the-Heck-is-Hex_003f" href="#What-the-Heck-is-Hex_003f">1.1 What the Heck is <em>Hex</em>?</a></li>
    <li><a name="toc-New-Features-in-Hex-Version-8_002e0_002e0" href="#New-Features-in-Hex-Version-8_002e0_002e0">1.2 New Features in <em>Hex</em> Version 8.0.0</a></li>
    <li><a name="toc-How-to-Read-this-Manual" href="#How-to-Read-this-Manual">1.3 How to Read this Manual</a></li>
  </ul></li>
  <li><a name="toc-Getting-Started-1" href="#Getting-Started">2. Getting Started</a>
  <ul class="toc">
    <li><a name="toc-Short_002dCut-Buttons-1" href="#Short_002dCut-Buttons">2.1 Short-Cut Buttons</a></li>
    <li><a name="toc-Function-Buttons-1" href="#Function-Buttons">2.2 Function Buttons</a></li>
    <li><a name="toc-Keyboard-Keys-1" href="#Keyboard-Keys">2.3 Keyboard Keys</a></li>
  </ul></li>
  <li><a name="toc-Molecular-Graphics-1" href="#Molecular-Graphics">3. Molecular Graphics</a>
  <ul class="toc">
    <li><a name="toc-Basic-Display-Styles-1" href="#Basic-Display-Styles">3.1 Basic Display Styles</a></li>
    <li><a name="toc-Solid-Models-1" href="#Solid-Models">3.2 Solid Models</a></li>
    <li><a name="toc-Cartoons-1" href="#Cartoons">3.3 Cartoons</a></li>
    <li><a name="toc-Solid-Surfaces-1" href="#Solid-Surfaces">3.4 Solid Surfaces</a></li>
    <li><a name="toc-Spherical-Harmonic-Surfaces-1" href="#Spherical-Harmonic-Surfaces">3.5 Spherical Harmonic Surfaces</a></li>
    <li><a name="toc-Dot-Surfaces-1" href="#Dot-Surfaces">3.6 Dot Surfaces</a></li>
    <li><a name="toc-Changing-the-Scene-Origin-1" href="#Changing-the-Scene-Origin">3.7 Changing the Scene Origin</a></li>
    <li><a name="toc-Clipping-the-Scene-1" href="#Clipping-the-Scene">3.8 Clipping the Scene</a></li>
    <li><a name="toc-Animation-1" href="#Animation">3.9 Animation</a></li>
    <li><a name="toc-Full-Screen-Mode-1" href="#Full-Screen-Mode">3.10 Full Screen Mode</a></li>
    <li><a name="toc-Stereo-Displays-1" href="#Stereo-Displays">3.11 Stereo Displays</a></li>
  </ul></li>
  <li><a name="toc-Manoeuvring-Molecules-1" href="#Manoeuvring-Molecules">4. Manoeuvring Molecules</a>
  <ul class="toc">
    <li><a name="toc-Editing-Origins-and-Orientations-1" href="#Editing-Origins-and-Orientations">4.1 Editing Origins and Orientations</a></li>
  </ul></li>
  <li><a name="toc-Docking-Molecules-1" href="#Docking-Molecules">5. Docking Molecules</a>
  <ul class="toc">
    <li><a name="toc-Rotational-Search-1" href="#Rotational-Search">5.1 Rotational Search</a></li>
    <li><a name="toc-Distance-Sub_002dStepping-1" href="#Distance-Sub_002dStepping">5.2 Distance Sub-Stepping</a></li>
    <li><a name="toc-Docking-Examples-1" href="#Docking-Examples">5.3 Docking Examples</a></li>
    <li><a name="toc-Disc-Cache-1" href="#Disc-Cache">5.4 Disc Cache</a></li>
    <li><a name="toc-Calculating-Bumps-1" href="#Calculating-Bumps">5.5 Calculating Bumps</a></li>
    <li><a name="toc-Molecular-Mechanics-Refinement-1" href="#Molecular-Mechanics-Refinement">5.6 Molecular Mechanics Refinement</a></li>
    <li><a name="toc-Clustering-Docking-Results-1" href="#Clustering-Docking-Results">5.7 Clustering Docking Results</a></li>
    <li><a name="toc-Saving-Docking-Results-1" href="#Saving-Docking-Results">5.8 Saving Docking Results</a></li>
    <li><a name="toc-Docking-Multiple-Structures-1" href="#Docking-Multiple-Structures">5.9 Docking Multiple Structures</a></li>
    <li><a name="toc-Docking-Very-Large-Molecules-_0028Macro-Docking_0029" href="#Macro-Docking">5.10 Docking Very Large Molecules (Macro Docking)</a></li>
    <li><a name="toc-Additional-Docking-Parameters-1" href="#Additional-Docking-Parameters">5.11 Additional Docking Parameters</a></li>
    <li><a name="toc-Molecular-Matching-1" href="#Molecular-Matching">5.12 Molecular Matching</a></li>
  </ul></li>
  <li><a name="toc-Miscellaneous-1" href="#Miscellaneous">6. Miscellaneous</a>
  <ul class="toc">
    <li><a name="toc-Macros-2" href="#Macros">6.1 Macros</a></li>
    <li><a name="toc-Hetero-Atoms-1" href="#Hetero-Atoms">6.2 Hetero Atoms</a></li>
    <li><a name="toc-Hardcopy-Output-1" href="#Hardcopy-Output">6.3 Hardcopy Output</a></li>
    <li><a name="toc-Compressed-Files-1" href="#Compressed-Files">6.4 Compressed Files</a></li>
    <li><a name="toc-Environment-Variables-1" href="#Environment-Variables">6.5 Environment Variables</a></li>
    <li><a name="toc-Command-Line-Options-1" href="#Command-Line-Options">6.6 Command Line Options</a>
    <ul class="toc">
      <li><a name="toc-Loading-Three-Structures-Together" href="#Loading-Three-Structures-Together">6.6.1 Loading Three Structures Together</a></li>
      <li><a name="toc-Loading-and-Superposing-Two-Structures" href="#Loading-and-Superposing-Two-Structures">6.6.2 Loading and Superposing Two Structures</a></li>
      <li><a name="toc-Docking-with-Macro-File-And-Log-File" href="#Docking-with-Macro-File-And-Log-File">6.6.3 Docking with Macro File And Log File</a></li>
      <li><a name="toc-Docking-with-File-I_002fO-Redirection" href="#Docking-with-File-I_002fO-Redirection">6.6.4 Docking with File I/O Redirection</a></li>
      <li><a name="toc-Serialising-Several-Batch-Jobs" href="#Serialising-Several-Batch-Jobs">6.6.5 Serialising Several Batch Jobs</a></li>
      <li><a name="toc-Using-Hex-as-a-Web-Browser" href="#Using-Hex-as-a-Web-Browser">6.6.6 Using <em>Hex</em> as a Web Browser</a></li>
    </ul></li>
    <li><a name="toc-Examples-Directory-1" href="#Examples-Directory">6.7 Examples Directory</a></li>
    <li><a name="toc-Bugs-and-Known-Limitations-1" href="#Bugs-and-Known-Limitations">6.8 Bugs and Known Limitations</a>
    <ul class="toc">
      <li><a name="toc-Macros-1" href="#Macros-Bug">6.8.1 Macros</a></li>
      <li><a name="toc-Non_002dStandard-Residues-1" href="#Non_002dStandard-Residues">6.8.2 Non-Standard Residues</a></li>
      <li><a name="toc-Postscript-Fonts-1" href="#Postscript-Fonts">6.8.3 Postscript Fonts</a></li>
    </ul></li>
    <li><a name="toc-The-Test-Functions-1" href="#The-Test-Functions">6.9 The Test Functions</a></li>
    <li><a name="toc-Technical-Information-1" href="#Technical-Information">6.10 Technical Information</a></li>
    <li><a name="toc-Contacts-1" href="#Contacts">6.11 Contacts</a></li>
    <li><a name="toc-Acknowledgements" href="#Acknowledgements">6.12 Acknowledgements</a></li>
    <li><a name="toc-References-1" href="#References">6.13 References</a></li>
    <li><a name="toc-Citing-Hex-1" href="#Citing-Hex">6.14 Citing Hex</a></li>
  </ul></li>
  <li><a name="toc-Licences-1" href="#Licences">A. Licences</a>
  <ul class="toc">
    <li><a name="toc-Hex-Licence-1" href="#Hex-Licence">A.1 Hex Licence</a></li>
    <li><a name="toc-Stride-Licence-1" href="#Stride-Licence">A.2 Stride Licence</a></li>
    <li><a name="toc-Kiss-FFT-Licence-1" href="#Kiss-FFT-Licence">A.3 Kiss FFT Licence</a></li>
  </ul></li>
  <li><a name="toc-Installation-Guide-1" href="#Installation-Guide">B. Installation Guide</a>
  <ul class="toc">
    <li><a name="toc-Microsoft-Windows-Installation" href="#Microsoft-Windows-Installation">B.1 Microsoft Windows Installation</a></li>
    <li><a name="toc-Mac-OS-X-Installation" href="#Mac-OS-X-Installation">B.2 Mac OS X Installation</a></li>
    <li><a name="toc-Linux-Installation" href="#Linux-Installation">B.3 Linux Installation</a></li>
  </ul></li>
  <li><a name="toc-Feature-History-1" href="#Feature-History">C. Feature History</a>
  <ul class="toc">
    <li><a name="toc-New-Features-in-Hex-Version-6_002e3" href="#New-Features-in-Hex-Version-6_002e3">C.1 New Features in <em>Hex</em> Version 6.3</a></li>
    <li><a name="toc-New-Features-in-Hex-Version-6_002e1" href="#New-Features-in-Hex-Version-6_002e1">C.2 New Features in <em>Hex</em> Version 6.1</a></li>
    <li><a name="toc-New-Features-in-Hex-Version-6_002e0" href="#New-Features-in-Hex-Version-6_002e0">C.3 New Features in <em>Hex</em> Version 6.0</a></li>
    <li><a name="toc-New-Features-in-Hex-Version-5_002e1" href="#New-Features-in-Hex-Version-5_002e1">C.4 New Features in <em>Hex</em> Version 5.1</a></li>
    <li><a name="toc-New-Features-in-Hex-Version-5_002e0" href="#New-Features-in-Hex-Version-5_002e0">C.5 New Features in <em>Hex</em> Version 5.0</a></li>
    <li><a name="toc-New-Features-in-Version-4_002e5" href="#New-Features-in-Version-4_002e5">C.6 New Features in Version 4.5</a></li>
    <li><a name="toc-New-Features-in-Version-4_002e2" href="#New-Features-in-Version-4_002e2">C.7 New Features in Version 4.2</a></li>
    <li><a name="toc-New-Features-in-Version-4_002e1" href="#New-Features-in-Version-4_002e1">C.8 New Features in Version 4.1</a></li>
    <li><a name="toc-New-Features-in-Version-4_002e0" href="#New-Features-in-Version-4_002e0">C.9 New Features in Version 4.0</a></li>
    <li><a name="toc-New-Features-in-Version-3_002e4" href="#New-Features-in-Version-3_002e4">C.10 New Features in Version 3.4</a></li>
    <li><a name="toc-New-Features-in-Version-3_002e3" href="#New-Features-in-Version-3_002e3">C.11 New Features in Version 3.3</a></li>
    <li><a name="toc-New-Features-in-Version-3_002e2" href="#New-Features-in-Version-3_002e2">C.12 New Features in Version 3.2</a></li>
    <li><a name="toc-New-Features-in-Version-3_002e1" href="#New-Features-in-Version-3_002e1">C.13 New Features in Version 3.1</a></li>
    <li><a name="toc-New-Features-in-Version-3_002e0" href="#New-Features-in-Version-3_002e0">C.14 New Features in Version 3.0</a></li>
  </ul></li>
  <li><a name="toc-nVidia-CUDA-Configuration-1" href="#nVidia-CUDA-Configuration">D. nVidia CUDA Configuration</a></li>
  <li><a name="toc-nVidia-Stereo-Configuration-1" href="#nVidia-Stereo-Configuration">E. nVidia Stereo Configuration</a></li>
  <li><a name="toc-Frequently-Asked-Questions" href="#FAQs">F. Frequently Asked Questions</a></li>
  <li><a name="toc-Subject-Index" href="#Subject-Index">Subject Index</a></li>
</ul>
</div>
<hr size="1">
<a name="SEC_About"></a>
<table cellpadding="1" cellspacing="1" border="0">
<tr><td valign="middle" align="left">[<a href="#Introduction" title="Cover (top) of document">Top</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_Contents" title="Table of contents">Contents</a>]</td>
<td valign="middle" align="left">[<a href="#Subject-Index" title="Index">Index</a>]</td>
<td valign="middle" align="left">[<a href="#SEC_About" title="About (help)"> ? </a>]</td>
</tr></table>
<h1>About This Document</h1>
<p>
  This document was generated by <em>Dave Ritchie</em> on <em>October 27, 2013</em> using <a href="http://www.nongnu.org/texi2html/"><em>texi2html 1.82</em></a>.
</p>
<p>
  The buttons in the navigation panels have the following meaning:
</p>
<table border="1">
  <tr>
    <th> Button </th>
    <th> Name </th>
    <th> Go to </th>
    <th> From 1.2.3 go to</th>
  </tr>
  <tr>
    <td align="center"> [ &lt; ] </td>
    <td align="center">Back</td>
    <td>Previous section in reading order</td>
    <td>1.2.2</td>
  </tr>
  <tr>
    <td align="center"> [ &gt; ] </td>
    <td align="center">Forward</td>
    <td>Next section in reading order</td>
    <td>1.2.4</td>
  </tr>
  <tr>
    <td align="center"> [ &lt;&lt; ] </td>
    <td align="center">FastBack</td>
    <td>Beginning of this chapter or previous chapter</td>
    <td>1</td>
  </tr>
  <tr>
    <td align="center"> [ Up ] </td>
    <td align="center">Up</td>
    <td>Up section</td>
    <td>1.2</td>
  </tr>
  <tr>
    <td align="center"> [ &gt;&gt; ] </td>
    <td align="center">FastForward</td>
    <td>Next chapter</td>
    <td>2</td>
  </tr>
  <tr>
    <td align="center"> [Top] </td>
    <td align="center">Top</td>
    <td>Cover (top) of document</td>
    <td> &nbsp; </td>
  </tr>
  <tr>
    <td align="center"> [Contents] </td>
    <td align="center">Contents</td>
    <td>Table of contents</td>
    <td> &nbsp; </td>
  </tr>
  <tr>
    <td align="center"> [Index] </td>
    <td align="center">Index</td>
    <td>Index</td>
    <td> &nbsp; </td>
  </tr>
  <tr>
    <td align="center"> [ ? ] </td>
    <td align="center">About</td>
    <td>About (help)</td>
    <td> &nbsp; </td>
  </tr>
</table>

<p>
  where the <strong> Example </strong> assumes that the current position is at <strong> Subsubsection One-Two-Three </strong> of a document of the following structure:
</p>

<ul>
  <li> 1. Section One
    <ul>
      <li>1.1 Subsection One-One
        <ul>
          <li>...</li>
        </ul>
      </li>
      <li>1.2 Subsection One-Two
        <ul>
          <li>1.2.1 Subsubsection One-Two-One</li>
          <li>1.2.2 Subsubsection One-Two-Two</li>
          <li>1.2.3 Subsubsection One-Two-Three &nbsp; &nbsp;
            <strong>&lt;== Current Position </strong></li>
          <li>1.2.4 Subsubsection One-Two-Four</li>
        </ul>
      </li>
      <li>1.3 Subsection One-Three
        <ul>
          <li>...</li>
        </ul>
      </li>
      <li>1.4 Subsection One-Four</li>
    </ul>
  </li>
</ul>

<hr size="1">
<p>
 <font size="-1">
  This document was generated by <em>Dave Ritchie</em> on <em>October 27, 2013</em> using <a href="http://www.nongnu.org/texi2html/"><em>texi2html 1.82</em></a>.
 </font>
 <br>

</p>
</body>
</html>
